<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>序列比对其他相关问题 | 思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1576309192768">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">

<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">



  </head>
  <body>
    <div class="main">
      <div class="main-content">
        <div class="site-header">
  <a href="https://kaopubear.top/">
  <img class="avatar" src="https://kaopubear.top//images/avatar.png?v=1576309192768" alt="">
  </a>
  <h1 class="site-title">
    思考问题的熊
  </h1>
  <p class="site-description">
    很高兴和你相遇<br>这里正在记录我的所思所学
  </p>
  <div class="menu-container">
    
      
        <a href="/" class="menu">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu">
          标签
        </a>
      
    
      
        <a href="/post/about" class="menu">
          关于
        </a>
      
    
      
        <a href="/post/resource" class="menu">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu">
          项目
        </a>
      
    
  </div>
  <div class="social-container">
    
      
    
      
    
      
    
      
    
      
    
  </div>
</div>

        <div class="post-detail">
          <article class="post">
            <h2 class="post-title">
              序列比对其他相关问题
            </h2>
            <div class="post-info">
              <span>
                2017-08-14
              </span>
              <span>
                6 min read
              </span>
              
                <a href="https://kaopubear.top//tag/DvI6ypPftI" class="post-tag">
                  # 生物信息
                </a>
              
            </div>
            
              <img class="post-feature-image" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-d77ac044d4b6743f.jpg" alt="">
            
            <div class="post-content-wrapper">
              <div class="post-content">
                <h2 id="全局比对-global-alignment">全局比对 Global Alignment</h2>
<p>由芝加哥的 Needleman 和 Wunsch 两位于上个世纪 70 年代初提出，常被称之为 Needleman-Wunsch 算法。算法针对用户指定的打分函数，确定性地找出两条序列间的最优比对。</p>
<p>Needle-Wunsch 算法对两条序列所有残基进行全局比对的局限性。</p>
<ul>
<li>功能相关的蛋白之间虽然可能在整体序列上相差甚远， 却常常会具有相同的功能域</li>
<li>序列片段能够独立发挥特定的生物学功能，却在不同蛋白之间相当保守</li>
<li>仅靠全局比对的算法无法发现这样的片段</li>
<li>内含子的发现使得在做核酸水平的序列比对时必须要正确处理内含子导致的大片段的差异</li>
</ul>
<!--more-->
<h2 id="局部比对-local-alignment">局部比对 Local Alignment</h2>
<p>1981 年，物理学家 Temple Smith 和数学家 Michael Waterman 在 Journal of Molcular Biology 上发表了一篇仅有四页的文章，提出 Smith-Waterman 局部比对算法</p>
<p>race back begins at the highest score inthe matrix and continues until you reach 0。And also the secondary best alignment</p>
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-1ccb0e4b771c757c.jpg" alt="img"></figure>
<p>核心思想是给分数增加了下限 0 分</p>
<p>所有的回溯都是局部的，所有的最终比对也是局部的。引入止损下限，差异扩大之后重启比对，找到局部水平的相似性</p>
<h2 id="空位罚分的改进">空位罚分的改进</h2>
<p>Affine gap penalty</p>
<ul>
<li>opening a gap receives a score ofd</li>
<li>extending a gap receives a score of e</li>
<li>Penalty = d + (n-1)* e</li>
</ul>
<p>有限向量机<br>
<img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-7102a385725a4e40.jpg" alt="alt"></p>
<p>扩展公式</p>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-80a9e576ea651253.jpg" alt="alt"></figure>
<h2 id="全局比对的时间复杂性">全局比对的时间复杂性</h2>
<p>O(mn) 正比于 m*n</p>
<h2 id="全基因组比对">全基因组比对</h2>
<h3 id="同源-homology-的分类">同源 homology 的分类</h3>
<ul>
<li>
<p>直系同源 ortholog 来自于不同的物种，演化过程中基因没有丢失，各物种中都有</p>
<p>chaining</p>
</li>
<li>
<p>旁系同源 paralog 来自于一个物种内部基因组的复制</p>
<p>netting</p>
</li>
</ul>
<h2 id="ngs-sequence-alignment">NGS: Sequence alignment</h2>
<p>Map the large numbers of short reads to a reference genome</p>
<ul>
<li>In a broader sense: Identify similar sequences (DNA, RNA, or protein) inconsequence of functional, structural, or evolutionary relationships between the them</li>
<li>Applications: Genome assembly, SNP detection, homology search, etc</li>
</ul>
<p>short: greater search sensitivity</p>
<p>large: faster search speed</p>
<h3 id="in-exact-alignment">In-exact alignment</h3>
<p>BWA 和 bowtie 的相关算法，大大减少了对服务器的要求</p>
<p><strong>如何快速的知道某段序列大约在基因组的哪个位置</strong></p>
<ol>
<li>如何定义大约这个概念</li>
</ol>
<ul>
<li>Hamming Distance or Sequence Similarity</li>
<li>Ungapped vs Gapped Global vs Local</li>
<li>All positions or the single best</li>
</ul>
<ol start="2">
<li>Efficiency depends on the data characteristics &amp; goals</li>
</ol>
<ul>
<li>Smith-Waterman: Exhaustive search for optimal alignments</li>
<li>BLAST: Hash-table based homology searches</li>
<li>Bowtie: BWT alignment for short read mapping</li>
</ul>
<h3 id="bwt-算法">BWT 算法</h3>
<p>核心是绕最后一个序列转圈</p>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-d77ac044d4b6743f.jpg" alt="alt"></figure>
<ul>
<li>
<p>先给每一个 T 做 rotations, 再进行 sort, 生成 bw 矩阵，最后一列从头到尾就是 BWT</p>
</li>
<li>
<p>回溯</p>
<p>给每一个 T 的字母一个出现次数的排序，图示如下</p>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-7f6db222b92d5b06.jpg" alt="alt"></figure>
</li>
</ul>
<h3 id="suffix后缀-arrays">Suffix（后缀） Arrays</h3>
<blockquote>
<p>类似于电话本中的索引结构</p>
</blockquote>
<p><strong>What if we need to check many queries</strong></p>
<ul>
<li>
<p>We don't need to check every page of the phone book to find 'Ma'</p>
</li>
<li>
<p>Sorting alphabetically lets us immediately skip 96% (25/26) of the book withoutany loss in accuracy</p>
</li>
</ul>
<p><strong>Sorting the genome: Suffix Array (Manber &amp; Myers1991)</strong></p>
<ul>
<li>Sort every suffix of the genome</li>
</ul>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-1bf7617aa186f44f.jpg" alt="alt"></figure>
<p>所有具有相同 prefix（前缀）的 suffixes（后缀）会聚在一起，这样就可以进行类似于二分法的排除</p>
<p><strong>全基因组建立 index</strong></p>
<figure data-type="image" tabindex="6"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-2728285ca4cda020.jpg" alt="alt"></figure>
<ul>
<li>
<p>An array of integers giving the startingpositions of suffixes of a string inlexicographical order</p>
</li>
<li>
<p>从中间的 index 开始找，过滤一半</p>
</li>
</ul>
<figure data-type="image" tabindex="7"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-8c9e641581731988.jpg" alt="alt"></figure>
<p><strong>效率</strong></p>
<p>Total Runtime: O(m log n)</p>
<ul>
<li>More complicated, but much faster!</li>
<li>Looking up a query loops 32 times instead of 3B</li>
</ul>
<p>Searching the array is very fast, but it takes time to construct</p>
<ul>
<li>This time will be amortized over many, many searches</li>
<li>Run it once &quot;overnight&quot; and save it away for all future queries</li>
<li>非常消耗内存</li>
</ul>
<h3 id="blastdot-matrix">BLAST/Dot matrix</h3>
<p><strong>Indexing-based local alignment</strong></p>
<figure data-type="image" tabindex="8"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-71d09c0f3d71dd54.jpg" alt="alt"></figure>
<p><strong>Basic Local Alignment Search Tool</strong></p>
<blockquote>
<p>围绕最优比对路径进行计算</p>
</blockquote>
<p>BLAST Ideas 核心思想：Seeding‐and‐extending</p>
<ul>
<li>
<p>Find matches (seed) between the query and subject 找到高度相似的小片段，种子</p>
</li>
<li>
<p>Extend seed into High Scoring Segment Pairs (HSPs) 向两端延伸并进行比对</p>
<p>Run Smith‐Waterman algorithm on the specified region only</p>
</li>
<li>
<p>Assess the reliability of the alignment 打分评估</p>
</li>
</ul>
<p>将序列切分，在数据库中定位候选序列和位置</p>
<figure data-type="image" tabindex="9"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-e4f4ca9c765403dd.jpg" alt="alt"></figure>
<p>得到候选序列和查询序列的 heatmap</p>
<p>去掉零散的 hit, 直留下对角线，形成 hit cluster</p>
<p>以 hit cluster 为基础向左右进行延伸直到分数不符合要去</p>
<p>在扩展的区域进行局部比对</p>
<figure data-type="image" tabindex="10"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-2f6e770b76db1ad6.jpg" alt="alt"></figure>
<p><strong>blast 加速</strong></p>
<ul>
<li>标记低复杂度，易产生假阳性</li>
<li>考虑与种子相似的邻居单字</li>
</ul>
<p><strong>分数评估，避免随机因素</strong></p>
<p>E value</p>
<figure data-type="image" tabindex="11"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-83fdec479a8701f3.jpg" alt="alt"></figure>
<ul>
<li>n 数据库大小</li>
<li>k 和打分矩阵相关</li>
<li>m 长度</li>
<li>s 比对的分数</li>
</ul>
<p>在随机情况下获得比当前分数高的可能比对条数，不是概率是个望值。p 为 0.05 时，E 也是 0.05。</p>
<p>BLAST 是一种启发式算法，不确定有最优解</p>
<figure data-type="image" tabindex="12"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-6fa0263571430462.jpg" alt="alt"></figure>
<p>只在有效区域应用动态规划算法</p>
<hr>
<figure data-type="image" tabindex="13"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="14"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

              </div>
              <div class="toc-container">
                <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E5%85%A8%E5%B1%80%E6%AF%94%E5%AF%B9-global-alignment">全局比对 Global Alignment</a></li>
<li><a href="#%E5%B1%80%E9%83%A8%E6%AF%94%E5%AF%B9-local-alignment">局部比对 Local Alignment</a></li>
<li><a href="#%E7%A9%BA%E4%BD%8D%E7%BD%9A%E5%88%86%E7%9A%84%E6%94%B9%E8%BF%9B">空位罚分的改进</a></li>
<li><a href="#%E5%85%A8%E5%B1%80%E6%AF%94%E5%AF%B9%E7%9A%84%E6%97%B6%E9%97%B4%E5%A4%8D%E6%9D%82%E6%80%A7">全局比对的时间复杂性</a></li>
<li><a href="#%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%AF%94%E5%AF%B9">全基因组比对</a>
<ul>
<li><a href="#%E5%90%8C%E6%BA%90-homology-%E7%9A%84%E5%88%86%E7%B1%BB">同源 homology 的分类</a></li>
</ul>
</li>
<li><a href="#ngs-sequence-alignment">NGS: Sequence alignment</a>
<ul>
<li><a href="#in-exact-alignment">In-exact alignment</a></li>
<li><a href="#bwt-%E7%AE%97%E6%B3%95">BWT 算法</a></li>
<li><a href="#suffix%E5%90%8E%E7%BC%80-arrays">Suffix（后缀） Arrays</a></li>
<li><a href="#blastdot-matrix">BLAST/Dot matrix</a></li>
</ul>
</li>
</ul>
</li>
</ul>

              </div>
            </div>
          </article>
        </div>

        
          <div class="next-post">
            <div class="next">下一篇</div>
            <a href="https://kaopubear.top//post/2017-08-13-longxing-bioinfo-blast">
              <h3 class="post-title">
                序列比对与动态规划
              </h3>
            </a>
          </div>
        

        

        <div class="site-footer">
  欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a> | 
  <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
</div>

<script>
  hljs.initHighlightingOnLoad()

  let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

  // This should probably be throttled.
  // Especially because it triggers during smooth scrolling.
  // https://lodash.com/docs/4.17.10#throttle
  // You could do like...
  // window.addEventListener("scroll", () => {
  //    _.throttle(doThatStuff, 100);
  // });
  // Only not doing it here to keep this Pen dependency-free.

  window.addEventListener("scroll", event => {
    let fromTop = window.scrollY;

    mainNavLinks.forEach((link, index) => {
      let section = document.getElementById(decodeURI(link.hash).substring(1));
      let nextSection = null
      if (mainNavLinks[index + 1]) {
        nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
      }
      if (section.offsetTop <= fromTop) {
        if (nextSection) {
          if (nextSection.offsetTop > fromTop) {
            link.classList.add("current");
          } else {
            link.classList.remove("current");    
          }
        } else {
          link.classList.add("current");
        }
      } else {
        link.classList.remove("current");
      }
    });
  });

</script>

      </div>
    </div>
  </body>
</html>
