<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1578673710658">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.remixicon.com/releases/v1.3.1/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<link rel="alternate" type="application/atom+xml" title="思考问题的熊 - Atom Feed" href="https://kaopubear.top//atom.xml">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div id="app" class="main px-4 flex flex-col lg:flex-row">
      <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="remixicon-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1578673710658" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E4%B8%8D%E5%90%8C%E5%B1%82%E6%AC%A1%E7%9A%84%E8%A1%A8%E8%A7%82">不同层次的表观</a></li>
<li><a href="#%E8%A1%A8%E8%A7%82%E5%AE%9E%E4%BE%8B">表观实例</a></li>
<li><a href="#%E8%A1%A8%E8%A7%82%E9%81%97%E4%BC%A0%E5%AE%9A%E4%B9%89">表观遗传定义</a></li>
<li><a href="#%E8%A1%A8%E8%A7%82%E5%9F%BA%E5%9B%A0%E7%BB%84-epigenome">表观基因组 epigenome</a></li>
<li><a href="#epigenetic-mechanisms">Epigenetic mechanisms</a>
<ul>
<li><a href="#dna-methylation">DNA methylation</a></li>
<li><a href="#histone-modification">Histone modification</a></li>
</ul>
</li>
<li><a href="#dna-methylation-2">DNA methylation</a>
<ul>
<li><a href="#human-%E7%89%B9%E5%BE%81">human 特征</a></li>
<li><a href="#%E6%A4%8D%E7%89%A9%E4%B8%AD%E6%9C%89%E7%89%B9%E6%AE%8A%E7%9A%84-rddm">植物中有特殊的 RdDM</a></li>
<li><a href="#bisulfite-sequencing">bisulfite sequencing</a></li>
<li><a href="#methylated-dna-immunoprecipitation-medip">Methylated DNA immunoprecipitation (MeDIP)</a></li>
<li><a href="#methyl-sensitive-restriction-enzymes-mres">Methyl-sensitive restriction enzymes (MREs)</a></li>
</ul>
</li>
<li><a href="#%E7%BB%84%E8%9B%8B%E7%99%BD%E4%BF%AE%E9%A5%B0">组蛋白修饰</a>
<ul>
<li><a href="#h3k4me3-%E5%90%AB%E4%B9%89">H3K4me3 含义</a></li>
<li><a href="#%E7%BB%84%E8%9B%8B%E7%99%BD%E4%BF%AE%E9%A5%B0%E5%8A%9F%E8%83%BD">组蛋白修饰功能</a></li>
<li><a href="#%E5%A6%82%E4%BD%95%E6%95%B4%E5%90%88%E4%BC%97%E5%A4%9A%E7%9A%84-chip-seq-%E6%95%B0%E6%8D%AE">如何整合众多的 ChIP-seq 数据</a>
<ul>
<li><a href="#chromhmm">ChromHMM</a></li>
<li><a href="#segway">Segway</a></li>
</ul>
</li>
<li><a href="#%E5%9F%BA%E5%9B%A0%E6%B5%8F%E8%A7%88%E5%99%A8%E5%B1%95%E7%A4%BA">基因浏览器展示</a></li>
</ul>
</li>
<li><a href="#human-%E8%A1%A8%E8%A7%82%E6%95%B0%E6%8D%AE%E5%BA%93">human 表观数据库</a>
<ul>
<li><a href="#encode">encode</a></li>
<li><a href="#roadmap">roadmap</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="remixicon-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
      <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
        <h1 class="text-3xl font-bold lg:mt-16">表观遗传与生物信息</h1>
        <div class="text-sm text-gray-700 lg:my-8">
          2017-08-17 / 6 min read
        </div>
        
          <img class="post-feature-image rounded-lg mx-auto my-4" src="ttps://raw.githubusercontent.com/fei0810/image-host/master/img/177622-38231516d4752d09.jpg" alt="">
        
        <div class="post-content yue">
          <h2 id="不同层次的表观">不同层次的表观</h2>
<!--more-->
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-354dda16cd5cdf70.jpg" alt="" loading="lazy"></figure>
<ul>
<li>Broadly, features at different levels of chromatin organization are generally associated with inactive (off) or active (on) transcription.</li>
<li>From the top, genomic DNA is methylated(Me) on cytosine bases in specific contexts and is packaged into nucleosomes, which vary in histone composition and histone modifications (for example, histone H3 lysine 9 trimethylation (H3K9me3)); these features constitute the primary layer of chromatinstructure.</li>
<li>Here, different histone modifications are indicated by coloured dots and histone variants such as H2A.Z are brown.</li>
<li>DNA in chromatin may remain accessible to DNA-binding proteins such as transcription factors (TFs) and RNA polymerase II (RNAPII)or may be further compacted.</li>
<li>Chromatin can also organize into higher-order structuressuch as nuclear lamina-associated domains and transcription factories.</li>
<li>DOI:10.1038/nrg2905</li>
</ul>
<h2 id="表观实例">表观实例</h2>
<p>基因印迹</p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a618faca55fcf197.jpg" alt="" loading="lazy"></figure>
<h2 id="表观遗传定义">表观遗传定义</h2>
<ul>
<li>A mitotically or meiotically heritable state of different gene activity and expression (phenotype) that is independent of differences in DNA sequence (genotype) – based on Conrad Waddington, 1942</li>
<li>The sum of the alterations to the chromatin template that collectively establish and propagate different patterns of gene expression (transcription) and silencing from the same genome.</li>
<li>Epigenetic changes influence the phenotype without altering the genotype.</li>
<li>While epigenetics often refers to the study of single genes or sets of genes, epigenomics refers to more global analyses of epigenetic changes across the entire genome.</li>
</ul>
<h2 id="表观基因组-epigenome">表观基因组 epigenome</h2>
<ul>
<li>The complete number,location, and types ofepigenetic modificationsthat occur in a given cell</li>
</ul>
<h2 id="epigenetic-mechanisms">Epigenetic mechanisms</h2>
<h3 id="dna-methylation">DNA methylation</h3>
<ul>
<li>Normal cells — role in gene expression and chromosome stability</li>
<li>Cancer cells — consequences of aberrant hypo- and hyper-methylation</li>
</ul>
<h3 id="histone-modification">Histone modification</h3>
<ul>
<li>Normal cells — the histone code</li>
<li>Cancer cells — consequences of altered histone modifying enzymes</li>
</ul>
<p>Interaction between DNA methylation, histone modifications, and other players such as non-coding RNAs</p>
<p>Cell and tissue type specificity</p>
<p>Gene-environment interaction, disease susceptibility</p>
<h2 id="dna-methylation-2">DNA methylation</h2>
<h3 id="human-特征">human 特征</h3>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a3496c1dd5bb3f73.jpg" alt="" loading="lazy"></figure>
<ul>
<li>DNA methylation occurs mostly in CG context, but also non-CG sequences (mCHG, mCHH)</li>
<li>DNA methylation is generally depleted from promoters, and enriched within gene bodies</li>
<li>DNA methylation is partially depleted at regulatory element</li>
</ul>
<h3 id="植物中有特殊的-rddm">植物中有特殊的 RdDM</h3>
<h3 id="bisulfite-sequencing">bisulfite sequencing</h3>
<p><strong>基本原理</strong></p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-9cfabba1d47bd07f.jpg" alt="" loading="lazy"></figure>
<ul>
<li>DNA 的重亚硫酸盐转化来检测未甲基化的胞嘧啶</li>
<li>在文库制备过程中，重亚硫酸盐将未甲基化的胞嘧啶转化成尿嘧啶</li>
<li>转化后的碱基在测序中被识别为胸腺嘧啶（PCR 后），并通过序列计数来确定甲基化胞嘧啶的比例</li>
</ul>
<p><strong>比对的过程需要进行转换</strong></p>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-ab2b76773d283de7.jpg" alt="" loading="lazy"></figure>
<p><strong>常用 mapping 软件算法</strong></p>
<ul>
<li>bismark</li>
</ul>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-1a2c6471b8e64af5.jpg" alt="" loading="lazy"></figure>
<ul>
<li>bsmap</li>
</ul>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-ffb9789d4a025013.jpg" alt="" loading="lazy"></figure>
<ul>
<li>
<p>Bisulfite seed table, using the original seed and bisulfite variants as keys and corresponding coordinates in the reference genome as values. Each read was looked up in the seed table for potential mapping positions.</p>
</li>
<li>
<p>A positional specific mask of the corresponding reference sequence was generated by setting 01 to C(light blue) and 11 to A, G, T(black). The original read was masked by a bitwise AND operation with the positional specific mask.</p>
</li>
<li>
<p>The reference sequence and the masked read were compared with a bitwise XOR operation. Non-zero XOR results were counted as mismatches (red). Bisulfite alignment is marked in green.</p>
</li>
</ul>
<h3 id="methylated-dna-immunoprecipitation-medip">Methylated DNA immunoprecipitation (MeDIP)</h3>
<ul>
<li>甲基化 DNA 免疫共沉淀测序</li>
<li>use 5-methylcytosine antibody</li>
<li>富集甲基化的 DNA 序列</li>
<li>有甲基化的地方会有 peak</li>
<li>不能精确到单碱基分辨率，这能知道一段区域的甲基化程度</li>
</ul>
<h3 id="methyl-sensitive-restriction-enzymes-mres">Methyl-sensitive restriction enzymes (MREs)</h3>
<ul>
<li>
<p>Unmethylated CpGs are identified by sequencing size selected fragments from parallel DNA digestions with the methyl-sensitive restriction enzymes (MREs)</p>
</li>
<li>
<p>没有甲基化的地方会有 peak</p>
</li>
<li>
<p>甲基化敏感性限制酶测序 (MRE-seq) 技术主要利用甲基化敏感性限制性酶 (methylation-sensitive restriction enzyme, MRE) 对基因组 DNA 进行切割，未甲基化位点可以被酶切而产生不同长度的片段，然后，结合高通量测序技术获得序列的甲基化信息</p>
</li>
<li>
<p>和 MeDIP 结合使用</p>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-b2ecf644450ff688.jpg" alt="" loading="lazy"></figure>
</li>
<li>
<p>plant DOI:10.1146/annurev-arplant-043014-114633</p>
</li>
<li>
<p>human DOI:10.1038/nature08514</p>
</li>
</ul>
<h2 id="组蛋白修饰">组蛋白修饰</h2>
<h3 id="h3k4me3-含义">H3K4me3 含义</h3>
<figure data-type="image" tabindex="9"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-b1902497c38fc017.jpg" alt="" loading="lazy"></figure>
<h3 id="组蛋白修饰功能">组蛋白修饰功能</h3>
<figure data-type="image" tabindex="10"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-22de42c4286ff453.jpg" alt="" loading="lazy"></figure>
<ul>
<li>Promoters, gene bodies, an enhancer and a boundary element are indicated on a schematic genomic region.</li>
<li>Active promoters are commonly marked by H3K4me2, H3K4me3, acetylation (ac), and H2A.Z.</li>
<li>Transcribed regions are enriched for H3K36me3 and H3K79me2.</li>
<li>Repressed genes may be located in large domains of H3K9me2 and/or H3K9me3 or H3K27me3.</li>
<li>Enhancers are relatively enriched for H3K4me1, H3K4me2, H3K27ac and the histone acetyltransferase p300.</li>
<li>CTCF binds many sites that may function as boundary elements, insulators or structural scaffolds</li>
</ul>
<h3 id="如何整合众多的-chip-seq-数据">如何整合众多的 ChIP-seq 数据</h3>
<p><strong>半自动基因注释</strong></p>
<figure data-type="image" tabindex="11"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-38231516d4752d09.jpg" alt="" loading="lazy"></figure>
<p><strong>两种常用算法</strong></p>
<h4 id="chromhmm">ChromHMM</h4>
<figure data-type="image" tabindex="12"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-34fa1a2cd2666169.jpg" alt="" loading="lazy"></figure>
<p>输入参数</p>
<figure data-type="image" tabindex="13"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-40c25e82cee9682d.jpg" alt="" loading="lazy"></figure>
<ul>
<li>
<p>算法是 HMM，具体隐马尔可夫模型参考另一篇笔记</p>
</li>
<li>
<p>属于 HMM 中的 learning 类问题</p>
</li>
<li>
<p>使用迭代方法是 EM</p>
</li>
<li>
<p>200 bp resolution</p>
</li>
<li>
<p>Bernoulli to model emission</p>
</li>
<li>
<p>Posterior decoding</p>
</li>
<li>
<p>Single model for all cell types</p>
</li>
</ul>
<h4 id="segway">Segway</h4>
<figure data-type="image" tabindex="14"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-128e2ca0197855f0.jpg" alt="" loading="lazy"></figure>
<ul>
<li>基于贝叶斯动态网络</li>
<li>1 bp resolution</li>
<li>Gaussian to model emission</li>
<li>Viterbi decoding</li>
<li>One model per cell type</li>
</ul>
<h3 id="基因浏览器展示">基因浏览器展示</h3>
<figure data-type="image" tabindex="15"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-e1d1e808beafe838.jpg" alt="" loading="lazy"></figure>
<p>具体学习地址</p>
<ul>
<li>
<p><a href="http://www.ensembl.org/info/genome/funcgen/regulatory_segmentation.html">ensembl</a></p>
</li>
<li>
<p><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=602267329_ZKudaZlz5EEJC7GtzQakQ3aDfF1w&amp;c=chr21&amp;g=wgEncodeAwgSegmentation">Genome Segmentations from ENCODE</a></p>
</li>
</ul>
<h2 id="human-表观数据库">human 表观数据库</h2>
<h3 id="encode">encode</h3>
<ul>
<li>网站 <a href="https://www.encodeproject.org/">https://www.encodeproject.org/</a></li>
<li>数据下载地址<a href="http://genome.ucsc.edu/ENCODE/downloads.html">http://genome.ucsc.edu/ENCODE/downloads.html</a></li>
</ul>
<h3 id="roadmap">roadmap</h3>
<ul>
<li>网站 <a href="http://www.roadmapepigenomics.org/">http://www.roadmapepigenomics.org/</a></li>
</ul>
<hr>
<figure data-type="image" tabindex="16"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="17"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

        </div>

        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/DvI6ypPftI">
            <span class="flex-auto">生物信息</span>
          </a>
        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/MoSquIwnNqz">
            <span class="flex-auto">表观遗传</span>
          </a>
        


        <div class="flex justify-between py-8">
          
            <div class="prev-post">
              <a href="https://kaopubear.top/post/2017-08-18-longxing-bioinfo-tf">
                <h3 class="post-title">
                  <i class="remixicon-arrow-left-line"></i>
                  转录调控与生物信息
                </h3>
              </a>
            </div>
          

          
            <div class="next-post">
              <a href="https://kaopubear.top/post/2017-08-16-longxing-bioinfo-markov">
                <h3 class="post-title">
                  马尔科夫模型
                  <i class="remixicon-arrow-right-line"></i>
                </h3>
              </a>
            </div>
          
        </div>

        
          
            <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
<script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>

<div id="gitalk-container"></div>

<script>

  var gitalk = new Gitalk({
    clientID: '77b0e0937753a24c7510',
    clientSecret: 'd312ec269767c6310a14f017359734b75a4dc4fc',
    repo: 'fei0810.github.io',
    owner: 'fei0810',
    admin: ['fei0810'],
    id: (location.pathname).substring(0, 49),      // Ensure uniqueness and length less than 50
    distractionFreeMode: false  // Facebook-like distraction free mode
  })

  gitalk.render('gitalk-container')

</script>

          

          
        

      </div>
    </div>

    <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()

let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
  </body>
</html>
