<html>

<head>
  <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1583596426161">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/remixicon@2.3.0/fonts/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


</head>

<body>
  <div id="app" class="main px-4 flex flex-col lg:flex-row">
    <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="ri-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1583596426161" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E5%88%8A%E9%A6%96%E8%AF%AD">刊首语</a></li>
<li><a href="#%E4%B8%93%E4%B8%9A%E6%96%87%E7%8C%AE">专业文献</a>
<ul>
<li><a href="#the-bioinformatics-toolbox-for-circrna-discovery-and-analysis">The Bioinformatics Toolbox for circRNA Discovery and Analysis</a></li>
<li><a href="#quality-control-and-processing-of-nascent-rna-profiling-data">Quality control and processing of nascent RNA profiling data</a></li>
<li><a href="#pareameters-a-tool-for-computational-inference-of-plant-mirnamrna-targeting-rules-using-small-rna-and-degradome-sequencing-data">PAREameters: a tool for computational inference of plant miRNA–mRNA targeting rules using small RNA and degradome sequencing data</a></li>
</ul>
</li>
<li><a href="#%E6%95%B0%E6%8D%AE%E5%BA%93">数据库</a>
<ul>
<li><a href="#r%E7%A4%BE%E5%8C%BA%E7%9A%84%E5%8F%91%E5%B1%95%E6%8E%A2%E7%B4%A2">R社区的发展探索</a></li>
</ul>
</li>
<li><a href="#%E5%A5%BD%E4%B9%A6%E5%A5%BD%E6%96%87">好书好文</a>
<ul>
<li><a href="#r%E7%9A%8420%E5%B9%B4">R的20年</a></li>
<li><a href="#10%E4%B8%AA%E8%8A%82%E7%9C%81%E6%97%B6%E9%97%B4%E7%9A%84r%E6%93%8D%E4%BD%9C%E6%8A%80%E5%B7%A7">10个节省时间的R操作技巧</a></li>
<li><a href="#mac%E8%A7%A6%E6%8E%A7%E6%9D%BFforcetouch%E5%8F%AF%E4%BB%A5%E5%81%9A%E4%BB%80%E4%B9%88">Mac触控板ForceTouch可以做什么</a></li>
</ul>
</li>
<li><a href="#%E5%AD%A6%E4%B9%A0%E7%B4%A0%E6%9D%90">学习素材</a>
<ul>
<li><a href="#lncrna%E5%AE%9A%E9%87%8F%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9">lncRNA定量注意事项</a></li>
<li><a href="#canadian-bioinformatics-workshops">Canadian Bioinformatics Workshops</a></li>
<li><a href="#ggplot2-%E6%95%99%E7%A8%8B%E5%90%88%E9%9B%86">ggplot2 教程合集</a></li>
<li><a href="#ggplot2-elegant-graphics-for-data-analysis">ggplot2: Elegant Graphics for Data Analysis</a></li>
</ul>
</li>
<li><a href="#%E5%BD%B1%E9%9F%B3%E6%8E%A8%E8%8D%90">影音推荐</a>
<ul>
<li><a href="#%E7%90%86%E8%A7%A3-rnaseq-%E6%98%AF%E5%A6%82%E4%BD%95%E5%B7%A5%E4%BD%9C%E7%9A%84">理解 RNAseq 是如何工作的</a></li>
</ul>
</li>
<li><a href="#%E5%B7%A5%E5%85%B7">工具</a>
<ul>
<li><a href="#datawrapper">Datawrapper</a></li>
<li><a href="#antv">AntV</a></li>
</ul>
</li>
<li><a href="#%E6%8A%80%E5%B7%A7">技巧</a>
<ul>
<li><a href="#%E5%BE%AE%E4%BF%A1%E5%8F%8C%E5%BC%80">微信双开</a></li>
</ul>
</li>
<li><a href="#%E8%AE%A2%E9%98%85">订阅</a></li>
<li><a href="#%E4%BA%A4%E6%B5%81">交流</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="ri-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">加入<a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a><br ><a href="https://tinyletter.com/kaopubear" target="_blank">邮件订阅博客内容</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
    <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
      <h1 class="text-3xl font-bold lg:mt-16">靠谱熊的每周分享 第2期</h1>
      <div class="text-sm text-gray-700 lg:my-8">
        2020-03-06 / 9 min read
      </div>
      
      <img class="post-feature-image rounded-lg mx-auto my-4" src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306200456.jpg" alt="">
      
      <div class="post-content yue">
        <h2 id="刊首语">刊首语</h2>
<p>本周把自己一年多没更新的个人公众号(思考问题的熊)更新了一次，结果很快就有了五十多条留言。<br>
很多人都在反馈自己过去三四年发生和经历的事情，很神奇的感受，看似没有交集但是却也没有断了联系。</p>
<!-- more -->
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181149.png" alt="" loading="lazy"></figure>
<p>不知道你有没有一个自己很久没有留意却突然发现还有人在关注的东西，不妨分享一下。</p>
<h2 id="专业文献">专业文献</h2>
<h3 id="the-bioinformatics-toolbox-for-circrna-discovery-and-analysis">The Bioinformatics Toolbox for circRNA Discovery and Analysis</h3>
<p>Circular RNAs (circRNAs) are a unique class of RNA molecule identified more than 40 years ago which are produced by a covalent linkage via back-splicing of linear RNA. Recent advances in sequencing technologies and bioinformatics tools have led directly to an ever-expanding field of types and biological functions of circRNAs. In parallel with technological developments, practical applications of circRNAs have arisen including their utilization as biomarkers of human disease. Currently, circRNA-associated bioinformatics tools can support projects including circRNA annotation, circRNA identification and network analysis of competing endogenous RNA (ceRNA). In this review, we collected about 100 circRNA-associated bioinformatics tools and summarized their current attributes and capabilities. We also performed network analysis and text mining on circRNA tool publications in order to reveal trends in their ongoing development.**</p>
<ul>
<li>PMID: <strong>32103237</strong></li>
<li>DOI: <a href="https://doi.org/10.1093/bib/bbaa001">10.1093/bib/bbaa001</a></li>
</ul>
<p>circRNA 发展历史 <br>
<img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181246.png" alt="" loading="lazy"><br>
circRNA 分析工具<br>
<img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181241.png" alt="" loading="lazy"></p>
<h3 id="quality-control-and-processing-of-nascent-rna-profiling-data">Quality control and processing of nascent RNA profiling data</h3>
<p>Experiments that profile nascent RNA are growing in popularity; however, there is no standard analysis pipeline to uniformly process the data and assess quality. Here, we introduce PEPPRO, a comprehensive, scalable workflow for GRO-seq, PRO-seq, and ChRO-seq data. PEPPRO produces uniform processed output files for downstream analysis, including alignment files, signal tracks, and count matrices. Furthermore, PEPPRO simplifies downstream analysis by using a standard project definition format which can be read using metadata APIs in R and Python. For quality control, PEPPRO provides several novel statistics and plots, including assessments of adapter abundance, RNA integrity, library complexity, nascent RNA purity, and run-on efficiency. PEPPRO is restartable and fault-tolerant, records copious logs, and provides a web-based project report for navigating results. It can be run on local hardware or using any cluster resource manager, using either native software or our provided modular Linux container environment. PEPPRO is thus a robust and portable first step for genomic nascent RNA analysis.<br>
Publisher URL: <a href="http://biorxiv.org/cgi/content/short/2020.02.27.956110v1?rss=1&amp;utm_source=researcher_app&amp;utm_medium=referral&amp;utm_campaign=RESR_MRKT_Researcher_inbound">http://biorxiv.org/cgi/content/short/2020.02.27.956110v1</a><br>
DOI: <a href="https://doi.org/10.1101/2020.02.27.956110">https://doi.org/10.1101/2020.02.27.956110</a></p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181233.png" alt="" loading="lazy"></figure>
<h3 id="pareameters-a-tool-for-computational-inference-of-plant-mirnamrna-targeting-rules-using-small-rna-and-degradome-sequencing-data">PAREameters: a tool for computational inference of plant miRNA–mRNA targeting rules using small RNA and degradome sequencing data</h3>
<p>MicroRNAs (miRNAs) are short, non-coding RNAs that modulate the translation-rate of messenger RNAs (mRNAs) by directing the RNA-induced silencing complex to sequence-specific targets. In plants, this typically results in cleavage and subsequent degradation of the mRNA. Degradome sequencing is a high-throughput technique developed to capture cleaved mRNA fragments and thus can be used to support miRNA target prediction. The current criteria used for miRNA target prediction were inferred on a limited number of experimentally validated <em>A. thaliana</em> interactions and were adapted to fit these specific interactions; thus, these fixed criteria may not be optimal across all datasets (organisms, tissues or treatments). We present a new tool, PAREameters, for inferring targeting criteria from small RNA and degradome sequencing datasets. We evaluate its performance using a more extensive set of experimentally validated interactions in multiple <em>A. thaliana</em> datasets. We also perform comprehensive analyses to highlight and quantify the differences between subsets of miRNA–mRNA interactions in model and non-model organisms. Our results show increased sensitivity in <em>A. thaliana</em> when using the PAREameters inferred criteria and that using data-driven criteria enables the identification of additional interactions that further our understanding of the RNA silencing pathway in both model and non-model organisms.<br>
Publisher URL: <a href="https://academic.oup.com/nar/article/48/5/2258/5707202?rss=1&amp;utm_source=researcher_app&amp;utm_medium=referral&amp;utm_campaign=RESR_MRKT_Researcher_inbound">https://academic.oup.com/nar/article/48/5/2258/5707202</a><br>
DOI: <a href="https://doi.org/10.1093/nar/gkz1234">https://doi.org/10.1093/nar/gkz1234</a></p>
<figure data-type="image" tabindex="3"><img src="https://cdn.nlark.com/yuque/0/2020/png/233470/1583487016630-47ed7b6f-b7ed-4187-bd29-4c624db14a93.png#align=left&amp;display=inline&amp;height=959&amp;name=image.png&amp;originHeight=1918&amp;originWidth=757&amp;size=245920&amp;status=done&amp;style=none&amp;width=378.5" alt="" loading="lazy"></figure>
<h2 id="数据库">数据库</h2>
<h3 id="r社区的发展探索">R社区的发展探索</h3>
<p><a href="https://benubah.github.io/r-community-explorer/rugs.html">https://benubah.github.io/r-community-explorer/rugs.html</a></p>
<p>该页面总结了 R 语言近年来社区的发展情况，女性用户的分析以及历届Google Summer of Code中R相关的项目。</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181624.png" alt="" loading="lazy"></figure>
<h2 id="好书好文">好书好文</h2>
<h3 id="r的20年">R的20年</h3>
<p>最近有 <a href="https://jozef.io/r921-happy-birthday-r/">一篇文章</a> 从三个纬度介绍了 R 这 20 年的发展。</p>
<ul>
<li>R 在这些年里发展速度有多快</li>
<li>2000年以来发布了多少个 R 包</li>
<li>包的下载量如何增长</li>
</ul>
<h3 id="10个节省时间的r操作技巧">10个节省时间的R操作技巧</h3>
<p><a href="https://drkeithmcnulty.com/author/keithmcnulty/">Keith McNulty</a> 总结了可以节省时间的<a href="https://towardsdatascience.com/ten-time-saving-r-hacks-b411add26b96">10个R操作技巧</a></p>
<ul>
<li>Downloading and reading files straight from source</li>
<li>Storing your credentials for regular use</li>
<li>RStudio’s shortcut keys</li>
<li>Global chunk options in RMarkdown</li>
<li>Easy pasting of ggplots with the <code>patchwork</code> package</li>
<li>Smoother dependency management using <code>Renv</code></li>
<li>Multitask with RStudio’s Jobs</li>
<li>Rename all variables in scope</li>
<li>Using <code>.</code> to keep piping</li>
<li>Immediately invoked display</li>
</ul>
<h3 id="mac触控板forcetouch可以做什么">Mac触控板ForceTouch可以做什么</h3>
<p>9to5mac 发布了一篇关于的教程，介绍了<a href="https://9to5mac.com/2020/03/05/mac-macbook-what-can-force-touch-trackpad-do/">Mac触控板ForceTouch可以做什么</a>，包括如何设置ForceTouch以及哪些应用可以使用ForceTouch。</p>
<h2 id="学习素材">学习素材</h2>
<h3 id="lncrna定量注意事项">lncRNA定量注意事项</h3>
<p>关于如何定量lncRNA，19年有一篇发表在Gigascience的文章进行了不同方法的测试，整体的结论如下：</p>
<blockquote>
<p>Pseudoalignment methods and RSEM detect more lncRNAs and correlate highly with simulated ground truth. On the contrary, HTSeq and featureCounts often underestimate lncRNA expression. Antisense lncRNAs are poorly quantified by alignment-based gene quantification methods, which can be improved using stranded protocols and pseudoalignment methods.</p>
</blockquote>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6897288/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6897288/</a></p>
<p>伪比对方法和 RSEM 检测到更多的 lncRNAs，反义 lncRNAs 通过基于比对的基因量化方法难以量化。关于这个问题在 GitHub 有一个简单的讨论：<a href="https://github.com/alexdobin/STAR/issues/848">https://github.com/alexdobin/STAR/issues/848</a></p>
<h3 id="canadian-bioinformatics-workshops">Canadian Bioinformatics Workshops</h3>
<p><a href="https://bioinformaticsdotca.github.io/">https://bioinformaticsdotca.github.io/</a><br>
Canadian Bioinformatics Workshops 旗下的生物信息教程合集基本上所有内容都可以公开获取</p>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181636.png" alt="" loading="lazy"></figure>
<h3 id="ggplot2-教程合集">ggplot2 教程合集</h3>
<p>Erik Gahner Larsen 整理了目前可以找到的 ggplot2 相关教程合集 <a href="https://github.com/erikgahner/awesome-ggplot2">awesome-ggplot2</a> <br>
各个角度的教程都有，可以学习。</p>
<h3 id="ggplot2-elegant-graphics-for-data-analysis">ggplot2: Elegant Graphics for Data Analysis</h3>
<p>Hadley Wickham 主编的<a href="https://ggplot2-book.org/index.html">第三版 ggplot2 教程</a>已经在路上了，服气。</p>
<h2 id="影音推荐">影音推荐</h2>
<h3 id="理解-rnaseq-是如何工作的">理解 RNAseq 是如何工作的</h3>
<p>这个简短的视频说明了 RNAseq 是如何工作的，以及如何研究中使用它。Jose Manuel Garcia Manteiga  解释了如何有效地设计 RNAseq 实验以及可以从这项技术中获得的结果。</p>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181304.png" alt="image.png" loading="lazy"></figure>
<p>十分钟的视频把转录组分析相关内容的方方面面今本都涉及到了，很喜欢这样的讲课方式。如果有机会做一些类似的小视频就好了，不过机会在哪呢╮(￣▽￣)╭<br>
(我从油管做了一个搬运放到了B站，嵌入了英文字幕）</p>
<iframe src="https://player.bilibili.com/player.html?aid=93932327" frameborder="no" allowfullscreen="true"></iframe>
<h2 id="工具">工具</h2>
<h3 id="datawrapper">Datawrapper</h3>
<p><a href="https://www.datawrapper.de/">https://www.datawrapper.de/</a><br>
Datawrapper是一种简单的可视化工具，无需编程基础。大量的媒体和记者都在使用这个工具进行数据的可视化，官方也提供了详细的教程。</p>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181312.png" alt="" loading="lazy"></figure>
<p>如何使用datawrapper <a href="https://academy.datawrapper.de/">https://academy.datawrapper.de/</a></p>
<h3 id="antv">AntV</h3>
<p>和语雀出自同一家，AntV 是蚂蚁金服全新一代数据可视化解决方案，致力于提供一套简单方便、专业可靠、无限可能的数据可视化最佳实践。近日其 G2 正式发布了 4.0 版本。</p>
<p><a href="https://antv.vision/zh">https://antv.vision/zh</a></p>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181318.png" alt="" loading="lazy"></figure>
<h2 id="技巧">技巧</h2>
<h3 id="微信双开">微信双开</h3>
<p>微信这个让人又恨又不得不用的东西，有时候不仅不得不用，可能一次还得用两个。这里就涉及到了微信双开的问题。在android系统上，现在多数国产厂商都适配了微信双开。至于Windows和macOS其实也都有自己的方法（其实都是命令行启动），供各位参考。</p>
<p>Windows 上的使用方式<s>就是以极快的速度连续点击微信图标</s>有两种思路，我比较喜欢的是在微信安装目录新建一个bat文件，里面写两行 <code>start WeChat.ext</code> 。在把这个文件做一个快捷方式到桌面。也可以直接写绝对路径，把文件直接放到桌面。</p>
<figure data-type="image" tabindex="9"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200306181328.png" alt="image.png" loading="lazy"></figure>
<p>macOS本身是利用 nohup 命令让微信进行后台运行，如下所示，原则上，这种方法可以同时开启N个微信。</p>
<pre><code class="language-shell">nohup /Applications/WeChat.app/Contents/MacOS/WeChat &gt; /dev/null 2&gt;&amp;1 &amp;
</code></pre>
<h2 id="订阅">订阅</h2>
<p>该周刊每周六发布，暂时作为「素材分享学习小组」内部资料，我们会挑选部分内容进行公开分享。<br>
完成前期测试运行后将择期通过博客和公众号等形式同步更新，届时将发布具体的订阅形式。</p>
<p>因为该周刊采用「轮值编辑」的形式，我个人的每周分享如果没有特殊情况将会继续更新在自己的<a href="https://kaopubear.top">博客</a>等平台，同时也可以通过<a href="https://tinyletter.com/kaopubear">邮件订阅</a>。</p>
<h2 id="交流">交流</h2>
<p>如果想要加入「素材分享学习小组」可以参考<a href="https://kaopubear.top/post/2020-01-19-refgroup2020/">申请说明，</a>如果想和我们交流，欢迎在评论区留言，我们下周见！</p>

      </div>
      
      <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 "
        href="https://kaopubear.top/tag/tg-VHYKad/">
        <span class="flex-auto">工具</span>
      </a>
      
      <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 "
        href="https://kaopubear.top/tag/DvI6ypPftI/">
        <span class="flex-auto">生物信息</span>
      </a>
      
      <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 "
        href="https://kaopubear.top/tag/2IWlJgErkt/">
        <span class="flex-auto">日常</span>
      </a>
      


      <div class="flex justify-between py-8">
        
        <div class="prev-post">
          <a href="https://kaopubear.top/post/2020-03-07-HanlonRazor/">
            <h3 class="post-title">
              <i class="ri-arrow-left-line"></i>
              非蠢既坏 别这么想
            </h3>
          </a>
        </div>
        

        
        <div class="next-post">
          <a href="https://kaopubear.top/post/2020-02-29-r20years/">
            <h3 class="post-title">
              R 的 20 年
              <i class="ri-arrow-right-line"></i>
            </h3>
          </a>
        </div>
        
      </div>

      

    </div>
  </div>

  <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()
let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
</body>

</html>