<html>

<head>
  <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1583496325103">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/remixicon@2.3.0/fonts/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


</head>

<body>
  <div id="app" class="main px-4 flex flex-col lg:flex-row">
    <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="ri-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1583496325103" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E6%96%87%E7%8C%AE%E9%A2%98%E7%9B%AE-relative-abundance-of-transcripts-rats-identifying-differential-isoform-abundance-from-rna-seq">文献题目 Relative Abundance of Transcripts (RATs): Identifying differential isoform abundance from RNA-seq</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9">相关内容</a></li>
</ul>
</li>
<li><a href="#%E6%96%87%E7%8C%AE%E9%A2%98%E7%9B%AE-error-noise-and-bias-in-de-novo-transcriptome-assemblies">文献题目 Error, noise and bias in de novo transcriptome assemblies</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9-2">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89-2">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9-2">相关内容</a></li>
</ul>
</li>
<li><a href="#%E6%96%87%E7%8C%AE%E9%A2%98%E7%9B%AE-predicting-the-effects-of-snps-on-transcription-factor-binding-affinity">文献题目 Predicting the effects of SNPs on transcription factor binding affinity</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9-3">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89-3">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9-3">相关内容</a></li>
</ul>
</li>
<li><a href="#%E6%96%87%E7%8C%AE%E9%A2%98%E7%9B%AE-lnisks-reference-free-mutation-identification-for-large-and-complex-crop-genomes">文献题目 LNISKS: Reference-free mutation identification for large and complex crop genomes</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9-4">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89-4">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9-4">相关内容</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="ri-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">加入<a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a><br ><a href="https://tinyletter.com/kaopubear" target="_blank">邮件订阅博客内容</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
    <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
      <h1 class="text-3xl font-bold lg:mt-16">每周文献-190330-DTU分析及转录组denovo拼接解析等</h1>
      <div class="text-sm text-gray-700 lg:my-8">
        2019-03-30 / 6 min read
      </div>
      
      <img class="post-feature-image rounded-lg mx-auto my-4" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606090751.png" alt="">
      
      <div class="post-content yue">
        <h2 id="文献题目-relative-abundance-of-transcripts-rats-identifying-differential-isoform-abundance-from-rna-seq">文献题目 Relative Abundance of Transcripts (RATs): Identifying differential isoform abundance from RNA-seq</h2>
<!--more-->
<p><strong>DOI(url)</strong>: https://doi.org/10.12688/f1000research.17916.1</p>
<p><strong>发表日期</strong>：24 Feb 2019</p>
<h3 id="关键点">关键点</h3>
<p>差异表达中比较少见的一种方式</p>
<h3 id="参考意义">参考意义</h3>
<p>在 F1000Research 发表的这篇文章介绍了一个利用 alignment-free RNA-seq quantifications 结果进行差异分析的工具，主要是用来对转录本进行差异定量分析。在日常的分析中我们进行差异分析最常见的是对差异基因进行定量分析，这里没有考虑到每个基因内部转录本的情况。</p>
<p>在这篇文章中提到了一个三种差异分析的方法，分别是：</p>
<ul>
<li>differential gene expression (DGE)</li>
<li>differential transcript expression (DTE)</li>
<li>differential transcript usage (DTU)</li>
</ul>
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606090840.png" alt="" loading="lazy"></figure>
<p>根据需求的不同，这三种方法会分析出非常不同的结果。在 DTU 中，即便是两个表达总量没有差异的基因其也可能发生 isoform switching，及 dominant isoform 的改变。而文章中作者发表的 R 包即可进行 DTU 的分析，其输入数据可以是 Kallisto 或者 Salmon 的定量结果。</p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606090853.png" alt="" loading="lazy"></figure>
<p>R 包地址：https://github.com/bartongroup/Rats</p>
<h3 id="相关内容">相关内容</h3>
<p>考察一个基因内不同转录本在不同情况下的表达丰度，这个需求在我的实际分析中还没有用到。不过把转录本定量的结果转为基因定量的结果，使用 R 包 <a href="https://bioconductor.org/packages/release/bioc/html/tximport.html">tximport</a>。</p>
<h2 id="文献题目-error-noise-and-bias-in-de-novo-transcriptome-assemblies">文献题目 Error, noise and bias in de novo transcriptome assemblies</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/585745</p>
<p><strong>发表日期</strong>：March 22, 2019</p>
<h3 id="关键点-2">关键点</h3>
<p>详细讨论转录组 de novo 拼接的那点事。</p>
<h3 id="参考意义-2">参考意义</h3>
<p>这篇发表在 biorxiv 的文章，从多个方面阐述了转录本拼接本身存在的问题。这其中包括作者评估的几个算法都没有拼出数百个真是表达的基因，一大部分拼接处的 contigs 完全由内含子和 UTR 组成；对转录本有效长度的不准确给定量带来了很大的偏差等等。最后建议现在测序价格便宜了，能拼基因组就拼基因组吧。（这道理难道我不懂么，我只是没钱）。</p>
<p>这篇文章更值得参考的是他做的分析和采用的方法，50 页的文本非常详细的记录了具体的分析过程以及参数。也提醒了我们在做类似的分析是有哪些角度可以思考。</p>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606090941.png" alt="" loading="lazy"></figure>
<p><strong>主要分析流程</strong></p>
<ul>
<li>Short read processing：trimgalore</li>
<li>De novo transcriptome assembly
<ul>
<li>TRINITY</li>
<li>SHANNON</li>
<li>BINPACKER</li>
<li>BUSCO</li>
</ul>
</li>
<li>Assembly and read functional composition</li>
<li>Coverage of reference transcripts and genes
<ul>
<li>RSEM</li>
<li>TRANSDECODER</li>
</ul>
</li>
<li>Expression estimation</li>
<li>Analysis of missing genes
<ul>
<li>PANTHER</li>
</ul>
</li>
<li>Evaluating assembly redundancy</li>
<li>SNP-based analysis of assembly composition</li>
</ul>
<h3 id="相关内容-2">相关内容</h3>
<p>敢把流程写的及其详细的文章自然就敢把代码全部 show 给你，GitHub 地址 <a href="https://github.com/harvardinformatics/TranscriptomeAssemblyEvaluation">TranscriptomeAssemblyEvaluation</a> 。</p>
<h2 id="文献题目-predicting-the-effects-of-snps-on-transcription-factor-binding-affinity">文献题目 Predicting the effects of SNPs on transcription factor binding affinity</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/581306</p>
<p><strong>发表日期</strong>：March 18, 2019</p>
<h3 id="关键点-3">关键点</h3>
<p>非编码区 SNP 可以研究的方向之一，对转录因子结合能力的影响。</p>
<h3 id="参考意义-3">参考意义</h3>
<p>GWAS 显示 88％的疾病相关 SNP 位于非编码区。然而，非编码 SNP 仍未得到充分研究，这其中一部分原因是它们难以确定实验验证的优先级。这篇文章作者提出一种确定非编码区 SNP 重要性的评判标准以及方法。通过观察全基因组功能性转录因子结合位点内 SNP 的 ChIP-seq 信号强度差异来估计转录因子结合亲和力<br>
的变化。</p>
<p>目前用来分析转录因子结合位点的 martix 叫做 position weight matrix (PWM)，而文章中则提出了 SNP effect matrices (SEM)。</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606090955.png" alt="" loading="lazy"></figure>
<p><strong>SNP Effect Matrix pipeline</strong></p>
<p>使用数据</p>
<ul>
<li>ChIP-seq data： provides a transcription factor's endogenous binding in the genome</li>
<li>DNase I hypersensitive sites sequencing (DNaseseq) data:  represents regions of open chromatin where transcription factors are known to function</li>
<li>position weight matrices (PWMs): denote previous knowledge of the binding pattern of transcription factors</li>
</ul>
<p>流程</p>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606091008.png" alt="" loading="lazy"></figure>
<h3 id="相关内容-3">相关内容</h3>
<p>GitHub https://github.com/Boyle-Lab/SEM_CPP</p>
<h2 id="文献题目-lnisks-reference-free-mutation-identification-for-large-and-complex-crop-genomes">文献题目 LNISKS: Reference-free mutation identification for large and complex crop genomes</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/580829</p>
<p><strong>发表日期</strong>：March 19, 2019</p>
<h3 id="关键点-4">关键点</h3>
<p>针对 Bulk segregant analysis (BSA) 设计的无参考基因组 call snp 流程</p>
<h3 id="参考意义-4">参考意义</h3>
<p>对于植物（作物）来说，通常手里拿到的材料并不是真正的测序品种。比如水稻就包括籼稻和粳稻，而且很可能手里的粳稻也不是真正的粳稻。面对这种情况，其实不少用到和参考基因组比对的工作都或多或少的存在一些问题，因为那个基因组在很多位置其实你都无法「参考」，而其中影响最大的一类工作就是 call snp 相关的工作。当你的材料和参考基因组差别比较大的时候就会存在很多黑箱。如果能够不依赖基因组信息而直接 call snp 一定程度上就可以规避这类问题。但是准确度如何需要进一步考证。</p>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190606091020.png" alt="" loading="lazy"></figure>
<h3 id="相关内容-4">相关内容</h3>
<p>相关的几篇文章：</p>
<ul>
<li><a href="https://doi.org/10.1038/nbt.2515">Mutation identification by direct comparison of whole-genome sequencing data from mutant and wild-type individuals using k-mers</a></li>
<li><a href="https://doi.org/10.1038/nbt.3877">Rapid cloning of genes in hexaploid wheat using cultivar-specific long-range chromosome assembly</a></li>
<li><a href="https://doi.org/10.1038/s41587-018-0007-9">Resistance gene cloning from a wild crop relative by sequence capture and association genetics</a></li>
</ul>
<hr>
<figure data-type="image" tabindex="7"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名 - 非商业性使用 - 禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

      </div>
      
      <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 "
        href="https://kaopubear.top/tag/EqrCpMmApu/">
        <span class="flex-auto">文献</span>
      </a>
      


      <div class="flex justify-between py-8">
        
        <div class="prev-post">
          <a href="https://kaopubear.top/post/2019-04-04-facet-ggplot2/">
            <h3 class="post-title">
              <i class="ri-arrow-left-line"></i>
              用漫威数据学习 ggplot2 facet
            </h3>
          </a>
        </div>
        

        
        <div class="next-post">
          <a href="https://kaopubear.top/post/2019-03-22-weeklypaper/">
            <h3 class="post-title">
              每周文献-190322-DNA甲基化分析工具及silencers鉴定
              <i class="ri-arrow-right-line"></i>
            </h3>
          </a>
        </div>
        
      </div>

      

    </div>
  </div>

  <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()
let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
</body>

</html>