<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>每周文献-181216-lncRNA差异分析流程测评 | 思考问题的熊</title>
<meta name="description" content="很高兴和你相遇&lt;br&gt;这里正在记录我的所思所学">

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1576308317609">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">



<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>

<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />



  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://kaopubear.top/">
        <img src="https://kaopubear.top//images/avatar.png?v=1576308317609" class="site-logo">
        <h1 class="site-title">思考问题的熊</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="/" class="site-nav">
            首页
          </a>
        
      
        
          <a href="/archives" class="site-nav">
            归档
          </a>
        
      
        
          <a href="/tags" class="site-nav">
            标签
          </a>
        
      
        
          <a href="/post/about" class="site-nav">
            关于
          </a>
        
      
        
          <a href="/post/resource" class="site-nav">
            资料
          </a>
        
      
        
          <a href="/post/project" class="site-nav">
            项目
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
      
        
      
        
      
        
      
        
      
    </div>
    <div class="site-description">
      很高兴和你相遇<br>这里正在记录我的所思所学
    </div>
    <div class="site-footer">
      欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a> | <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">每周文献-181216-lncRNA差异分析流程测评</h2>
            <div class="post-date">2018-12-16</div>
            
              <div class="feature-container" style="background-image: url('https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-edb8eb425df5ea97.png')">
              </div>
            
            <div class="post-content">
              <h2 id="文献信息">文献信息</h2>
<p>**标题：**Differential gene <strong>expression</strong> analysis tools exhibit substandard performance for long non-coding RNA-sequencing data</p>
<p><strong>DOI(url):</strong> https://doi.org/10.1186/s13059-018-1466-5</p>
<p>**发表日期：**24 July 2018</p>
<p><strong>关键词：</strong> lncRNA,  Differential gene expression, RNA-seq, differential expression</p>
<h2 id="文献概述">文献概述</h2>
<p>这篇文章详细的分析了不同标准化和差异分析方法在 lncRNA 分析中的差别。一共使用了25个分析流程，主要关注点是 lncRNA  和一些低表达 mRNA。使用15种指标来评估差异基因的分析方法和标准化方法，一共使用了 6 中不同 RNA-seq 数据集，同时还提供了一个 shiny 网页可视化工具用来展示这些分析结果。按道理类似类型的文章应该达不到这个水平的杂志，lncRNA 分析方法的测评能够发到 Genome Biology 上也是牛，想必定有过人之处。</p>
<p>简单说最后的结论是使用 limma 和 SAMSeq 分析lncRNA 或者表达量很低的mRNA 效果要稍微好些，值得注意的是，为了获得至少 50％ 的 sensitivity，在实际环境（如临床癌症研究）中研究表达水平时需要超过80个样本（what ?）。测试使用的大约一半的方法显示出过多的假阳性，非常不可靠。</p>
<!--more-->
<h2 id="笔记">笔记</h2>
<p>lncRNA 研究的主要问题就是表达量太低，在一些软件中往往是要求去除掉表达量很低的基因，这个时候就非常尴尬。</p>
<p>作者为此挑选了一些引用率较高的软件，这些软件的共同点是都有 R 包可以使用，而且都是用原始的 read counts 作为输出。在数据方面，作者使用了不同规模的 6 个数据，基本上可以概括进行差异分析的不同情况。用作者的话说：据我们所知，我们的研究是迄今为止所进行的最大的实例	评估，包括所使用的真实数据集的数量，评估的指标数量以及DE流程数量。</p>
<h3 id="差异分析工具">差异分析工具</h3>
<p>使用的差异分析工具如下：</p>
<table>
<thead>
<tr>
<th>Tool (package version)</th>
<th style="text-align:left">Pipelines</th>
<th>Citationsa</th>
</tr>
</thead>
<tbody>
<tr>
<td>edgeR (3.14.0)</td>
<td style="text-align:left">(1) Exact test based on NB distribution, (2) GLM with NB family, (3) QL, (4–7) robust GLM with four different prior DF</td>
<td>5406</td>
</tr>
<tr>
<td>DESeq (1.24.0)</td>
<td style="text-align:left">(1) Default, exact test based on NB distribution</td>
<td>4655</td>
</tr>
<tr>
<td>DESeq2(1.12.4)</td>
<td style="text-align:left">Fits GLM with NB family. (1) Default, (2) independent filtering disabled (setting1), (3) independent filtering disabled and outlier-detection off (setting2)</td>
<td>1364</td>
</tr>
<tr>
<td>limma (3.25.21)</td>
<td style="text-align:left">Fits linear models on log-transformed counts. (1) Voom, (2) voom (robust), (3) trended, (4) trended (robust), (5) voom+QW, (6) limmaVST, (7) limmaQN</td>
<td>1828</td>
</tr>
<tr>
<td>NOISeq (2.12.1)</td>
<td style="text-align:left">(1) Default, data-adaptive and non-parametric method</td>
<td>524</td>
</tr>
<tr>
<td>baySeq (2.6.0)</td>
<td style="text-align:left">(1) Default, Bayesian methods with empirical prior distributions</td>
<td>315</td>
</tr>
<tr>
<td>SAMSeq (samr, 2.0)</td>
<td style="text-align:left">(1) Default, non-parametric method based on Wilcoxon rank sum statistic</td>
<td>140</td>
</tr>
<tr>
<td>PoissonSeq (1.1.2)</td>
<td style="text-align:left">(1) Default, uses poisson log-linear model</td>
<td>92</td>
</tr>
<tr>
<td>QuasiSeq (1.0.8)</td>
<td style="text-align:left">Fits GLM with NB family. (1) QL, (2) QLShrink, (3) QLSPline</td>
<td>57</td>
</tr>
</tbody>
</table>
<h3 id="分析过程">分析过程</h3>
<ol>
<li>first we evaluated various normalization procedures</li>
<li>second we compared the level of agreement among DE pipelines using various publicly available RNA-seq datasets;</li>
<li>third we explored the ability of the DE pipelines to recover known evidence of differential expression;</li>
<li>fourth we used simulation procedures to evaluate and compare the performance of the tools under a variety of gene expression experiment scenarios, such as variability, sample size, and fraction of DE genes.</li>
</ol>
<figure data-type="image" tabindex="1"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig1_HTML.png" alt=""></figure>
<h3 id="标准化方法比较">标准化方法比较</h3>
<p>结论是除了 quantile normalization (QN)，其它几种标准化的方法都差不多。 另外，终于在文献里看到了upsetR 画出来的图。</p>
<figure data-type="image" tabindex="2"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig2_HTML.png" alt=""></figure>
<h3 id="一致性分析">一致性分析</h3>
<p>主要检查指标：</p>
<ol>
<li>number of genes identified as significantly differentially expressed (SDE);</li>
<li>similarity in terms of the set of SDE genes;</li>
<li>the degree of agreement on gene ranking;</li>
<li>similarity of fold-change estimates;</li>
<li>handling of genes with special characteristics (lncRNAs, genes with low counts, genes with outliers)</li>
<li>computation time</li>
</ol>
<p>这里的分析结果是使用聚类方法展示的，如果看懂了就很能说明问题。</p>
<p>下图中的abcd 分别代表： <strong>a</strong> fraction of significantly differentially expressed (SDE) genes detected at 5% FDR, <strong>b</strong> overlap among pipelines in detecting SDE genes at 5% FDR, <strong>c</strong> gene ranking agreement, and <strong>d</strong> similarity of log fold-change (LFC) estimates)</p>
<figure data-type="image" tabindex="3"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig3_HTML.png" alt=""></figure>
<p>DESeq, baySeq, limmaQN,和 NOISeq 聚在一起，总之就是比较差。 QuasiSeq (both settings), edgeR robust (with both tested prior degrees of freedom), limmaVoom+QW, PoissonSeq 和 SAMSeq 相对而言就都比价不错。</p>
<p>文章后面有介绍了一些其它数据的比价结果就不一一详细罗列，最后的建议是：**limma (with variance stabilizing transformation; voom with or without quality weighting; trend) and SAMSeq control the actual FDR reasonably well, while not sacrificing sensitivity。**另外，就是如果做lncRNA，多送一些样本吧。</p>
<p>文章上传了分析代码，所有几个主要附件都是网页格式，展示所有评测结果还做了个<a href="http://statapps.ugent.be/tools/AppDGE/">网站</a>。</p>
<p>最后一点让我好奇的是，文章从投稿到接受一共花了8个月的时间，这期间都发生了什么，review 又给了哪些意见，什么数据是在 review 下补的 …</p>
<hr>
<figure data-type="image" tabindex="4"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

            </div>
            
              <div class="tag-container">
                
                  <a href="https://kaopubear.top//tag/EqrCpMmApu" class="tag">
                    文献
                  </a>
                
              </div>
            
            
              <div class="next-post">
                <div class="next">下一篇</div>
                <a href="https://kaopubear.top//post/2018-12-13-rggplot">
                  <h3 class="post-title">
                    R 语言可视化之原理概述篇
                  </h3>
                </a>
              </div>
            

            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>

<script type="application/javascript">

AOS.init();

hljs.initHighlightingOnLoad()

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>




  </body>
</html>
