<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1580465005059">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.remixicon.com/releases/v1.3.1/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<link rel="alternate" type="application/atom+xml" title="思考问题的熊 - Atom Feed" href="https://kaopubear.top//atom.xml">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div id="app" class="main px-4 flex flex-col lg:flex-row">
      <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="remixicon-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1580465005059" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E6%96%87%E7%8C%AE%E4%BF%A1%E6%81%AF">文献信息</a></li>
<li><a href="#%E6%96%87%E7%8C%AE%E6%A6%82%E8%BF%B0">文献概述</a></li>
<li><a href="#%E7%AC%94%E8%AE%B0">笔记</a>
<ul>
<li><a href="#%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90%E5%B7%A5%E5%85%B7">差异分析工具</a></li>
<li><a href="#%E5%88%86%E6%9E%90%E8%BF%87%E7%A8%8B">分析过程</a></li>
<li><a href="#%E6%A0%87%E5%87%86%E5%8C%96%E6%96%B9%E6%B3%95%E6%AF%94%E8%BE%83">标准化方法比较</a></li>
<li><a href="#%E4%B8%80%E8%87%B4%E6%80%A7%E5%88%86%E6%9E%90">一致性分析</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="remixicon-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
      <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
        <h1 class="text-3xl font-bold lg:mt-16">每周文献-181216-lncRNA差异分析流程测评</h1>
        <div class="text-sm text-gray-700 lg:my-8">
          2018-12-16 / 6 min read
        </div>
        
          <img class="post-feature-image rounded-lg mx-auto my-4" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-edb8eb425df5ea97.png" alt="">
        
        <div class="post-content yue">
          <h2 id="文献信息">文献信息</h2>
<p>**标题：**Differential gene <strong>expression</strong> analysis tools exhibit substandard performance for long non-coding RNA-sequencing data</p>
<!--more-->
<p><strong>DOI(url):</strong> https://doi.org/10.1186/s13059-018-1466-5</p>
<p>**发表日期：**24 July 2018</p>
<p><strong>关键词：</strong> lncRNA,  Differential gene expression, RNA-seq, differential expression</p>
<h2 id="文献概述">文献概述</h2>
<p>这篇文章详细的分析了不同标准化和差异分析方法在 lncRNA 分析中的差别。一共使用了25个分析流程，主要关注点是 lncRNA  和一些低表达 mRNA。使用15种指标来评估差异基因的分析方法和标准化方法，一共使用了 6 中不同 RNA-seq 数据集，同时还提供了一个 shiny 网页可视化工具用来展示这些分析结果。按道理类似类型的文章应该达不到这个水平的杂志，lncRNA 分析方法的测评能够发到 Genome Biology 上也是牛，想必定有过人之处。</p>
<p>简单说最后的结论是使用 limma 和 SAMSeq 分析lncRNA 或者表达量很低的mRNA 效果要稍微好些，值得注意的是，为了获得至少 50％ 的 sensitivity，在实际环境（如临床癌症研究）中研究表达水平时需要超过80个样本（what ?）。测试使用的大约一半的方法显示出过多的假阳性，非常不可靠。</p>
<h2 id="笔记">笔记</h2>
<p>lncRNA 研究的主要问题就是表达量太低，在一些软件中往往是要求去除掉表达量很低的基因，这个时候就非常尴尬。</p>
<p>作者为此挑选了一些引用率较高的软件，这些软件的共同点是都有 R 包可以使用，而且都是用原始的 read counts 作为输出。在数据方面，作者使用了不同规模的 6 个数据，基本上可以概括进行差异分析的不同情况。用作者的话说：据我们所知，我们的研究是迄今为止所进行的最大的实例	评估，包括所使用的真实数据集的数量，评估的指标数量以及DE流程数量。</p>
<h3 id="差异分析工具">差异分析工具</h3>
<p>使用的差异分析工具如下：</p>
<table>
<thead>
<tr>
<th>Tool (package version)</th>
<th style="text-align:left">Pipelines</th>
<th>Citationsa</th>
</tr>
</thead>
<tbody>
<tr>
<td>edgeR (3.14.0)</td>
<td style="text-align:left">(1) Exact test based on NB distribution, (2) GLM with NB family, (3) QL, (4–7) robust GLM with four different prior DF</td>
<td>5406</td>
</tr>
<tr>
<td>DESeq (1.24.0)</td>
<td style="text-align:left">(1) Default, exact test based on NB distribution</td>
<td>4655</td>
</tr>
<tr>
<td>DESeq2(1.12.4)</td>
<td style="text-align:left">Fits GLM with NB family. (1) Default, (2) independent filtering disabled (setting1), (3) independent filtering disabled and outlier-detection off (setting2)</td>
<td>1364</td>
</tr>
<tr>
<td>limma (3.25.21)</td>
<td style="text-align:left">Fits linear models on log-transformed counts. (1) Voom, (2) voom (robust), (3) trended, (4) trended (robust), (5) voom+QW, (6) limmaVST, (7) limmaQN</td>
<td>1828</td>
</tr>
<tr>
<td>NOISeq (2.12.1)</td>
<td style="text-align:left">(1) Default, data-adaptive and non-parametric method</td>
<td>524</td>
</tr>
<tr>
<td>baySeq (2.6.0)</td>
<td style="text-align:left">(1) Default, Bayesian methods with empirical prior distributions</td>
<td>315</td>
</tr>
<tr>
<td>SAMSeq (samr, 2.0)</td>
<td style="text-align:left">(1) Default, non-parametric method based on Wilcoxon rank sum statistic</td>
<td>140</td>
</tr>
<tr>
<td>PoissonSeq (1.1.2)</td>
<td style="text-align:left">(1) Default, uses poisson log-linear model</td>
<td>92</td>
</tr>
<tr>
<td>QuasiSeq (1.0.8)</td>
<td style="text-align:left">Fits GLM with NB family. (1) QL, (2) QLShrink, (3) QLSPline</td>
<td>57</td>
</tr>
</tbody>
</table>
<h3 id="分析过程">分析过程</h3>
<ol>
<li>first we evaluated various normalization procedures</li>
<li>second we compared the level of agreement among DE pipelines using various publicly available RNA-seq datasets;</li>
<li>third we explored the ability of the DE pipelines to recover known evidence of differential expression;</li>
<li>fourth we used simulation procedures to evaluate and compare the performance of the tools under a variety of gene expression experiment scenarios, such as variability, sample size, and fraction of DE genes.</li>
</ol>
<figure data-type="image" tabindex="1"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig1_HTML.png" alt="" loading="lazy"></figure>
<h3 id="标准化方法比较">标准化方法比较</h3>
<p>结论是除了 quantile normalization (QN)，其它几种标准化的方法都差不多。 另外，终于在文献里看到了upsetR 画出来的图。</p>
<figure data-type="image" tabindex="2"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig2_HTML.png" alt="" loading="lazy"></figure>
<h3 id="一致性分析">一致性分析</h3>
<p>主要检查指标：</p>
<ol>
<li>number of genes identified as significantly differentially expressed (SDE);</li>
<li>similarity in terms of the set of SDE genes;</li>
<li>the degree of agreement on gene ranking;</li>
<li>similarity of fold-change estimates;</li>
<li>handling of genes with special characteristics (lncRNAs, genes with low counts, genes with outliers)</li>
<li>computation time</li>
</ol>
<p>这里的分析结果是使用聚类方法展示的，如果看懂了就很能说明问题。</p>
<p>下图中的abcd 分别代表： <strong>a</strong> fraction of significantly differentially expressed (SDE) genes detected at 5% FDR, <strong>b</strong> overlap among pipelines in detecting SDE genes at 5% FDR, <strong>c</strong> gene ranking agreement, and <strong>d</strong> similarity of log fold-change (LFC) estimates)</p>
<figure data-type="image" tabindex="3"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-018-1466-5/MediaObjects/13059_2018_1466_Fig3_HTML.png" alt="" loading="lazy"></figure>
<p>DESeq, baySeq, limmaQN,和 NOISeq 聚在一起，总之就是比较差。 QuasiSeq (both settings), edgeR robust (with both tested prior degrees of freedom), limmaVoom+QW, PoissonSeq 和 SAMSeq 相对而言就都比价不错。</p>
<p>文章后面有介绍了一些其它数据的比价结果就不一一详细罗列，最后的建议是：**limma (with variance stabilizing transformation; voom with or without quality weighting; trend) and SAMSeq control the actual FDR reasonably well, while not sacrificing sensitivity。**另外，就是如果做lncRNA，多送一些样本吧。</p>
<p>文章上传了分析代码，所有几个主要附件都是网页格式，展示所有评测结果还做了个<a href="http://statapps.ugent.be/tools/AppDGE/">网站</a>。</p>
<p>最后一点让我好奇的是，文章从投稿到接受一共花了8个月的时间，这期间都发生了什么，review 又给了哪些意见，什么数据是在 review 下补的 …</p>
<hr>
<figure data-type="image" tabindex="4"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

        </div>

        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/EqrCpMmApu">
            <span class="flex-auto">文献</span>
          </a>
        


        <div class="flex justify-between py-8">
          
            <div class="prev-post">
              <a href="https://kaopubear.top/post/2018-12-19-biolitmap">
                <h3 class="post-title">
                  <i class="remixicon-arrow-left-line"></i>
                  生物信息发文章哪家强
                </h3>
              </a>
            </div>
          

          
            <div class="next-post">
              <a href="https://kaopubear.top/post/2018-12-14-usetrello">
                <h3 class="post-title">
                  任务协作管理工具 Trello 使用简介
                  <i class="remixicon-arrow-right-line"></i>
                </h3>
              </a>
            </div>
          
        </div>

        
          
            <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
<script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>

<div id="gitalk-container"></div>

<script>

  var gitalk = new Gitalk({
    clientID: '77b0e0937753a24c7510',
    clientSecret: 'd312ec269767c6310a14f017359734b75a4dc4fc',
    repo: 'fei0810.github.io',
    owner: 'fei0810',
    admin: ['fei0810'],
    id: (location.pathname).substring(0, 49),      // Ensure uniqueness and length less than 50
    distractionFreeMode: false  // Facebook-like distraction free mode
  })

  gitalk.render('gitalk-container')

</script>

          

          
        

      </div>
    </div>

    <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()

let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
  </body>
</html>
