<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1577962727761">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.remixicon.com/releases/v1.3.1/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<link rel="alternate" type="application/atom+xml" title="思考问题的熊 - Atom Feed" href="https://kaopubear.top//atom.xml">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div id="app" class="main px-4 flex flex-col lg:flex-row">
      <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="remixicon-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1577962727761" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#cis-regulatory-motifs">cis-regulatory motifs</a>
<ul>
<li><a href="#%E8%BD%AC%E5%BD%95%E8%B0%83%E6%8E%A7">转录调控</a></li>
<li><a href="#%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%A0%94%E7%A9%B6%E8%B0%83%E6%8E%A7%E5%8E%9F%E4%BB%B6%E7%9A%84%E4%B8%BB%E8%A6%81%E6%96%B9%E6%B3%95">全基因组研究调控原件的主要方法</a></li>
</ul>
</li>
<li><a href="#%E7%BB%93%E5%90%88%E5%9C%A8%E5%93%AA%E9%87%8C">结合在哪里</a></li>
<li><a href="#%E5%B8%B8%E7%94%A8%E7%9A%84%E8%A1%A8%E7%A4%BA%E6%96%B9%E6%B3%95">常用的表示方法</a></li>
<li><a href="#how-is-specificity-of-binding-achieved">How is specificity of binding achieved</a>
<ul>
<li><a href="#motif-%E5%AE%9A%E4%B9%89">motif 定义</a></li>
<li><a href="#%E6%9C%89%E6%97%B6%E5%B9%B6%E9%9D%9E%E7%9B%B4%E6%8E%A5%E5%92%8C-dna-%E7%BB%93%E5%90%88">有时并非直接和 DNA 结合</a></li>
</ul>
</li>
<li><a href="#how-to-identify-tf-binding-sites">How to identify TF binding sites?</a>
<ul>
<li><a href="#%E6%B2%A1%E6%9C%89%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9%E7%9A%84%E6%97%B6%E5%80%99">没有序列比对的时候</a></li>
<li><a href="#%E6%9C%80%E5%8E%9F%E5%A7%8B%E7%9A%84%E6%96%B9%E6%B3%95%E6%98%AF%E5%A4%9A%E9%87%8D%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9-msa">最原始的方法是多重序列比对 MSA</a></li>
<li><a href="#phylocon-comparative-genomic-approach">PhyloCon — comparative genomic approach</a></li>
<li><a href="#expectation-maximization-em-%E7%9B%AE%E5%89%8D%E6%9C%80%E5%B8%B8%E7%94%A8%E7%9A%84%E6%96%B9%E6%B3%95meme">Expectation-Maximization (EM) 目前最常用的方法（MEME）</a></li>
</ul>
</li>
<li><a href="#%E5%9C%A8-meme-%E4%B8%AD">在 MEME 中</a>
<ul>
<li><a href="#%E7%9B%B8%E5%85%B3%E6%96%87%E7%8C%AE">相关文献</a></li>
<li><a href="#%E8%BF%87%E7%A8%8B">过程</a></li>
</ul>
</li>
<li><a href="#what-does-transcription-factor-binding-mean-in-terms-of-regulating-target-genes">What does transcription factor binding mean in terms of regulating target genes?</a></li>
<li><a href="#chip-seq-call-peak-%E7%9A%84%E7%AD%96%E7%95%A5%E5%92%8C%E6%80%9D%E6%83%B3">ChIP-seq call peak 的策略和思想</a>
<ul>
<li><a href="#chip-seq-%E6%8A%80%E6%9C%AF">ChIP-seq 技术</a></li>
<li><a href="#macs-%E5%8E%9F%E7%90%86">macs 原理</a></li>
<li><a href="#chip-seq-%E5%90%8E%E7%BB%AD%E5%88%86%E6%9E%90">ChIP-seq 后续分析</a></li>
</ul>
</li>
<li><a href="#%E5%9F%BA%E5%9B%A0%E8%B0%83%E6%8E%A7%E7%BD%91%E7%BB%9C">基因调控网络</a>
<ul>
<li><a href="#%E8%B4%9D%E5%8F%B6%E6%96%AF%E7%BD%91%E7%BB%9C">贝叶斯网络</a></li>
<li><a href="#model-%E5%9B%BE%E5%BD%A2%E8%AF%B4%E6%98%8E">model 图形说明</a></li>
<li><a href="#%E6%A8%A1%E5%9E%8B%E9%80%89%E6%8B%A9">模型选择</a></li>
<li><a href="#%E8%AE%AD%E7%BB%83%E5%8F%82%E6%95%B0">训练参数</a></li>
<li><a href="#cpd-for-discrete-expression-level">CPD for discrete expression level</a></li>
<li><a href="#%E5%AE%9E%E9%99%85%E5%90%AB%E4%B9%89">实际含义</a></li>
<li><a href="#module-network-%E6%A8%A1%E5%9D%97%E7%BD%91%E7%BB%9C">module network 模块网络</a></li>
</ul>
</li>
<li><a href="#%E6%97%A0%E6%96%B9%E5%90%91%E5%9B%BE">无方向图</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="remixicon-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
      <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
        <h1 class="text-3xl font-bold lg:mt-16">转录调控与生物信息</h1>
        <div class="text-sm text-gray-700 lg:my-8">
          2017-08-18 / 11 min read
        </div>
        
          <img class="post-feature-image rounded-lg mx-auto my-4" src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a26326baa2031fb8.jpg" alt="">
        
        <div class="post-content yue">
          <h2 id="cis-regulatory-motifs">cis-regulatory motifs</h2>
<h3 id="转录调控">转录调控</h3>
<!--more-->
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a26326baa2031fb8.jpg" alt="" loading="lazy"></figure>
<ul>
<li>Transcription start site (TSS)</li>
<li>Transcription factor binding sites (TFBS)</li>
<li>Cis-regulatory module (CRM) 有多个 TF 在一起</li>
<li>Proximal promoter and distal enhancer 近端的启动子远端的增强子</li>
</ul>
<p>在人中，有 300 个 TF 结合在核心启动子区域；有 1500 个结合在基因其他区域，可以调节一系列基因</p>
<p>图示</p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-85acad960c29db7c.jpg" alt="" loading="lazy"></figure>
<ul>
<li>其中的 insulator 可以阻隔 enhancer 起作用</li>
</ul>
<h3 id="全基因组研究调控原件的主要方法">全基因组研究调控原件的主要方法</h3>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-ccc8ba0f510f85ac.jpg" alt="" loading="lazy"></figure>
<p>ChIP-seq:</p>
<ul>
<li>peak 的高低体现了蛋白的富集程度</li>
<li>只能研究单一的蛋白，具有特异性</li>
<li>有时候不能找到合适的抗体</li>
</ul>
<p>DNase-seq</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-1990297397cfba93.jpg" alt="" loading="lazy"></figure>
<ul>
<li>
<p>DNase I 超敏感位点（DHS）是对 DNaseⅠ高度敏感的活性染色质区域，DNase 测序（DNase-seq）是进行全基因组 DHS 分析的常用方法</p>
</li>
<li>
<p>DNase I 是一种非特异性核酸内切酶，基于它们对切割的过敏性，长期以来就被用于对“开放”染色质位点的作图</p>
</li>
<li>
<p>chromatin open 的位置很容易有其他蛋白的结合</p>
</li>
<li>
<p>由于多种蛋白质可以与相同序列相结合，有必要整合 DNase-seq 测序数据和 ChIP-seq 测序数据来对引起特定 DNase 足迹的蛋白质进行定性鉴定</p>
</li>
<li>
<p>不依赖于抗体或表位标签，DNase-seq 可以用来在一次实验中分析大量蛋白质的基因组分布</p>
</li>
<li>
<p>从大范围来看，结合的位置凸起。如果从小范围来看，空着的位置刚好是一个可能的 motif</p>
</li>
</ul>
<p>ATAC-seq (Assay for transposase- accessiblechromatin using sequencing)</p>
<ul>
<li>通过 Tn5 转座酶，优先标记和测序核小体之间的 DNA</li>
<li>ATAC-seq 提供的信息与新 DNase-seq 法差不多，但步骤更为简单，需要的细胞也更少</li>
<li>在无法获得大量细胞的情况下，ATAC-seq 更有帮助。</li>
</ul>
<p>文章原图</p>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-35db6f988c423e23.jpg" alt="" loading="lazy"></figure>
<ul>
<li>
<p>A genomic locus analysed by complementary chromatinprofiling experiments reveals different aspects ofchromatin structure</p>
</li>
<li>
<p>ChIP–seq reveals binding sites of specifictranscription factors (TFs); DNase-seq, ATAC-seq andFAIRE–seq reveal regions of open chromatin; andMNase-seq identifies well-positioned nucleosomes.</p>
</li>
<li>
<p>In ChIP–seq, specific antibodies are used toextract DNA fragments that are bound to the targetprotein, either directly or through other proteins ina complex that contains the target factor.</p>
</li>
<li>
<p>In DNase-seq, chromatin is lightly digested by theDNase I endonuclease. Size selection is used toenrich for fragments that are produced in regions ofchromatin where the DNA is highly sensitive to DNaseI attack.</p>
</li>
<li>
<p>ATAC-seq is an alternative method to DNase-seq thatuses an engineered Tn5 transposase to cleave DNA andto integrate primer DNA sequences into the cleavedgenomic DNA (that is, tagmentation).</p>
</li>
<li>
<p>Micrococcal nuclease (MNase) is an endo–exonucleasethat processively digests DNA until an obstruction,such as a nucleosome, is reached.</p>
</li>
<li>
<p>In FAIRE–seq, formaldehyde is used to crosslink chromatin, and phenol–chloroform is used to isolate sheared DNA.</p>
</li>
</ul>
<h2 id="结合在哪里">结合在哪里</h2>
<p>Some TFs almost always bind in proximal promoter regions</p>
<p>Others bind to many regions</p>
<h2 id="常用的表示方法">常用的表示方法</h2>
<p>Position weight matrix (PWM)</p>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-ec68f83e1d83b9bd.jpg" alt="" loading="lazy"></figure>
<ul>
<li>
<p>把所有碱基出现的次数相加，高度表示可信度</p>
</li>
<li>
<p>这种方法过于简单，不能表示出碱基之间的关系。</p>
</li>
<li>
<p>假设各个碱基之间均为独立</p>
</li>
</ul>
<h2 id="how-is-specificity-of-binding-achieved">How is specificity of binding achieved</h2>
<h3 id="motif-定义">motif 定义</h3>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-b433ef63a0e65f2f.jpg" alt="" loading="lazy"></figure>
<h3 id="有时并非直接和-dna-结合">有时并非直接和 DNA 结合</h3>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-f7f7bd39e27d11a7.jpg" alt="" loading="lazy"></figure>
<h2 id="how-to-identify-tf-binding-sites">How to identify TF binding sites?</h2>
<figure data-type="image" tabindex="9"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-4e6a38e15b458a94.jpg" alt="" loading="lazy"></figure>
<h3 id="没有序列比对的时候">没有序列比对的时候</h3>
<p>Given a collection of genes that are likely to be regulated by the same TFs (or orthologous genes across different species — methods based on phylogenetic footprinting principles), find the TF-binding motifs in common</p>
<p>但是问题是不知道 motif 是什么，找不到相关的基因，而且如何排除背景干扰</p>
<h3 id="最原始的方法是多重序列比对-msa">最原始的方法是多重序列比对 MSA</h3>
<p>比较保守的非编码区域可能有</p>
<h3 id="phylocon-comparative-genomic-approach">PhyloCon — comparative genomic approach</h3>
<ul>
<li>结合序列比对和共表达基因</li>
<li>共表达基因很可能收到相同的 motif 调控</li>
<li>但并不是所有的 elements 都保守</li>
</ul>
<h3 id="expectation-maximization-em-目前最常用的方法meme">Expectation-Maximization (EM) 目前最常用的方法（MEME）</h3>
<p><strong>Expectation-Maximization</strong></p>
<p>In each iteration, it learns the PWM model and identifies examples of the matrix (sites in the input sequences) 在每一次迭代中，学习一个 PWMmodel 然后再通过输入的序列进行比对</p>
<p>MEME works by iteratively refining PWMs and identifying sites for each PWM（不同的迭代直到找到一个最合适的 PWM)</p>
<p>The intuitive idea is as follows:</p>
<ol>
<li>Estimate motif model (PWM)</li>
</ol>
<p>Start with a k-mer seed (random or specified) 通常是 6 个<br>
Build a PWM by incorporating some of background frequencies 根据背景生成一个初始的 PWM</p>
<ol start="2">
<li>Identify examples of the model</li>
</ol>
<p>For every k-mer in the input sequences, identify its probability given the PWM model 计算 k-mer 在输入序列中给出 PWM 出现的概率</p>
<ol start="3">
<li>Re-estimate the motif model</li>
</ol>
<p>Calculate a new PWM, based on the weighted frequencies of all k-mers in the input sequences<br>
根据 input 序列中 k-mer 出现频率的权重更新 PWM</p>
<ol start="4">
<li>Iteratively refine the PWMs and identify sites until convergence</li>
</ol>
<p>例子 1</p>
<figure data-type="image" tabindex="10"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-696e009de8179d38.jpg" alt="" loading="lazy"></figure>
<p>1.1</p>
<figure data-type="image" tabindex="11"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-420901f4091c12df.jpg" alt="" loading="lazy"></figure>
<p>1.2</p>
<figure data-type="image" tabindex="12"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-dabdaec5cf2ee9c5.jpg" alt="" loading="lazy"></figure>
<p>1.3</p>
<figure data-type="image" tabindex="13"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-9fe5da09f37947d5.jpg" alt="" loading="lazy"></figure>
<h2 id="在-meme-中">在 MEME 中</h2>
<h3 id="相关文献">相关文献</h3>
<figure data-type="image" tabindex="14"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-5a5adb20d0f23c74.jpg" alt="" loading="lazy"></figure>
<h3 id="过程">过程</h3>
<p>首先设置 model, 然后经历 Estep 和 Mstep, 找到合适的 PWM</p>
<p>然后将 PWM 进行极大似然转换并取 log</p>
<figure data-type="image" tabindex="15"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a0a8f0aa6c7a63b9.jpg" alt="" loading="lazy"></figure>
<p>然后看输入序列中出现该 motif 的概率</p>
<figure data-type="image" tabindex="16"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-005ed16a78029e4c.jpg" alt="" loading="lazy"></figure>
<h2 id="what-does-transcription-factor-binding-mean-in-terms-of-regulating-target-genes">What does transcription factor binding mean in terms of regulating target genes?</h2>
<p>人的大多数结合位点都是在内含子和基因间区</p>
<p>Stronger sites are not closer to differentially regulated genes (not necessarily more functional)</p>
<p>Majority of functional sites not conserved</p>
<p>目前很难预测靶基因</p>
<figure data-type="image" tabindex="17"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-d0946ea9c0400b65.jpg" alt="" loading="lazy"></figure>
<h2 id="chip-seq-call-peak-的策略和思想">ChIP-seq call peak 的策略和思想</h2>
<h3 id="chip-seq-技术">ChIP-seq 技术</h3>
<figure data-type="image" tabindex="18"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-8da6e354dfa5a997.jpg" alt="" loading="lazy"></figure>
<p>核心思想</p>
<figure data-type="image" tabindex="19"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-4132776294f9f9ee.jpg" alt="" loading="lazy"></figure>
<h3 id="macs-原理">macs 原理</h3>
<p>TF 在基因组上的结合其实是一个随机过程，基因组的每个位置其实都有机会结合某个 TF，只是概率不一样</p>
<p>peak 出现的位置，是 TF 结合的热点，而 peak-calling 就是为了找到这些热点。</p>
<p>热点：位置多次被测得的 read 所覆盖（我们测的是一个细胞群体，read 出现次数多，说明该位置被 TF 结合的几率大）。</p>
<p>read 出现多少次算多：假设 TF 在基因组上的分布没有任何规律，测序得到的 read 在基因组上的分布也必然是随机的，某个碱基上覆盖的 read 的数目应该服从二项分布。</p>
<p>当 n 很大，p 很小时，二项分布可以近似用泊松分布替代</p>
<figure data-type="image" tabindex="20"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-34152b6a1715bd66.jpg" alt="" loading="lazy"></figure>
<p><span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>λ</mi></mrow><annotation encoding="application/x-tex">\lambda</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em;"></span><span class="mord mathdefault">λ</span></span></span></span>是泊松分布唯一的参数，n 是测序得到的 read 总数目，l 是单个 read 的长度，s 是基因组的大小。</p>
<p>我们可以算出在某个置信概率（如 0.00001）下，随机情况下，某个碱基上可以覆盖的 read 的数目的最小值，当实际观察到的 read 数目超过这个值（单侧检验）时，我们认为该碱基是 TF 的一个结合热点。反过来，针对每一个 read 数目，我们也可以算出对应的置信概率 P。</p>
<p>实际情况由于测序、mapping 过程内在的偏好性，以及不同染色质间的差异性，相比全基因组，某些碱基可能内在地会被更多的 read 所覆盖，这种情况得到的很多 peak 可能都是假的。</p>
<p>MACS 考虑到了这一点，当对某个碱基进行假设检验时，MACS 只考虑该碱基附近的染色质区段（如 10k），此时，上述公式中 n 表示附近 10k 区间内的 read 数目，s 被置为 10k。当有对照组实验（Control，相比实验组，没有用抗体捕获 TF，或用了一个通用抗体）存在时，利用 Control 组的数据构建泊松分布，当没有 Control 时，利用实验组，稍大一点的局部区间（比如 50k）的数据构建泊松分布。</p>
<p>read 只是跟随着 TF 一起沉淀下来的 DNA fragment 的末端，read 的位置并不是真实的 TF 结合的位置。</p>
<p>在 peak-calling 之前，延伸 read 是必须的。不同 TF 大小不一样，对 read 延伸的长度也理应不同。</p>
<p>我们知道测得的 read 最终其实会近似地平均分配到正负链上，这样对于一个 TF 结合热点而言，read 在附近正负链上会近似地形成“双峰”。</p>
<p>MACS 会以某个 window size 扫描基因组，统计每个 window 里面 read 的富集程度，然后抽取（比如 1000 个）合适的（read 富集程度适中，过少，无法建立模型，过大，可能反映的只是某种偏好性）window 作样本，建立“双峰模型”。</p>
<p>最后，两个峰之间的距离就被认为是 TF 的长度 D，每个 read 将延伸 D/2 的长度</p>
<figure data-type="image" tabindex="21"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-e80d17eda83aba19.jpg" alt="" loading="lazy"></figure>
<h3 id="chip-seq-后续分析">ChIP-seq 后续分析</h3>
<p>If we are given a set of ChIP-seq peaks, how to identify motif for the TF— use MEME</p>
<p>To find out what the sequence motif resembles — use TomTom</p>
<p>Use known motif to search peak regions — use FIMO</p>
<p>Study common biological pathways or functions of potential target genes of the TF — use GREAT</p>
<p>刘晓乐实验室 ChIP-seq 数据分析流程</p>
<ul>
<li><a href="http://cistrome.org/Cistrome/Cistrome_Project.html">http://cistrome.org/Cistrome/Cistrome_Project.html</a></li>
<li>说明 <a href="http://cistrome.org/chilin/index.html">http://cistrome.org/chilin/index.html</a></li>
</ul>
<h2 id="基因调控网络">基因调控网络</h2>
<h3 id="贝叶斯网络">贝叶斯网络</h3>
<p>定义：包括一个有向无环图（DAG）和一个条件概率表集合。DAG 中每一个节点表示一个随机变量，可以是可直接观测变量或隐藏变量，而有向边表示随机变量间的条件依赖；条件概率表中的每一个元素对应 DAG 中唯一的节点，存储此节点对于其所有直接前驱节点的联合条件概率</p>
<p>性质：每一个节点在其直接前驱节点的值制定后，这个节点条件独立于其所有非直接前驱前辈节点</p>
<p>类似 Markov 过程，贝叶斯网络可以看做是 Markov 链的非线性扩展。这条特性的重要意义在于明确了贝叶斯网络可以方便计算联合概率分布。</p>
<p>通过基因表达来推测网络</p>
<figure data-type="image" tabindex="22"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-c117b771a7925d90.jpg" alt="" loading="lazy"></figure>
<h3 id="model-图形说明">model 图形说明</h3>
<figure data-type="image" tabindex="23"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-62d13869cbcef1a6.jpg" alt="" loading="lazy"></figure>
<ul>
<li>DAG：有向无环图</li>
<li>Conditional probabilitydistribution (CPD) 条件概率分布</li>
<li>多变量非独立联合条件概率分布 P(G1，G2，G3，G4，G5) 求取公式</li>
</ul>
<h3 id="模型选择">模型选择</h3>
<figure data-type="image" tabindex="24"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-8a4599ddd62e720a.jpg" alt="" loading="lazy"></figure>
<h3 id="训练参数">训练参数</h3>
<figure data-type="image" tabindex="25"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-0c992efe8d33fe70.jpg" alt="" loading="lazy"></figure>
<h3 id="cpd-for-discrete-expression-level">CPD for discrete expression level</h3>
<figure data-type="image" tabindex="26"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-0fd44bbf35bbe7c8.jpg" alt="" loading="lazy"></figure>
<h3 id="实际含义">实际含义</h3>
<figure data-type="image" tabindex="27"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-9143271b6b0b757b.jpg" alt="" loading="lazy"></figure>
<h3 id="module-network-模块网络">module network 模块网络</h3>
<blockquote>
<p>每个节点不是一个基因合适若干基因</p>
</blockquote>
<p>经典文章</p>
<figure data-type="image" tabindex="28"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-04aa3460dd99e8f5.jpg" alt="" loading="lazy"></figure>
<p>主要过程</p>
<figure data-type="image" tabindex="29"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-b17e6972d976f571.jpg" alt="" loading="lazy"></figure>
<ul>
<li>具有隐藏变量</li>
<li>首先将基因进行 cluster</li>
<li>然后 M 步学习，E 步调整</li>
</ul>
<h2 id="无方向图">无方向图</h2>
<p>分析过程要给已经构建的相关性矩阵取逆</p>
<p>当样本很小时无法进行转换要使用 lasso 算法</p>
<figure data-type="image" tabindex="30"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-6f4c832b141e7c85.jpg" alt="" loading="lazy"></figure>
<p>关键在于如何确定公式中的 lamada</p>
<figure data-type="image" tabindex="31"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-f99b81c714949da4.jpg" alt="" loading="lazy"></figure>
<p>这样不需要所有节点之间都有边</p>
<hr>
<figure data-type="image" tabindex="32"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="33"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

        </div>

        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/DvI6ypPftI">
            <span class="flex-auto">生物信息</span>
          </a>
        


        <div class="flex justify-between py-8">
          
            <div class="prev-post">
              <a href="https://kaopubear.top/post/2017-09-08-LearnDocker1">
                <h3 class="post-title">
                  <i class="remixicon-arrow-left-line"></i>
                  Docker 学习笔记 I
                </h3>
              </a>
            </div>
          

          
            <div class="next-post">
              <a href="https://kaopubear.top/post/2017-08-17-longxing-bioinfo-epigenetic">
                <h3 class="post-title">
                  表观遗传与生物信息
                  <i class="remixicon-arrow-right-line"></i>
                </h3>
              </a>
            </div>
          
        </div>

        
          
            <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
<script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>

<div id="gitalk-container"></div>

<script>

  var gitalk = new Gitalk({
    clientID: '77b0e0937753a24c7510',
    clientSecret: 'd312ec269767c6310a14f017359734b75a4dc4fc',
    repo: 'fei0810.github.io',
    owner: 'fei0810',
    admin: ['fei0810'],
    id: (location.pathname).substring(0, 49),      // Ensure uniqueness and length less than 50
    distractionFreeMode: false  // Facebook-like distraction free mode
  })

  gitalk.render('gitalk-container')

</script>

          

          
        

      </div>
    </div>

    <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()

let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
  </body>
</html>
