<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>机器学习在生物信息学中的应用 | 思考问题的熊</title>
<meta name="description" content="很高兴和你相遇&lt;br&gt;这里正在记录我的所思所学">

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1576308317609">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">



<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>

<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />



  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://kaopubear.top/">
        <img src="https://kaopubear.top//images/avatar.png?v=1576308317609" class="site-logo">
        <h1 class="site-title">思考问题的熊</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="/" class="site-nav">
            首页
          </a>
        
      
        
          <a href="/archives" class="site-nav">
            归档
          </a>
        
      
        
          <a href="/tags" class="site-nav">
            标签
          </a>
        
      
        
          <a href="/post/about" class="site-nav">
            关于
          </a>
        
      
        
          <a href="/post/resource" class="site-nav">
            资料
          </a>
        
      
        
          <a href="/post/project" class="site-nav">
            项目
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
      
        
      
        
      
        
      
        
      
    </div>
    <div class="site-description">
      很高兴和你相遇<br>这里正在记录我的所思所学
    </div>
    <div class="site-footer">
      欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a> | <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">机器学习在生物信息学中的应用</h2>
            <div class="post-date">2017-08-15</div>
            
              <div class="feature-container" style="background-image: url('https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-c9e3a48658e8d2f2.jpg')">
              </div>
            
            <div class="post-content">
              <h2 id="recap">recap</h2>
<h3 id="machine-learning-methods">Machine learning methods</h3>
<p>general-purpose approaches to learn functional relationships from data without the need to define them a priori</p>
<h3 id="advantage">Advantage</h3>
<p>derive predictive models without the need for strong assumptions about underlying mechanisms, which are frequently unknown or insufficiently defined (especially for genomic data)</p>
<!--more-->
<h3 id="example">Example</h3>
<p>he most accurate prediction of gene expression levels is currently made from a broad set of epigenetic features using linear models or random forests</p>
<p>However, how the selected features determine the transcript levels remains an active research topic</p>
<h3 id="four-steps">four steps</h3>
<p>Most of machine learning applications in genomics can be described within the canonical machine learning workflow with four steps</p>
<ul>
<li>
<p>data cleaning and pre-processing</p>
</li>
<li>
<p>feature extraction</p>
</li>
<li>
<p>model fitting/training</p>
</li>
<li>
<p>evaluation</p>
</li>
</ul>
<p>It is customary to denote one data sample, including all covariates and features as input x (usually a vector of numbers), and label it with its response variable or output value y (usually a single number) when available.</p>
<h2 id="supervised-machine-learning-model">Supervised machine learning model</h2>
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-febd56d1bba026df.jpg" alt="img"></figure>
<h2 id="unsupervised-machine-learning">Unsupervised machine learning</h2>
<p>approaches aim to discover patterns from the data samples x themselves, without the need for output labels y.</p>
<p>Methods such as clustering and principal component analysis are typical examples of unsupervised models applied to biological data</p>
<p>深度神经网络</p>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-6f033979cf47f596.jpg" alt="img"></figure>
<h2 id="neural-networks">Neural networks</h2>
<p>An artificial neural network, initially inspired by neural networks in the brain, consists of layers of interconnected compute units (neurons).</p>
<p>The depth of a neural network corresponds to the number of hidden layers, and the width to the maximum number of neurons in one of itslayers.</p>
<p>When training networks with larger numbers of hidden layers, artificial neural networks were rebranded as “deep networks”.</p>
<p>The term “neural network” largely refers to the hypothesis class part of a machine learning algorithm:</p>
<ul>
<li>Hypothesis: non-linear hypothesis function, which involve compositions of multiple linear operators (e.g. matrix multiplications) and element wise nonlinear functions</li>
<li>Loss: “Typical” loss functions for classification and regression: logistic, softmax (multiclass logistic), etc.</li>
<li>Optimization: Gradient descent</li>
</ul>
<p>architectures</p>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-354e6277facce7dc.jpg" alt="img"></figure>
<p>Recurrent neural networks</p>
<ul>
<li>Maintain hidden state over time, hidden state is a function of currentinput and previous hidden state</li>
<li>Traditional RNNs have trouble capturing long-term dependencies</li>
<li>Problem has to do with vanishing gradient, for many activations like sigmoid, gradients get smaller and smaller over subsequent layers</li>
<li>One solution, long short term memory (LSTM) (Hochreiter and Schmidhuber 1997), has more complex structure that specifically encodes memory and pass-through features, able to model long-term dependencies</li>
</ul>
<p>application</p>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-d5fbb1ee36ed7bee.jpg" alt="img"></figure>
<h2 id="deepbind">DeepBind</h2>
<p>CNN architectures to predict specificities of DNA-binding and RNAbinding proteins</p>
<p>Outperformed existing methods, to recover known and novel sequence motifs, and could quantify the effect of sequence alterations and identify functional SNVs</p>
<p>The neurons in the convolutional layer scan for motif sequences and combinations thereof, similar to conventional PWMs</p>
<p>The learning signal from deeper layers informs the convolutional layer which motifs are the most relevant. The motifs recovered by the model can then be visualized as sequence logos</p>
<p>paper</p>
<ul>
<li>Predicting the sequence specificities of DNA- and RNA-binding proteins by deep learning, doi:10.1038/nbt.3300</li>
</ul>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-c9e3a48658e8d2f2.jpg" alt="img"></figure>
<h2 id="important-considerations-related-to-data">Important considerations related to data</h2>
<p>Need sufficient amount of labelled data</p>
<p>Need to be trained, selected and tested on independent data sets to avoid overfitting and assure that the model will generalize</p>
<ul>
<li>
<p>The training set is used to learn models with different parameters, which are then assessed on the validation set.</p>
</li>
<li>
<p>The model with best performance, for example prediction accuracy or meansquared error, is selected and further evaluated on the test set to quantify the performance on unseen data and for comparison to other methods.</p>
</li>
<li>
<p>Typical data set proportions are 60% for training, 10% for validation and 30% for model testing. If the data set is small, k-fold cross-validation can be used</p>
</li>
</ul>
<p>Categorical features such as DNA nucleotides need to be encoded numerically. Typically represented as binary vectors with all but one entry set to zero (one-hot coding).</p>
<p>The four bits of each encoded base are commonly considered analogously to color channels of an image to preserve the entity of a nucleotide</p>
<hr>
<figure data-type="image" tabindex="6"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="7"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

            </div>
            
              <div class="tag-container">
                
                  <a href="https://kaopubear.top//tag/DvI6ypPftI" class="tag">
                    生物信息
                  </a>
                
                  <a href="https://kaopubear.top//tag/P4GLM7-KRkk" class="tag">
                    机器学习
                  </a>
                
              </div>
            
            
              <div class="next-post">
                <div class="next">下一篇</div>
                <a href="https://kaopubear.top//post/2017-08-14-longxing-bioinfo-blastmore">
                  <h3 class="post-title">
                    序列比对其他相关问题
                  </h3>
                </a>
              </div>
            

            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>

<script type="application/javascript">

AOS.init();

hljs.initHighlightingOnLoad()

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>




  </body>
</html>
