<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>3D 基因组与生物信息 | 思考问题的熊</title>
<meta name="description" content="很高兴和你相遇&lt;br&gt;这里正在记录我的所思所学">

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1576308415693">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">



<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>

<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />



  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://kaopubear.top/">
        <img src="https://kaopubear.top//images/avatar.png?v=1576308415693" class="site-logo">
        <h1 class="site-title">思考问题的熊</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="/" class="site-nav">
            首页
          </a>
        
      
        
          <a href="/archives" class="site-nav">
            归档
          </a>
        
      
        
          <a href="/tags" class="site-nav">
            标签
          </a>
        
      
        
          <a href="/post/about" class="site-nav">
            关于
          </a>
        
      
        
          <a href="/post/resource" class="site-nav">
            资料
          </a>
        
      
        
          <a href="/post/project" class="site-nav">
            项目
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
      
        
      
        
      
        
      
        
      
    </div>
    <div class="site-description">
      很高兴和你相遇<br>这里正在记录我的所思所学
    </div>
    <div class="site-footer">
      欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a> | <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">3D 基因组与生物信息</h2>
            <div class="post-date">2018-08-11</div>
            
              <div class="feature-container" style="background-image: url('https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-828599c54149be5d.jpg')">
              </div>
            
            <div class="post-content">
              <h2 id="重要的-hi-c-相关文献">重要的 Hi-C 相关文献</h2>
<p>第一篇 Hi-C 文章： Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome； DOI: 10.1126/science.1181369</p>
<p>TAD 提出： Topological Domains in Mammalian Genomes Identified by Analysis of Chromatin Interactions ；doi:  10.1038/nature11082</p>
<p>高分辨率 Hi-C： A 3D Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping <a href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a></p>
<p>单细胞 Single cell： Hi-C reveals cell-to-cell variability in chromosome structure doi:  10.1038/nature12593；3D structures of individual mammalian genomes studied by single-cell Hi-C doi:10.1038/nature21429</p>
<p>综述：老师所讲 Hi-C 相关基础知识主要来自于综述 Organization and function of the 3D genome，doi:10.1038/nrg.2016.112</p>
<!--more-->
<h2 id="chromatin-interaction-in-different-resolutions">Chromatin interaction in different resolutions</h2>
<p>不同分辨率 Hi-C 可以看到的内容不同</p>
<p>5KB 可以看到各种 loop</p>
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-828599c54149be5d.jpg" alt="img"></figure>
<p>10KB 可以看到 TAD</p>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-02e99cd9e50fa311.jpg" alt="img"></figure>
<p>50kb 可以看到 TAD 之间的关联</p>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-44370d2fbc86ef43.jpg" alt="img"></figure>
<p>在整个染色体的水平可以看到染色质的位置分布</p>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-c2baa72eda2f9bb2.jpg" alt="img"></figure>
<h2 id="什么造成了所谓的-tad">什么造成了所谓的 TAD</h2>
<p><a href="https://img.mubu.com/document_image/e556f4a1-7f13-4a38-8943-9dc29cfb9314.jpg">img</a></p>
<h3 id="cohesin-complex">cohesin complex</h3>
<p>Cohesin is a protein complex that regulates the separation of sister chromatids during cell division, either mitosis or meiosis.</p>
<p>Cohesins hold sister chromatids together after DNA replication until anaphase when removal of cohesin leads to separation of sister chromatids.</p>
<h3 id="ctcf-proteins">CTCF proteins</h3>
<p>转录阻抑物 CTCF</p>
<p>CTCF 与靶顺序因子的结合可阻断增强子和启动子的相互作用，从而将增强子的活性限制在一定的功能区域</p>
<p>除了阻断增强子外，CTCF 还可作为染色质屏障阻止异染色质的传播</p>
<h2 id="predicting-enhancer-promoter-loops-如何预测-epl">Predicting enhancer-promoter loops 如何预测 EPL</h2>
<h3 id="两种类似的算法">两种类似的算法</h3>
<p>TargetFinder(Whalen et al. Nat Gen 2016)— an algorithm that uses many functional genomic datasets, including DNase-seq, histone marks, transcription factor (TF) ChIP-seq, gene expression, and DNA methylation data etc.</p>
<ul>
<li>
<p>Enhancer–promoter interactions are encoded by complex genomic signatures on looping chromatin ,doi:10.1038/ng.3539</p>
</li>
<li>
<p>pipeline</p>
</li>
</ul>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-368541bebfbdc802.jpg" alt="img"></figure>
<p>RIPPLE (Roy et al. NAR 2016) — Also uses functional genomic datasets for feature extraction.</p>
<ul>
<li>A predictive modeling approach for cell line-specific long-range regulatory interactions , <a href="https://doi.org/10.1093/nar/gkv865">https://doi.org/10.1093/nar/gkv865</a></li>
</ul>
<p>二者共同的发现</p>
<ul>
<li>signals from these functional genomic data are informative to computationally distinguish enhancer-promoter interactions from noninteracting enhancer-promoter pairs.</li>
</ul>
<h3 id="pep-只用序列信息来进行分析马坚实验室">PEP 只用序列信息来进行分析（马坚实验室）</h3>
<figure data-type="image" tabindex="6"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-eb196cce2ead61a9.jpg" alt="img"></figure>
<h2 id="hi-c-分析流程">Hi-C 分析流程</h2>
<p>Analysis methods for studying the 3D architecture of the genome ，<a href="https://doi.org/10.1186/s13059-015-0745-7">https://doi.org/10.1186/s13059-015-0745-7</a></p>
<p>流程</p>
<figure data-type="image" tabindex="7"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-7df3e724f72f8706.jpg" alt="img"></figure>
<h2 id="contact-map">contact map</h2>
<p>定义：A contact map is a matrix with rows and columns representing non-overlapping ‘bins’ across the genome.</p>
<p>Each entry in the matrix contains a count of read pairs that connect the corresponding bin pair in a Hi-C experiment.</p>
<p>How to determine bin size</p>
<ul>
<li>No standard rule. Rao et al. 2014 suggests using a bin size that results in at least 80% of all possible bins with &gt;1000 contacts.</li>
</ul>
<p>Two types of approaches to correct bias in the contact map</p>
<ul>
<li>
<p>Explicit approach — assuming some known bias</p>
<pre><code>- Restriction enzyme fragment lengths, GC content, and sequence mappability are three major sources of biases in Hi-C data (Yaffe and Tanay, Nat Genet 2011)
- HiCNorm — simpler and faster (Hu et al. Bioinformatics 2012)
</code></pre>
</li>
<li>
<p>Implicit approach — assume no known source of bias and that each locus receives equal sequence coverage after biases are removed</p>
<pre><code>- In other words, if there is no bias, the total genome-wide contact summation for each locus will be a constant, i.e., each locus has 'equal visibility'
</code></pre>
</li>
</ul>
<h2 id="contact-matrix-normalization">Contact matrix normalization</h2>
<p>如何进行标准化</p>
<figure data-type="image" tabindex="8"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-8e50a48b35e02af0.jpg" alt="img"></figure>
<h2 id="鉴别-tad-的算法">鉴别 TAD 的算法</h2>
<p><strong>HMM（任兵）</strong></p>
<figure data-type="image" tabindex="9"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-ee87211eb4b5a3fd.jpg" alt="img"></figure>
<p><strong>Arrowhead</strong></p>
<figure data-type="image" tabindex="10"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-53bc76992ca74202.jpg" alt="img"></figure>
<hr>
<figure data-type="image" tabindex="11"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="12"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

            </div>
            
              <div class="tag-container">
                
                  <a href="https://kaopubear.top//tag/DvI6ypPftI" class="tag">
                    生物信息
                  </a>
                
              </div>
            
            
              <div class="next-post">
                <div class="next">下一篇</div>
                <a href="https://kaopubear.top//post/2018-08-05-ecluseevernote">
                  <h3 class="post-title">
                    一位印象大使的印象笔记使用心得
                  </h3>
                </a>
              </div>
            

            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>

<script type="application/javascript">

AOS.init();

hljs.initHighlightingOnLoad()

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>




  </body>
</html>
