<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1582900886089">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/remixicon@2.3.0/fonts/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div id="app" class="main px-4 flex flex-col lg:flex-row">
      <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="ri-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1582900886089" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E9%87%8D%E8%A6%81%E7%9A%84-hi-c-%E7%9B%B8%E5%85%B3%E6%96%87%E7%8C%AE">重要的 Hi-C 相关文献</a></li>
<li><a href="#chromatin-interaction-in-different-resolutions">Chromatin interaction in different resolutions</a></li>
<li><a href="#%E4%BB%80%E4%B9%88%E9%80%A0%E6%88%90%E4%BA%86%E6%89%80%E8%B0%93%E7%9A%84-tad">什么造成了所谓的 TAD</a>
<ul>
<li><a href="#cohesin-complex">cohesin complex</a></li>
<li><a href="#ctcf-proteins">CTCF proteins</a></li>
</ul>
</li>
<li><a href="#predicting-enhancer-promoter-loops-%E5%A6%82%E4%BD%95%E9%A2%84%E6%B5%8B-epl">Predicting enhancer-promoter loops 如何预测 EPL</a>
<ul>
<li><a href="#%E4%B8%A4%E7%A7%8D%E7%B1%BB%E4%BC%BC%E7%9A%84%E7%AE%97%E6%B3%95">两种类似的算法</a></li>
<li><a href="#pep-%E5%8F%AA%E7%94%A8%E5%BA%8F%E5%88%97%E4%BF%A1%E6%81%AF%E6%9D%A5%E8%BF%9B%E8%A1%8C%E5%88%86%E6%9E%90%E9%A9%AC%E5%9D%9A%E5%AE%9E%E9%AA%8C%E5%AE%A4">PEP 只用序列信息来进行分析（马坚实验室）</a></li>
</ul>
</li>
<li><a href="#hi-c-%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B">Hi-C 分析流程</a></li>
<li><a href="#contact-map">contact map</a></li>
<li><a href="#contact-matrix-normalization">Contact matrix normalization</a></li>
<li><a href="#%E9%89%B4%E5%88%AB-tad-%E7%9A%84%E7%AE%97%E6%B3%95">鉴别 TAD 的算法</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="ri-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球「靠谱熊基地」</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
      <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
        <h1 class="text-3xl font-bold lg:mt-16">3D 基因组与生物信息</h1>
        <div class="text-sm text-gray-700 lg:my-8">
          2018-08-11 / 4 min read
        </div>
        
          <img class="post-feature-image rounded-lg mx-auto my-4" src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-828599c54149be5d.jpg" alt="">
        
        <div class="post-content yue">
          <h2 id="重要的-hi-c-相关文献">重要的 Hi-C 相关文献</h2>
<p>第一篇 Hi-C 文章： Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome； DOI: 10.1126/science.1181369</p>
<!--more-->
<p>TAD 提出： Topological Domains in Mammalian Genomes Identified by Analysis of Chromatin Interactions ；doi:  10.1038/nature11082</p>
<p>高分辨率 Hi-C： A 3D Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping <a href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a></p>
<p>单细胞 Single cell： Hi-C reveals cell-to-cell variability in chromosome structure doi:  10.1038/nature12593；3D structures of individual mammalian genomes studied by single-cell Hi-C doi:10.1038/nature21429</p>
<p>综述：老师所讲 Hi-C 相关基础知识主要来自于综述 Organization and function of the 3D genome，doi:10.1038/nrg.2016.112</p>
<h2 id="chromatin-interaction-in-different-resolutions">Chromatin interaction in different resolutions</h2>
<p>不同分辨率 Hi-C 可以看到的内容不同</p>
<p>5KB 可以看到各种 loop</p>
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-828599c54149be5d.jpg" alt="" loading="lazy"></figure>
<p>10KB 可以看到 TAD</p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-02e99cd9e50fa311.jpg" alt="" loading="lazy"></figure>
<p>50kb 可以看到 TAD 之间的关联</p>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-44370d2fbc86ef43.jpg" alt="" loading="lazy"></figure>
<p>在整个染色体的水平可以看到染色质的位置分布</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-c2baa72eda2f9bb2.jpg" alt="" loading="lazy"></figure>
<h2 id="什么造成了所谓的-tad">什么造成了所谓的 TAD</h2>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/e556f4a1-7f13-4a38-8943-9dc29cfb9314.jpg" alt="" loading="lazy"></figure>
<h3 id="cohesin-complex">cohesin complex</h3>
<p>Cohesin is a protein complex that regulates the separation of sister chromatids during cell division, either mitosis or meiosis.</p>
<p>Cohesins hold sister chromatids together after DNA replication until anaphase when removal of cohesin leads to separation of sister chromatids.</p>
<h3 id="ctcf-proteins">CTCF proteins</h3>
<p>转录阻抑物 CTCF</p>
<p>CTCF 与靶顺序因子的结合可阻断增强子和启动子的相互作用，从而将增强子的活性限制在一定的功能区域</p>
<p>除了阻断增强子外，CTCF 还可作为染色质屏障阻止异染色质的传播</p>
<h2 id="predicting-enhancer-promoter-loops-如何预测-epl">Predicting enhancer-promoter loops 如何预测 EPL</h2>
<h3 id="两种类似的算法">两种类似的算法</h3>
<p>TargetFinder(Whalen et al. Nat Gen 2016)— an algorithm that uses many functional genomic datasets, including DNase-seq, histone marks, transcription factor (TF) ChIP-seq, gene expression, and DNA methylation data etc.</p>
<ul>
<li>
<p>Enhancer–promoter interactions are encoded by complex genomic signatures on looping chromatin ,doi:10.1038/ng.3539</p>
</li>
<li>
<p>pipeline</p>
</li>
</ul>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-368541bebfbdc802.jpg" alt="" loading="lazy"></figure>
<p>RIPPLE (Roy et al. NAR 2016) — Also uses functional genomic datasets for feature extraction.</p>
<ul>
<li>A predictive modeling approach for cell line-specific long-range regulatory interactions , <a href="https://doi.org/10.1093/nar/gkv865">https://doi.org/10.1093/nar/gkv865</a></li>
</ul>
<p>二者共同的发现</p>
<ul>
<li>signals from these functional genomic data are informative to computationally distinguish enhancer-promoter interactions from noninteracting enhancer-promoter pairs.</li>
</ul>
<h3 id="pep-只用序列信息来进行分析马坚实验室">PEP 只用序列信息来进行分析（马坚实验室）</h3>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-eb196cce2ead61a9.jpg" alt="" loading="lazy"></figure>
<h2 id="hi-c-分析流程">Hi-C 分析流程</h2>
<p>Analysis methods for studying the 3D architecture of the genome ，<a href="https://doi.org/10.1186/s13059-015-0745-7">https://doi.org/10.1186/s13059-015-0745-7</a></p>
<p>流程</p>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-7df3e724f72f8706.jpg" alt="" loading="lazy"></figure>
<h2 id="contact-map">contact map</h2>
<p>定义：A contact map is a matrix with rows and columns representing non-overlapping ‘bins’ across the genome.</p>
<p>Each entry in the matrix contains a count of read pairs that connect the corresponding bin pair in a Hi-C experiment.</p>
<p>How to determine bin size</p>
<ul>
<li>No standard rule. Rao et al. 2014 suggests using a bin size that results in at least 80% of all possible bins with &gt;1000 contacts.</li>
</ul>
<p>Two types of approaches to correct bias in the contact map</p>
<ul>
<li>
<p>Explicit approach — assuming some known bias</p>
<pre><code>- Restriction enzyme fragment lengths, GC content, and sequence mappability are three major sources of biases in Hi-C data (Yaffe and Tanay, Nat Genet 2011)
- HiCNorm — simpler and faster (Hu et al. Bioinformatics 2012)
</code></pre>
</li>
<li>
<p>Implicit approach — assume no known source of bias and that each locus receives equal sequence coverage after biases are removed</p>
<pre><code>- In other words, if there is no bias, the total genome-wide contact summation for each locus will be a constant, i.e., each locus has 'equal visibility'
</code></pre>
</li>
</ul>
<h2 id="contact-matrix-normalization">Contact matrix normalization</h2>
<p>如何进行标准化</p>
<figure data-type="image" tabindex="9"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-8e50a48b35e02af0.jpg" alt="" loading="lazy"></figure>
<h2 id="鉴别-tad-的算法">鉴别 TAD 的算法</h2>
<p><strong>HMM（任兵）</strong></p>
<figure data-type="image" tabindex="10"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-ee87211eb4b5a3fd.jpg" alt="" loading="lazy"></figure>
<p><strong>Arrowhead</strong></p>
<figure data-type="image" tabindex="11"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-53bc76992ca74202.jpg" alt="" loading="lazy"></figure>
<hr>
<figure data-type="image" tabindex="12"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="13"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

        </div>

        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/DvI6ypPftI/">
            <span class="flex-auto">生物信息</span>
          </a>
        


        <div class="flex justify-between py-8">
          
            <div class="prev-post">
              <a href="https://kaopubear.top/post/2018-08-15-bioinfohistory/">
                <h3 class="post-title">
                  <i class="ri-arrow-left-line"></i>
                  生物信息学简史
                </h3>
              </a>
            </div>
          

          
            <div class="next-post">
              <a href="https://kaopubear.top/post/2018-08-05-ecluseevernote/">
                <h3 class="post-title">
                  一位印象大使的印象笔记使用心得
                  <i class="ri-arrow-right-line"></i>
                </h3>
              </a>
            </div>
          
        </div>

        

      </div>
    </div>

    <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()
let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
  </body>
</html>
