<html lang="en">
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<script src="https://kaopubear.top//media/script.js"></script>


  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
  
    gtag('config', 'UA-91022805-1');
  </script>
    

<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<script src="https://kaopubear.top//media/script.js"></script>


  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
  
    gtag('config', 'UA-91022805-1');
  </script>
    

  <script src="https://kaopubear.top//media/prism.js"></script>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
</head>
<body>
  <div class="antialiased">
    <div class="max-w-5xl mx-auto px-6 md:px-8">
      <div class="flex justify-between items-center mt-10">
  <h1 class="text-lg sm:text-xl md:text-2xl lg:text-3xl text-gray-900 font-medium md:font-normal leading-none">
    <a href="https://kaopubear.top/">思考问题的熊</a>
  </h1>
  <div class="block lg:hidden" id="thumb-menu">
    <button id="thumb-open" type="button" class="block">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M2 6H22V7H2V6ZM2 12H22V13H2V12ZM22 18H2V19H22V18Z" class="fill-current text-gray-900"></path></svg>
    </button>
    <button id="thumb-close" type="button" class="hidden">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M4 4L20 20M20 4L4 20" class="" stroke="#1A202C"></path></svg>
    </button>
  </div>
</div>
<div id="nav-list" class="lg:block hidden">
  <div class="w-full block lg:flex lg:justify-between lg:border-b lg:border-gray-400 mt-10 text-gray-700 font-medium lg:text-lg">
    <div class="lg:flex -mb-px">
      
        
          <a href="/" class="nav-link">首页</a>
        
      
        
          <a href="/archives" class="nav-link">归档</a>
        
      
        
          <a href="/tags" class="nav-link">标签</a>
        
      
        
          <a href="/post/about" class="nav-link">关于</a>
        
      
        
          <a href="/post/resource" class="nav-link">资料</a>
        
      
        
          <a href="/post/project" class="nav-link">项目</a>
        
      
    </div>
    <div class="lg:flex">
      
        
          <a href="https://github.com/fei0810" class="block lg:mt-0 lg:ml-8 flex align-center hover:text-gray-900" target="_blank">
            <div>GitHub</div>
            <svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg" class="ml-1 -mt-px"><path d="M16.5 13V18.5H6.5V8.5H12" stroke="#4A5568"></path> <path d="M10 15L18.5 6.5" stroke="#4A5568"></path> <path d="M14 6.5H18.5V11" stroke="#4A5568"></path></svg>
          </a>
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
      <a href="https://kaopubear.top//atom.xml" class="block mt-6 lg:mt-0 lg:ml-8 flex align-center hover:text-gray-900" target="_blank">
        <div>RSS</div>
        <svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg" class="ml-1 -mt-px"><path d="M16.5 13V18.5H6.5V8.5H12" stroke="#4A5568"></path> <path d="M10 15L18.5 6.5" stroke="#4A5568"></path> <path d="M14 6.5H18.5V11" stroke="#4A5568"></path></svg>
      </a>
    </div>
  </div>
</div>
      <div class="mb-20 md:flex mt-10 lg:mt-6 md:border-t md:border-gray-400 md:pt-2 lg:border-t-0 lg:pt-0">
        <div class="w-full md:w-2/3 mt-6">
          <h1 class="text-4xl">每周文献-190419-植物单细胞BAM重比对以及假基因研究</h1>
          
            <img class="max-w-full mt-6 rounded-sm" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606000459.png" alt="">
          
          <div class="mt-6 lg:mt-10 post-content">
            <h2 id="a-single-cell-rna-sequencing-profiles-the-developmental-landscape-of-arabidopsis-root">A Single-Cell RNA Sequencing Profiles the Developmental Landscape of Arabidopsis Root</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1016/j.molp.2019.04.004</p>
<p><strong>发表日期</strong>：April 17, 2019</p>
<h3 id="关键点">关键点</h3>
<p>国内首篇植物相关单细胞文章，两个一作还都很熟。</p>
<h3 id="参考意义">参考意义</h3>
<p>植物单细胞的阶段要开启了，今天提到的这篇文章是国内首篇，也是世界范围内的第四篇植物单细胞文章。除此之外，还有几篇已经在 bioRxiv 上上线了，不过还没有正式发表。这些文章都不约而同选择了植物中研究最广的模式之物拟南芥，而且全部研究的是根尖。后面在发展就要看看其它组织和物种的情况了。</p>
<p>如果一个东西在植物里已经出现要快速增长的趋势，那么它在人和动物里应该也就已经相对比较成熟了。此时，即便暂时还用不到，但是相关的技术和方法就需要留意和学习起来。</p>
<!--more-->
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606000700.png" alt=""></figure>
<p>在这篇文章中，作者在单细胞水平揭示了拟南芥根尖细胞的异质性重构了根尖分生组织细胞的发育轨迹。按照文章的说法，成功拿到了 7695 个根单细胞转录组数据。聚类后根细胞被划分为 24 个细胞类群，细胞类群注释分析鉴定了一些潜在的新细胞类型，并找到了一批细胞类型标记基因。</p>
<p>用 t-SNE 和 UMAP 重构了根发育的基本轨迹，实现了根分生组织细胞分裂和分化在单细胞水平上的准确投影。进一步利用伪时间（pseudo-time）分析，捕获了根尖分生组织细胞的分化轨迹和过渡态细胞，解析了根分生组织细胞如何通过协调细胞分裂和分化进程逐步形成根尖不同细胞类型的分子机理。此外，通过分析细胞类群对离子吸收和激素响应情况，揭示了不同根细胞类群的响应热图。该研究加深了我们对拟南芥根细胞组成和发育轨迹的认识，将根发育生物学从原先的组织器官水平提升到了单细胞水平。</p>
<p>(上面这一段话主要来自官方报道，具体的细节需要仔细读完文章在分享)</p>
<h3 id="相关内容">相关内容</h3>
<p>目前已经正式发表的其它三篇植物单细胞文章，看下来会发现分析思路和 figure 都异曲同工。</p>
<ul>
<li>Single-Cell RNA Sequencing Resolves Molecular Relationships Among Individual Plant Cells</li>
</ul>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606000942.png" alt=""></figure>
<ul>
<li>Spatiotemporal Developmental Trajectories in the Arabidopsis Root Revealed Using High-Throughput Single-Cell RNA Sequencing</li>
</ul>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001100.png" alt=""></figure>
<ul>
<li>Open Access Dynamics of gene expression in single root cells of A. thaliana</li>
</ul>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001218.png" alt=""></figure>
<h2 id="benchmarking-of-alignment-free-sequence-comparison-methods">Benchmarking of alignment-free sequence comparison methods</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/611137</p>
<p><strong>发表日期</strong>：April 16, 2019.</p>
<h3 id="关键点-2">关键点</h3>
<p>有哪些 alignment-free (AF) 相关的工具，以及如何评价。</p>
<h3 id="参考意义-2">参考意义</h3>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001228.png" alt=""></figure>
<p>AF 类的工具，在转录组分析层面使用最多的是定量分析。例如 salmon 和 kallisto，主要原理就是基于对 kmer 的各种操作。其实除了转录组的快速定量</p>
<p>这篇文章比较详细的介绍了目前主要 AF 相关工具的原理和工具。同时，作者使用了 24 个相关软件的 74 种方法，测试了五种应用场景，分别是：</p>
<ul>
<li>protein sequence classification</li>
<li>gene tree inference</li>
<li>regulatory element detection</li>
<li>genome-based phylogenetic inference</li>
<li>reconstruction of species trees under horizontal gene transfer and recombination events</li>
</ul>
<p>作者还提供了一个在线工具，用来展示这些结果。</p>
<h3 id="相关内容-2">相关内容</h3>
<ul>
<li><a href="https://github.com/kdmurray91/kWIP">kWIP</a></li>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1632-4">Skmer</a></li>
<li><a href="http://dx.doi.org/10.1101/549154"><strong>Kevlar</strong></a></li>
</ul>
<h2 id="bazam-a-rapid-method-for-read-extraction-and-realignment-of-high-throughput-sequencing-data">Bazam: a rapid method for read extraction and realignment of high-throughput sequencing data</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1186/s13059-019-1688-1</p>
<p><strong>发表日期</strong>：18 April 2019</p>
<h3 id="关键点-3">关键点</h3>
<p>bam 文件似乎可以方便的回滚了</p>
<h3 id="参考意义-3">参考意义</h3>
<p>随着参考基因组的更新和比对方法的更新，很多之前的 bam 文件似乎就变得过时了。除了找出原始的 fastq 文件再重新来过一次，现在有了另一个选择。<br>
bazam 首先可以从 bam 或者 cram 文件直接找到 pair reads 在比对回其它参考基因组而不需要中间步骤； 也可以按需要提取过滤后的 reads，例如与特定基因位置有 overlap 的 reads。结果可以直接传到下游工具，或以 fastq 格式存储以供进一步处理。另外还从 read 这个 input 层面提供了多线程比对的思路。加快了比对速度。</p>
<p>整体而言，和目前已有的一些可以转换 bam 文件的工具相比，其在内存和存储占用，已经方便程度上都有优势。</p>
<figure data-type="image" tabindex="6"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001239.png" alt=""></figure>
<p>目前我的问题是很多时候会对原始 bam 文件进行一波过滤，这个时候已经丢掉了很多 fastq 的 reads。</p>
<h3 id="相关内容-3">相关内容</h3>
<p>其它几个已有工具的比较</p>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Storage used</th>
<th>Memory</th>
<th>Effective Cores</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sort-Extract-Realign</td>
<td>282 GB</td>
<td>20 GB</td>
<td>16</td>
<td>13 h, 15 min</td>
</tr>
<tr>
<td>Picard SamToFastq</td>
<td>148 GB</td>
<td>78 GB</td>
<td>16</td>
<td>16 h, 14 min</td>
</tr>
<tr>
<td>Biobambam bamtofastq</td>
<td>149 GB</td>
<td>30 GB</td>
<td>16</td>
<td>15 h 30 min</td>
</tr>
<tr>
<td>Bazam (no sharding)</td>
<td>68 GB</td>
<td>28 GB</td>
<td>16</td>
<td>14 h, 55 min</td>
</tr>
<tr>
<td>Bazam 10-way sharding</td>
<td>102 GB</td>
<td>20 GB</td>
<td>160</td>
<td>1 h, 11 min</td>
</tr>
</tbody>
</table>
<ul>
<li><a href="https://github.com/ssadedin/bazam">bazam</a></li>
</ul>
<h2 id="evolutionary-origins-of-pseudogenes-and-their-association-with-regulatory-sequences-in-plants">Evolutionary Origins of Pseudogenes and Their Association with Regulatory Sequences in Plants</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1105/tpc.18.00601</p>
<p><strong>发表日期</strong>：March 2019</p>
<h3 id="关键点-4">关键点</h3>
<p>在我看来，假基因和 lncRNA 这类非编码 RNA 其实大多数是定义方式不同，重合度不低。</p>
<h3 id="参考意义-4">参考意义</h3>
<p>假基因（Ψs）一般是和功能基因的序列相近的非功能性基因，通过复制或逆转录方式形成，通常会含有各种突变导致基因功能的丧失。在这篇文章中，作者检查了七种被子植物（拟南芥，短柄草，大豆，苜蓿，水稻，杨树和高粱）假基因的起源，进化和表达模式及其与非编码序列的关系。作者鉴定了大约 250,000 个假基因，发现非常大比例的非转座因子调控非编码 RNA（microRNA 和 lncRNA）起源于假基因近端上游区域的转录。 还发现与随机基因间区相比转录因子结合位点优先发生在假基因近端上游区域，这表明假基因可能通过提供用作启动子和增强子的转录因子结合位点来调节基因组进化。</p>
<p>假基因定义流程：</p>
<figure data-type="image" tabindex="7"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001315.png" alt=""></figure>
<p>假基因鉴定情况</p>
<figure data-type="image" tabindex="8"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606001355.png" alt=""></figure>
<h3 id="相关内容-4">相关内容</h3>
<p>主要鉴定步骤</p>
<ul>
<li>(1) identify intergenic regions (masked genic and transposon regions) with sequence similarity to known proteins using exonerate;</li>
<li>(2) quality control, identity ≥ 20%, match length ≥ 30 amino acids, match length ≥ 5% of the query sequence, and only the best match is retained;
<ul>
<li>exonerate <code>--model protein2genome --showquerygff no --showtargetgff yes --maxintron 5000 --showvulgar yes --ryo \&quot;%ti\\t%qi\\t%tS\\t%qS\\t%tl\\t%ql\\t%tab\\t%tae\\t%tal\\t%qab\\t%qae\\t%qal\\t%pi\\n\&quot;.</code></li>
</ul>
</li>
<li>(3) link homologous segments into contigs (set I Ψs);</li>
<li>(4) realign using tfasty to identify features that disrupt contiguous protein sequences
<ul>
<li>tfasty34, with parameters <code>-A -m 3 q</code></li>
</ul>
</li>
<li>(5) distinguish WGD-derived Ψs and set II Ψs.
<ul>
<li>MCScanX <code>-k 50 -g -1 -s 5 -m 25</code></li>
</ul>
</li>
</ul>
<p>以及</p>
<p><a href="https://github.com/bjfupoplar/PlantPseudo">PlantPseudo</a></p>
<hr>
<figure data-type="image" tabindex="9"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名 - 非商业性使用 - 禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="10"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

          </div>
          
        </div>
        
          <div class="w-full md:w-1/3">
            <div class="mt-6 md:pl-8">
              <h4 class="text-base font-thin text-gray-700 mb-2">发布时间</h4>
              <div class="text-lg mb-6">2019-04-19</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">标签</h4>
              <div class="mb-6">
                
                  <a href="https://kaopubear.top//tag/EqrCpMmApu" class="py-1 px-4 border-gray-300 border text-sm inline-block text-gray-700 hover:text-gray-900 rounded mr-2 mb-2">文献</a>
                
              </div>
              <h4 class="text-base font-thin text-gray-700 mb-2">阅读时间</h4>
              <div class="text-lg mb-6">8 min read</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">相关阅读</h4>
              <div class="border-b border-gray-400">
                

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://kaopubear.top//post/2019-09-11-howtolearnzotero" class="text-lg hover:underline my-2 inline-block">Zotero入门学习路径</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://kaopubear.top//post/2019-06-22-weeklypaper" class="text-lg hover:underline my-2 inline-block">每周文献-190622-ATAC-seq和转录本从头组装新方法</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://kaopubear.top//post/2019-06-06-weeklypaper" class="text-lg hover:underline my-2 inline-block">每周文献-190606-多篇结构变异和转录组分析方法文章</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  


              </div>
            </div>
          </div>
        
      </div>
      <div class="footer border-gray-400 border-t pt-6 mt-6 sm:mt-8 md:mt-10 mb-20 text-gray-700 text-sm sm:text-base lg:text-lg">
  欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a>
</div>
    </div>
  </div>
  <script>
    Prism.highlightAll()
  </script>
</body>
</html>