<html lang="en">
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://fei0810.github.io/styles/main.css">
<script src="https://fei0810.github.io/media/script.js"></script>

    

<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://fei0810.github.io/styles/main.css">
<script src="https://fei0810.github.io/media/script.js"></script>

    

  <script src="https://fei0810.github.io/media/prism.js"></script>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
</head>
<body>
  <div class="antialiased">
    <div class="max-w-5xl mx-auto px-6 md:px-8">
      <div class="flex justify-between items-center mt-10">
  <h1 class="text-lg sm:text-xl md:text-2xl lg:text-3xl text-gray-900 font-medium md:font-normal leading-none">
    <a href="https://fei0810.github.io">思考问题的熊</a>
  </h1>
  <div class="block lg:hidden" id="thumb-menu">
    <button id="thumb-open" type="button" class="block">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M2 6H22V7H2V6ZM2 12H22V13H2V12ZM22 18H2V19H22V18Z" class="fill-current text-gray-900"></path></svg>
    </button>
    <button id="thumb-close" type="button" class="hidden">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M4 4L20 20M20 4L4 20" class="" stroke="#1A202C"></path></svg>
    </button>
  </div>
</div>
<div id="nav-list" class="lg:block hidden">
  <div class="w-full block lg:flex lg:justify-between lg:border-b lg:border-gray-400 mt-10 text-gray-700 font-medium lg:text-lg">
    <div class="lg:flex -mb-px">
      
        
          <a href="/" class="nav-link">首页</a>
        
      
        
          <a href="/archives" class="nav-link">归档</a>
        
      
        
          <a href="/tags" class="nav-link">标签</a>
        
      
        
          <a href="/post/about" class="nav-link">关于</a>
        
      
        
          <a href="/post/resource" class="nav-link">资料</a>
        
      
    </div>
    <div class="lg:flex">
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
      <a href="https://fei0810.github.io/atom.xml" class="block mt-6 lg:mt-0 lg:ml-8 flex align-center hover:text-gray-900" target="_blank">
        <div>RSS</div>
        <svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg" class="ml-1 -mt-px"><path d="M16.5 13V18.5H6.5V8.5H12" stroke="#4A5568"></path> <path d="M10 15L18.5 6.5" stroke="#4A5568"></path> <path d="M14 6.5H18.5V11" stroke="#4A5568"></path></svg>
      </a>
    </div>
  </div>
</div>
      <div class="mb-20 md:flex mt-10 lg:mt-6 md:border-t md:border-gray-400 md:pt-2 lg:border-t-0 lg:pt-0">
        <div class="w-full md:w-2/3 mt-6">
          <h1 class="text-4xl">每周文献-190606-多篇结构变异和转录组分析方法文章</h1>
          
          <div class="mt-6 lg:mt-10 post-content">
            <h2 id="alignment-and-mapping-methodology-influence-transcript-abundance-estimation">Alignment and mapping methodology influence transcript abundance estimation</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/657874</p>
<p><strong>发表日期</strong>：June 03, 2019</p>
<h3 id="关键点">关键点</h3>
<p>不对比对方法对转录本定量的影响有哪些（读完感觉是给 Salmon 最近一次升级写的软文）</p>
<h3 id="参考意义">参考意义</h3>
<p>使用 RNA-seq 数据进行转录本定量的准确性取决于许多因素，比如比对的方法和所采用的定量模型。虽然有不少文章已经讲过定量模型的重要性，但比较各种比对方法对定量准确度的影响并没有那么受关注。作者在这篇文章中研究了比对方法对定量准确性以及对差异基因表达分析的影响。</p>
<!--more-->
<p>即使定量模型本身不变，选择不同的比对方法，或使用不同的参数对定量的影响有时可能很大并影响下游分析。作者也强调当评估过于注重模拟数据时，这些影响可能会被我们忽视，因为在模拟数据中，比对这一步往往比实验获得的样本更简单。文章讨论了用于定量目的的最佳比对方法，同时也引入了一种新的混合比对方法，称为 <strong>selective alignment</strong>(SA)。</p>
<p>文章中，作者选择了三种比对策略：</p>
<ul>
<li>unspliced alignment of RNA-seq reads directly to the transcriptome</li>
<li>spliced alignment of RNA-seq reads to the annotated genome (with subsequent projection to the transcriptome)</li>
<li>(unspliced) lightweight mapping (quasi-mapping) of the RNA-seq reads directly to the transcriptome</li>
</ul>
<p>具体的比对方法：</p>
<ul>
<li>Bowtie2 – Alignment with Bowtie2 to the target transcriptome and allowing alignments with indels, followed by quantification using Salmon in alignment mode.</li>
<li>Bowtie2 strict – Alignment with Bowtie2 to the target transcriptome and disallowing alignments with indels (i.e. using the same parameters as those used by RSEM), followed by quantification using Salmon in alignment mode.</li>
<li>Bowtie2 RSEM – Alignment with Bowtie2 to the target transcriptome and disallowing alignments with indels, followed by quantification using RSEM.</li>
<li>STAR – Alignment with STAR to the target genome (aided with the GTF annotation of the transcriptome) and projected to the transcriptome allowing alignments with indels and soft clipping, followed by quantification using Salmon in alignment mode.</li>
<li>STAR strict – Alignment with STAR to the target genome (aided with the GTF annotation of the transcriptome) and projected to the transcriptome and disallowing alignments with indels or soft clipping, followed by quantification using Salmon in alignment mode.</li>
<li>STAR RSEM – Alignment with STAR to the target genome (aided with the GTF annotation of the transcriptome) and projected to the transcriptome and disallowing alignments with indels or soft clipping, followed by quantification using RSEM.</li>
<li>quasi – Quasi-mapping directly to the target transcriptome, coupled with quantification using Salmon in non-alignment mode.</li>
<li>SA– Selective alignment directly to the target transcriptome and a set of decoy sequences, coupled with quantification using Salmon in non-alignment mode.</li>
</ul>
<h3 id="相关内容">相关内容</h3>
<pre><code class="language-sh"># For indexing, we use the following extra command line arguments, along with the regular indexing and
threads parameters:
STAR --genomeFastaFiles &lt;fasta file&gt; --sjdbGTFfile &lt;gtf file&gt; --sjdbOverhang 100
Bowtie2 default
salmon -k 23 --keepDuplicates
kallisto -k 23
# For quantification, we use the following extra command line, \
# along with regular index and threads, with each tools we compare against:
SA --mimicBT2 --useEM
quasi --rangeFactorization 4 --discardOrphansQuasi --useEM
Bowtie2 --sensitive -k 200 -X 1000 --no-discordant --no-mixed
Bowtie2 strict --sensitive --dpad 0 --gbar 99999999 --mp 1,1 \
--np 1 --score-min L,0,-0.1 --no-mixed --no-discordant -k 200 -I 1 -X 1000
Bowtie2 RSEM --sensitive --dpad 0 --gbar 99999999 --mp 1,1 \
--np 1 --score-min L,0,-0.1 --no-mixed --no-discordant -k 200 -I 1 -X 1000

STAR --outFilterType BySJout --alignSJoverhangMin 8 --outFilterMultimapNmax 20 \
--alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 \
--outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 \
--alignMatesGapMax 1000000 --readFilesCommand zcat --outSAMtype BAM Unsorted \
--quantMode TranscriptomeSAM --outSAMattributes NH HI AS NM MD \
--quantTranscriptomeBan Singleend

STAR strict --outFilterType BySJout --alignSJoverhangMin 8 --outFilterMultimapNmax \
20 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 \
--outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 \
--alignMatesGapMax 1000000 --readFilesCommand zcat --outSAMtype BAM Unsorted \
--quantMode TranscriptomeSAM --outSAMattributes NH HI AS NM MD \
--quantTranscriptomeBan IndelSoftclipSingleend

STAR RSEM --outFilterType BySJout --alignSJoverhangMin 8 --outFilterMultimapNmax \
20 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 \
--outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 \
--alignMatesGapMax 1000000 --readFilesCommand zcat --outSAMtype BAM Unsorted \
--quantMode TranscriptomeSAM --outSAMattributes NH HI AS NM MD \
--quantTranscriptomeBan IndelSoftclipSingleend \

RSEM default

kallisto default or --rf-stranded as appropriate
</code></pre>
<p>这里所谓的 SA 模式其实就是在 salmon 最近升级中加入的一个参数，具体可以了解 <a href="https://github.com/COMBINE-lab/salmon/releases/tag/v0.14.0">官方说明</a></p>
<h2 id="a-practical-guide-to-methods-controlling-false-discoveries-in-computational-biology">A practical guide to methods controlling false discoveries in computational biology</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1186/s13059-019-1716-1</p>
<p><strong>发表日期</strong>：4 June 2019</p>
<h3 id="关键点-2">关键点</h3>
<p>在数据分析的过程中如何更好的控制 false discoveries</p>
<h3 id="参考意义-2">参考意义</h3>
<p>以下是 8 中可用的 FDR-controlling methods ，其中 IHW 和 BL 是考虑了协变量的现代方法。</p>
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606123631.png" alt=""></figure>
<p>不同方法的适用性评价，从结果来看更加推荐使用 IHW 和 BL 这两种方法。</p>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606124932.png" alt=""></figure>
<h3 id="相关内容-2">相关内容</h3>
<p>Independent and informative covariates used in case studies</p>
<table>
<thead>
<tr>
<th>Case study</th>
<th>Covariates found to be independent and informative</th>
</tr>
</thead>
<tbody>
<tr>
<td>Microbiome</td>
<td><strong>Ubiquity</strong>: the proportion of samples in which the feature is present. In microbiome data, it is common for many features to go undetected in many samples.</td>
</tr>
<tr>
<td></td>
<td><strong>Mean nonzero abundance</strong>: the average abundance of a feature among those samples in which it was detected. We note that this did not seem as informative as ubiquity in our case studies.</td>
</tr>
<tr>
<td>GWAS</td>
<td><strong>Minor allele frequency</strong>: the proportion of the population which exhibits the less common allele (ranges from 0 to 0.5) represents the rarity of a particular variant.</td>
</tr>
<tr>
<td></td>
<td><strong>Sample size (for meta-analyses)</strong>: the number of samples for which the particular variant was measured.</td>
</tr>
<tr>
<td>Gene set analyses</td>
<td><strong>Gene set size</strong>: the number of genes included in the particular set. Note that this is not independent under the null for over-representation tests, however (see Additional file <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1#MOESM1">1</a>: Supplementary Results).</td>
</tr>
<tr>
<td>Bulk RNA-seq</td>
<td><strong>Mean gene expression</strong>: the average expression level (calculated from normalized read counts) for a particular gene.</td>
</tr>
<tr>
<td>Single-Cell RNA-seq</td>
<td><strong>Mean nonzero gene expression</strong>: the average expression level (calculated from normalized read counts) for a particular gene, excluding zero counts.</td>
</tr>
<tr>
<td></td>
<td><strong>Detection rate</strong>: the proportion of samples in which the gene is detected. In single-cell RNA-seq it is common for many genes to go undetected in many samples.</td>
</tr>
<tr>
<td>ChIP-seq</td>
<td><strong>Mean read depth</strong>: the average coverage (calculated from normalized read counts) for the region</td>
</tr>
<tr>
<td></td>
<td><strong>Window Size</strong>: the length of the region</td>
</tr>
</tbody>
</table>
<p><a href="http://bioconductor.org/packages/release/bioc/html/swfdr.html">Science-wise false discovery rate and proportion of true null hypotheses estimation<br>
</a></p>
<p><a href="http://bioconductor.org/packages/release/bioc/html/IHW.html">Independent Hypothesis Weighting</a></p>
<h2 id="a-simple-deep-learning-approach-for-detecting-duplications-and-deletions-in-next-generation-sequencing-data">A Simple Deep Learning Approach for Detecting Duplications and Deletions in Next-Generation Sequencing Data</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/657361</p>
<p><strong>发表日期</strong>：June 03, 2019</p>
<h3 id="关键点-3">关键点</h3>
<p>使用机器学习方法在低丰度数据中鉴定 CNV</p>
<h3 id="参考意义-3">参考意义</h3>
<p>拷贝数变异 (CNV) 的检测仍然是一个难题，特别是在质量比较查或覆盖率较低的二代测序数据中。这篇文章介绍了一种在二代测序数据中检测 CNV 的方法。在低覆盖读数据中，机器学习在检测 CNV 方面似乎比之前的 gold-standard 更加准确，在高覆盖率数据中两者效果相当。甚至可以在以前使用长读数的数据中鉴定到新的 CNV。</p>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606133439.png" alt=""></figure>
<h3 id="相关内容-3">相关内容</h3>
<p>更多信息可以查看工具 <a href="https://github.com/tomh1lll/dudeml">GitHub 地址</a>，不过如果使用机器学习的方法应该最起码有已知的 CNV 信息才可以，这个东西从哪里来呢。还是使用其他工具据预测？</p>
<h2 id="comprehensive-evaluation-of-structural-variation-detection-algorithms-for-whole-genome-sequencing">Comprehensive evaluation of structural variation detection algorithms for whole genome sequencing</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1186/s13059-019-1720-5</p>
<p><strong>发表日期</strong>：3 June 2019</p>
<h3 id="关键点-4">关键点</h3>
<p>综合评估全基因组测序的结构变异检测算法</p>
<h3 id="参考意义-4">参考意义</h3>
<p>结构变异（SV）或拷贝数变异（CNV）极大地影响基因组中编码基因的功能并且和多种疾病有关。尽管许多现有的 SV 检测算法可以使用全基因组测序（WGS）数据检测多种类型的 SV，但是没有一种算法能够以高的 precision 和 recall 鉴定每种类型的 SV。</p>
<p>本文作者使用多个模拟和真实的 WGS 数据集评估了 69 个现有 SV 检测算法的表现。分析结果显示有一组算法根据 SV 的特定类型和大小范围准确鉴别 SV，并可以准确地确定 SV 的断点，大小和基因型。文中列举了针对每类 SV 优秀算法，其中 GRIDSS，Lumpy，SVseq2，SoftSV，Manta 和 Wham 是 deletion 或 duplication 这类 SV 更好的算法。</p>
<p>下图 A 是模拟数据，B 是真是数据，不同颜色代表不同的突变类型，包括插入、重复、到位和易位。检测 SV 的算法被分为以下几类：RP, read pairs; SR, split reads; RD, read depth; AS, assembly; LR, long reads 。以及他们的不同组合方法 RP-SR, RP-RD, RP-AS, RP-SR-AS 和 RP-SR-RD。</p>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606133451.png" alt=""></figure>
<p>针对不同长度不同工具的表现如下：</p>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606130804.png" alt=""></figure>
<p>SV 检测算法的运行时间和内存消耗如下图：</p>
<figure data-type="image" tabindex="6"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606131338.png" alt=""></figure>
<h3 id="相关内容-4">相关内容</h3>
<p>文章中使用到的软件信息：</p>
<figure data-type="image" tabindex="7"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606133500.png" alt=""></figure>
<p>所用软件的用法参数：<a href="https://static-content.springer.com/esm/art%3A10.1186%2Fs13059-019-1720-5/MediaObjects/13059_2019_1720_MOESM4_ESM.pdf">附件下载</a></p>
<h2 id="using-multiple-reference-genomes-to-identify-and-resolve-annotation-inconsistencies">Using multiple reference genomes to identify and resolve annotation inconsistencies</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1101/651984</p>
<p><strong>发表日期</strong>：May 30, 2019.</p>
<h3 id="关键点-5">关键点</h3>
<p>近似基因组间基因错误注释情况分析</p>
<h3 id="参考意义-5">参考意义</h3>
<p>大家越来越有钱，各种基因组测序结果越来越多。例如在植物中，往往一个物种就会存在很多个不同品种的基因组序列。虽然这些新基因组每一个都在彼此之间有很多共线性部分，但这些区域内的基因的注释结构却通常存在各种不同。有一种情况是 split-gene 的错误注释，也就是一个基因被错误地注释为两个不同的基因或两个基因被错误地注释为一个基因。这些错误注释可能对功能预测、定量分析以及许多下游分析产生重大影响。</p>
<p>本文作者开发了一种基于两两比较注释的高通量分析方法，可以检测潜在的分裂基因情况并评估不同基因是否应该合并为单个基因。文章使用来自玉米（B73，PH207 和 W22）的三个参考基因组的基因注释证明了方法的实用性。在每个两两比较中发现数百个潜在的分裂基因错误注释情况，对应于 3-5％的注释基因。同时还利用来自 10 种组织的 RNAseq 数据确定生物学上支持哪种状态。</p>
<figure data-type="image" tabindex="8"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606133507.png" alt=""></figure>
<h3 id="相关内容-5">相关内容</h3>
<p><a href="https://github.com/HirschLabUMN/Split_genes/">Split-gene misannotation pipeline</a></p>
<p><a href="https://github.com/HirschLabUMN/Split_genes/blob/master/analysis/SplitGenes.Rmd">The script that generated the tables, figures, and numbers</a></p>
<h2 id="ngsep3-accurate-variant-calling-across-species-and-sequencing-protocols">NGSEP3: accurate variant calling across species and sequencing protocols</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1093/bioinformatics/btz275</p>
<p><strong>发表日期</strong>：25 April 2019</p>
<h3 id="关键点-6">关键点</h3>
<p>一个可以检测基因组各种变异的集成分析流程</p>
<h3 id="参考意义-6">参考意义</h3>
<p>从软件名字可以看出，这个工具目前已经迭代到第三个版本，最早是 2013 年发表在 NAR 。下图为整体的分析流程，其中 STR 代表 short tandem repeats</p>
<figure data-type="image" tabindex="9"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606133512.png" alt=""></figure>
<p>整个 pipeline 支持的分析如下：</p>
<ol>
<li>Alignment of reads to a reference genome with bowtie2</li>
<li>Alignments sorting by reference coordinates</li>
<li>Integrated analysis of multiple samples for efficient discovery and genotyping of SNVs, indels and STRs. This is now the recommended option for GBS, RAD-sequencingExome sequencing, RNA-seq and low coverage WGS data.</li>
<li>Complete individual sample analysis for discovery and genotyping of SNVs, indels, STRs, and CNVs from WGS data.</li>
<li>Merging of genotype calls from different samples into a single VCF file</li>
<li>Functional annotation of genomic variants</li>
<li>Filtering of VCF files using quality, coverage, and functional criteria</li>
<li>Conversion of VCF files to input formats for several downstream analysis tools such as Mega, Splitstree, Structure, PowerMarker, Flapjack or HapMap</li>
<li>Quality and coverage statistics</li>
<li>Comparison of genotype calls between VCF files</li>
<li>Genome-wide comparison of read depth patterns between two samples</li>
<li>Deconvolution for single read experiments</li>
<li>Genotype imputation</li>
<li>Allele sharing statistics for inbred populations</li>
<li>A window-based analysis to discover haplotype introgressions from population VCF files</li>
<li>Distribution of k-mer abundances from fastq or fasta files</li>
<li>Distribution of relative allele counts from BAM files</li>
<li>Calculation of IBS distance matrices from VCF files</li>
<li>Construction of neighbor joining dendograms from distance matrices</li>
<li>Simulation of single individuals from a reference genome</li>
<li>Large scale alignment of two assembled and annotated genomes</li>
<li>Construction of a haploid genome for a sequenced individual from homozygous alternative variants</li>
<li>Benchmark statistics comparing test and gold standard VCF files</li>
<li>Calculation of variant density across the genome</li>
</ol>
<h3 id="相关内容-6">相关内容</h3>
<p>软件下载地址 https://sourceforge.net/p/ngsep/wiki/Home/</p>
<hr>
<figure data-type="image" tabindex="10"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="11"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

          </div>
          
            
  
            
              <div class="mt-6">
                <link rel="stylesheet" href="https://unpkg.com/disqusjs@1.1/dist/disqusjs.css">
<script src="https://unpkg.com/disqusjs@1.1/dist/disqus.js"></script>

<div id="disqus_thread"></div>

<script>

var options = {
  shortname: 'kaopubear',
  apikey: 'S3aqfVwpSDyKQ2A8Zpg9bXlFmZue0eFVbb3JWkNZtefxqCH39Fd2cLkwP7X0MPDi',
}
if ('https://disqus.skk.moe/disqus/') {
  options.api = 'https://disqus.skk.moe/disqus/'
}
var dsqjs = new DisqusJS(options)

</script>

              </div>
            
          
        </div>
        
          <div class="w-full md:w-1/3">
            <div class="mt-6 md:pl-8">
              <h4 class="text-base font-thin text-gray-700 mb-2">发布时间</h4>
              <div class="text-lg mb-6">2019-06-06</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">标签</h4>
              <div class="mb-6">
                
                  <a href="https://fei0810.github.io/tag/NJiuPskcxN" class="py-1 px-4 border-gray-300 border text-sm inline-block text-gray-700 hover:text-gray-900 rounded mr-2 mb-2">文献</a>
                
              </div>
              <h4 class="text-base font-thin text-gray-700 mb-2">阅读时间</h4>
              <div class="text-lg mb-6">13 min read</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">相关阅读</h4>
              <div class="border-b border-gray-400">
                

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-09-11-howtolearnzotero" class="text-lg hover:underline my-2 inline-block">Zotero入门学习路径</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-06-22-weeklypaper" class="text-lg hover:underline my-2 inline-block">每周文献-190622-ATAC-seq和转录本从头组装新方法</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-06-05-usebiorxiv" class="text-lg hover:underline my-2 inline-block">随时随地跟踪最新热点文献</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  


              </div>
            </div>
          </div>
        
      </div>
      <div class="footer border-gray-400 border-t pt-6 mt-6 sm:mt-8 md:mt-10 mb-20 text-gray-700 text-sm sm:text-base lg:text-lg">
  Powered by <a href="https://github.com/getgridea/gridea" target="_blank">Gridea</a> 我的 <a href="https://t.zsxq.com/UjiEaYZ"target="_blank">知识星球</a>
</div>
    </div>
  </div>
  <script>
    Prism.highlightAll()
  </script>
</body>
</html>