<html>

<head>
  <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1583298409684">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.jsdelivr.net/npm/remixicon@2.3.0/fonts/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


</head>

<body>
  <div id="app" class="main px-4 flex flex-col lg:flex-row">
    <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="ri-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1583298409684" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E6%96%87%E7%8C%AE%E4%BF%A1%E6%81%AF">文献信息</a></li>
<li><a href="#%E6%96%87%E7%8C%AE%E6%A6%82%E8%BF%B0">文献概述</a></li>
<li><a href="#%E7%AC%94%E8%AE%B0">笔记</a>
<ul>
<li><a href="#%E5%9C%A8%E6%8B%9F%E5%8D%97%E8%8A%A5%E4%B8%AD%E5%88%86%E7%A6%BB%E5%87%BA%E4%B8%8E-dna-%E7%94%B2%E5%9F%BA%E5%8C%96%E7%BB%93%E5%90%88%E7%9A%84%E8%9B%8B%E7%99%BD%E8%B4%A8">在拟南芥中分离出与 DNA 甲基化结合的蛋白质</a></li>
<li><a href="#suvh1-%E7%94%B1-rddm-%E7%9B%B8%E5%85%B3%E7%9A%84-mchh-%E5%8B%9F%E9%9B%86">SUVH1 由 RdDM 相关的 mCHH 募集</a></li>
<li><a href="#suvh1suvh3dnaj1-%E5%92%8C-dnaj2-%E7%9B%B8%E4%BA%92%E4%BD%9C%E7%94%A8-%E5%85%B1%E5%AE%9A%E4%BD%8D%E4%B8%94%E6%98%AF%E4%B8%B4%E8%BF%91%E5%9F%BA%E5%9B%A0%E8%A1%A8%E8%BE%BE%E5%BF%85%E9%9C%80%E7%9A%84">SUVH1，SUVH3，DNAJ1 和 DNAJ2 相互作用、共定位且是临近基因表达必需的</a></li>
<li><a href="#%E6%96%B9%E6%B3%95">方法</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E6%96%87%E7%8C%AE">相关文献</a></li>
</ul>
</li>
<li><a href="#%E5%85%B6%E5%AE%83%E5%87%A0%E7%AF%87%E6%96%B9%E6%B3%95%E6%96%87%E7%AB%A0">其它几篇方法文章</a>
<ul>
<li><a href="#%E6%A4%8D%E7%89%A9-mirna-%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B">植物 miRNA 分析流程</a></li>
<li><a href="#bam-%E6%96%87%E4%BB%B6%E8%BF%87%E6%BB%A4%E4%BF%A1%E6%81%AF%E7%BB%9F%E8%AE%A1%E5%8F%8A%E6%B3%A8%E9%87%8A%E5%B7%A5%E5%85%B7">BAM 文件过滤信息统计及注释工具</a></li>
<li><a href="#cnv-%E9%A2%84%E6%B5%8B%E5%B7%A5%E5%85%B7">CNV 预测工具</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="ri-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">加入<a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球</a><br ><a href="https://tinyletter.com/kaopubear" target="_blank">邮件订阅博客内容</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
    <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
      <h1 class="text-3xl font-bold lg:mt-16">每周文献-181208-DNA 甲基化和几个工具</h1>
      <div class="text-sm text-gray-700 lg:my-8">
        2018-12-08 / 7 min read
      </div>
      
      <img class="post-feature-image rounded-lg mx-auto my-4" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-0658aa8a483a7862.png" alt="">
      
      <div class="post-content yue">
        <h2 id="文献信息">文献信息</h2>
<p><strong>标题：</strong> A DNA methylation reader complex that enhances gene transcription</p>
<!--more-->
<p><strong>DOI(url):</strong> 10.1126/science.aar7854</p>
<p><strong>发表日期：</strong> 07 Dec 2018</p>
<p><strong>关键词：</strong> DNA methylation; reader complex; SUVH1 and SUVH3; activate</p>
<h2 id="文献概述">文献概述</h2>
<p>emmm , 挺长时间没有植物甲基化的文章发表在 science 上了，文章的通讯作者是美国科学院院士。</p>
<p>DNA 甲基化（DNA methylation）是最早发现的表观修饰遗传标记。DNA 甲基化在调控基因表达、维持染色质结构、基因印记以及胚胎发育等生物学过程中发挥着重大的作用。在真核生物中，DNA 甲基化通常标记转座元件。在植物中，RNA 介导的 DNA 甲基化（RNA-directed DNA methylation, RdDM）是一种重要的甲基化途径，植物中有三种甲基化形式 CG、CHG、CHH。CG 由 DNA 甲基化转移酶 MET1 维持，在植物中 DNMT1 的同源蛋白；植物特异的 CMT3 结合 H3k9me2 促进 CHG 甲基化；CHH 由 DRM2 及 CMT2 维持。转座子的插入可以对邻近的基因在转录水平产生影响，启动子的甲基化通常来说会抑制这些基因的表达，但也存在 DNA 甲基化促进基因表达的例外情况。</p>
<p>文章鉴定了拟南芥中的蛋白质复合物 SUVH1 和 SUVH3，其通过 DNA 甲基化被募集到染色质中，SUVH 蛋白与甲基化 DNA 结合并募集 DNAJ 蛋白以增强近端基因表达。SUVH1 和 SUVH3 在体外结合甲基化 DNA，与体内常染色质甲基化有关，并与含有两个含 DNAJ 结构域的同源物 DNAJ1 和 DNAJ2 形成复合物。 DNAJ1 的异位募集增强了植物，酵母和哺乳动物的基因转录。 因此，SUVH 蛋白结合甲基化 DNA 并募集 DNAJ 蛋白以增强近端基因表达，从而抵消转座子插入基因附近的抑制作用。通过平衡抑制和激活转录效应，DNA 甲基化可以起到微调基因表达的作用。</p>
<h2 id="笔记">笔记</h2>
<h3 id="在拟南芥中分离出与-dna-甲基化结合的蛋白质">在拟南芥中分离出与 DNA 甲基化结合的蛋白质</h3>
<p>首先挑选了 10 个和不同种类甲基化位点结合紧密的蛋白。其中  SUVH1 SUVH3 以及 DNAJ 蛋白的研究较少。</p>
<h3 id="suvh1-由-rddm-相关的-mchh-募集">SUVH1 由 RdDM 相关的 mCHH 募集</h3>
<p>ChIP-seq 数据显示，SUVH1 和 SUVH3 在基因组上的定位基本相同。且这两个蛋白的位置与通过 RdDM 途径相关的 CHH 甲基化共定位。从下图中的 drm1/2 突变体可以观察到。同时 SUVH1 也富集在 NRPE1 (RNA polymerase V 的最大亚基） 的 peak 区域，且在短 TE 和长 TE 的边缘处显示出更强的富集，这些位置也是 RdDM 定位的标志。使用随机森林回归 (random forest regression) 发现 mCHH 是体内 SUVH1 结合的最强预测因子。同时在 RdDM 相关的突变体中 SUVH1 的富集基本消失。</p>
<figure data-type="image" tabindex="1"><img src="http://science.sciencemag.org/content/sci/362/6419/1182/F2.large.jpg" alt="" loading="lazy"></figure>
<h3 id="suvh1suvh3dnaj1-和-dnaj2-相互作用-共定位且是临近基因表达必需的">SUVH1，SUVH3，DNAJ1 和 DNAJ2 相互作用、共定位且是临近基因表达必需的</h3>
<p>通过 IP 实验和 ChiP-seq 发现这四个蛋白可以相互作用和共定位，而且突变体中，离他们越近的基因表达受影响越明显。表明 DNAJ1 和 DNAJ2 与 SUVH1 和 SUVH3 相互作用，被募集到 RdDM 的位点，并促进近端基因的表达。然后使用 DNAJ1 的过表达突变体发现上调的基因和 DNAJ1 的结合位点有显著的富集。</p>
<h3 id="方法">方法</h3>
<p>文章正文比较短，图也很紧凑，但是附件有很多的数据和结果，一共有 55 页。从方法部分来看，能做的高通量分析基本都做了。比如质谱，ChIP-seq, ATAC-seq，Whole Genome Bisulfite Sequencing (WGBS) 和 RNAseq。</p>
<p>不出意外的是，RNA-seq 的 mapping 还是用的 TopHat2，差异表达用的还是 DEseq 第一版，WGBS 用的是 BSMAP。因为不同重复中甲基化差距不同，在下游分析中作者将重复合并在一起进行后续分析。画热图用的包是 ComplexHeatmap。ATAC-seq mapping 用的是 bowtie , ChIP-seq 用 bowtie2, call peak 使用 MACS 而不是第二版。可视化的几个图用了 NGS.plot DeepTools 这两个包。</p>
<p>对 SUVH1 的结合位点进行预测使用了随机森林回归算法。相关方法来自于 https://www.nature.com/articles/ncomms11025 这篇文章。</p>
<h3 id="相关文献">相关文献</h3>
<p>https://www.nature.com/articles/s41556-018-0089-0<br>
https://www.nature.com/articles/nature12931<br>
https://www.ncbi.nlm.nih.gov/pubmed/27903897?dopt=Abstract<br>
https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005998<br>
https://www.nature.com/articles/nsmb.2354</p>
<h2 id="其它几篇方法文章">其它几篇方法文章</h2>
<h3 id="植物-mirna-分析流程">植物 miRNA 分析流程</h3>
<p><strong>文献题目</strong>：miRDeep-P2: accurate and fast analysis of the microRNA transcriptome in plants</p>
<p><strong>DOI(url):</strong> https://doi.org/10.1093/bioinformatics/bty972</p>
<p><strong>发表日期：</strong> 2018 Dec 6</p>
<p><strong>关键点</strong></p>
<p>Two major challenges to identify microRNAs (miRNAs) in plants</p>
<ul>
<li>how to minimize the false-positive inheritable to computational predictions;</li>
<li>how to minimize the computational time required for analyzing the miRNA transcriptome in plants with complex and large genomes.</li>
</ul>
<p><strong>miRDP2</strong>: Based on ultra-deep sampling of small RNA libraries by next generation sequencing, miRDP2 is able to identify miRNA genes in plant species, even for those without detailed annotation, with extremely high speed and reliable performance.</p>
<p>http://sourceforge.net/projects/mirdp2/.</p>
<p><strong>参考意义</strong></p>
<p>流程有待测试，作者来自北京农科院。我对这个地方的印象就是虽然在北京，但是感觉已经快到我老家了。</p>
<h3 id="bam-文件过滤信息统计及注释工具">BAM 文件过滤信息统计及注释工具</h3>
<p><strong>文献题目：</strong> Alfred: Interactive multi-sample BAM alignment statistics, feature counting and feature annotation for long- and short-read sequencing</p>
<p><strong>DOI(url):</strong> 10.1093/bioinformatics/bty1007</p>
<p><strong>发表日期：</strong> 2018 Dec 6</p>
<p><strong>关键点</strong></p>
<p>主要功能：Alfred uses subcommands for quality control (qc), feature counting (count_dna, count_rna, count_jct), feature annotation (annotate, tracks), alignment (pwalign, consensus) and haplotype-resolved analysis (split, ase).</p>
<p>网址：https://gear.embl.de/docs/alfred/</p>
<p><strong>参考意义</strong></p>
<p>看起来功能挺多，与 samtools 相比更注重一些下游的分析。</p>
<h3 id="cnv-预测工具">CNV 预测工具</h3>
<p><strong>文献题目：</strong> CODEX2: full-spectrum copy number variation detection by high-throughput DNA sequencing.</p>
<p><strong>DOI(url):</strong> 10.1186/s13059-018-1578-y</p>
<p><strong>发表日期：</strong> 2018 Nov 26</p>
<p><strong>关键点</strong></p>
<p>CODEX2, as a statistical framework for full-spectrum CNV profiling that is sensitive for variants with both common and rare population frequencies and that is applicable to study designs with and without negative control samples.</p>
<p>We demonstrate and evaluate CODEX2 on whole-exome and targeted sequencing data, where biases are the most prominent.</p>
<p>CODEX2 outperforms existing methods and, in particular, significantly improves sensitivity for common CNVs.</p>
<p>https://github.com/yuchaojiang/CODEX2</p>
<hr>
<figure data-type="image" tabindex="2"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

      </div>
      
      <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 "
        href="https://kaopubear.top/tag/EqrCpMmApu/">
        <span class="flex-auto">文献</span>
      </a>
      


      <div class="flex justify-between py-8">
        
        <div class="prev-post">
          <a href="https://kaopubear.top/post/2018-12-11-rbasic/">
            <h3 class="post-title">
              <i class="ri-arrow-left-line"></i>
              R 语言必备基础知识
            </h3>
          </a>
        </div>
        

        
        <div class="next-post">
          <a href="https://kaopubear.top/post/2018-11-30-weeklypaper/">
            <h3 class="post-title">
              每周文献-181130-lncRNA
              <i class="ri-arrow-right-line"></i>
            </h3>
          </a>
        </div>
        
      </div>

      

    </div>
  </div>

  <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()
let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
</body>

</html>