<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1581055293966">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link href="https://cdn.remixicon.com/releases/v1.3.1/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.2/animate.min.css">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<link rel="alternate" type="application/atom+xml" title="思考问题的熊 - Atom Feed" href="https://kaopubear.top//atom.xml">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div id="app" class="main px-4 flex flex-col lg:flex-row">
      <div id="sidebar" class="sidebar-wrapper lg:static lg:w-1/4">
  <div class="lg:sticky top-0">
    <div class="sidebar-content">
      <div class="flex lg:block p-4 lg:px-0 items-center fixed lg:static lg:block top-0 right-0 left-0 bg-white z-50">
        <i class="remixicon-menu-2-line lg:mt-4 text-2xl cursor-pointer animated fadeIn" onclick="openMenu()"></i>
        <a href="https://kaopubear.top/">
          <img class="animated fadeInLeft avatar rounded-lg mx-4 lg:mt-32 lg:mx-0 mt-0 lg:w-24 lg:h-24 w-12 w-12" src="https://kaopubear.top//images/avatar.png?v=1581055293966" alt="">
        </a>
        <h1 class="animated fadeInLeft lg:text-4xl font-extrabold lg:mt-8 mt-0 text-xl" style="animation-delay: 0.2s">思考问题的熊</h1>
      </div>
      
        <div class="animated fadeInLeft" style="animation-delay: 0.4s">
          <p class="my-4 text-gray-600 font-light hidden lg:block">
            文章目录
          </p>
          <div class="toc-container hidden lg:block">
            <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#%E5%8D%95%E7%BB%86%E8%83%9E%E5%8F%8A%E5%8D%95%E5%88%86%E5%AD%90%E5%9C%A8%E8%A1%A8%E8%A7%82%E7%9A%84%E5%BA%94%E7%94%A8">单细胞及单分子在表观的应用</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9">相关内容</a></li>
</ul>
</li>
<li><a href="#%E4%B8%80%E7%A7%8D%E7%BB%84%E8%A3%85%E5%A4%A7%E7%89%87%E6%AE%B5%E9%87%8D%E5%A4%8D%E5%8C%BA%E5%9F%9F%E7%9A%84%E6%96%B9%E6%B3%95">一种组装大片段重复区域的方法</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9-2">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89-2">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9-2">相关内容</a></li>
</ul>
</li>
<li><a href="#rsem-%E8%BD%AC%E5%BD%95%E6%9C%AC%E5%AE%9A%E9%87%8F%E7%9A%84%E6%8F%90%E9%80%9F%E5%8D%87%E7%BA%A7%E7%89%88">RSEM 转录本定量的提速升级版</a>
<ul>
<li><a href="#%E5%85%B3%E9%94%AE%E7%82%B9-3">关键点</a></li>
<li><a href="#%E5%8F%82%E8%80%83%E6%84%8F%E4%B9%89-3">参考意义</a></li>
<li><a href="#%E7%9B%B8%E5%85%B3%E5%86%85%E5%AE%B9-3">相关内容</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      
    </div>
  </div>
</div>

<div class="menu-container">
  <i class="remixicon-arrow-left-line text-2xl cursor-pointer animated fadeIn close-menu-btn" onclick="closeMenu()"></i>
  <div>
    
      
        <a href="/" class="menu" style="animation-delay: 0s">
          首页
        </a>
      
    
      
        <a href="/archives" class="menu" style="animation-delay: 0.2s">
          归档
        </a>
      
    
      
        <a href="/tags" class="menu" style="animation-delay: 0.4s">
          标签
        </a>
      
    
      
        <a href="/post/resource" class="menu" style="animation-delay: 0.6000000000000001s">
          资料
        </a>
      
    
      
        <a href="/post/project" class="menu" style="animation-delay: 0.8s">
          项目
        </a>
      
    
      
        <a href="/post/about" class="menu" style="animation-delay: 1s">
          关于
        </a>
      
    
  </div>
  <div class="site-footer">
    <div class="py-4 text-gray-700">欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球「靠谱熊基地」</a></div>
    <a class="rss" href="https://kaopubear.top//atom.xml" target="_blank">RSS</a>
  </div>
</div>
<div class="mask" onclick="closeMenu()">
</div>
      <div class="content-wrapper py-32 lg:p-8 lg:w-3/4 post-detail animated fadeIn">
        <h1 class="text-3xl font-bold lg:mt-16">每周文献-181223-单细胞单分子表观</h1>
        <div class="text-sm text-gray-700 lg:my-8">
          2018-12-23 / 5 min read
        </div>
        
          <img class="post-feature-image rounded-lg mx-auto my-4" src="https://raw.githubusercontent.com/fei0810/image-host/master/img/177622-4d984291a0611a37.png" alt="">
        
        <div class="post-content yue">
          <h2 id="单细胞及单分子在表观的应用">单细胞及单分子在表观的应用</h2>
<p><strong>文献题目</strong>：Single-cell and single-molecule epigenomics to uncover genome regulation at unprecedented resolution</p>
<!--more-->
<p><strong>DOI(url):</strong> https://doi.org/10.1038/s41588-018-0290-x</p>
<p><strong>发表日期：</strong> 2018 Dec 17</p>
<h3 id="关键点">关键点</h3>
<p>这篇综述介绍了单细胞和单分子表观基因组学正在以前所未有的分辨率揭示基因组调控的各种机制。作者重点介绍了变革技术中的一部分，并讨论了它们如何被用于识别新的基因调控模式</p>
<p>这些技术如下表所示</p>
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-4164002027f6df25.png" alt="epi" loading="lazy"></figure>
<h3 id="参考意义">参考意义</h3>
<p>作者将这些分析与单细胞转录组学的最新进展进行了对比，论证了表观基因组学技术在理解细胞多样性和发现基因调控机制中的重要作用。</p>
<h3 id="相关内容">相关内容</h3>
<p>另外这边文章的「Box」作者还首先阐述了进行表观研究需要测定的一些信息。如果你对表观遗传完全没有概念推荐了解。</p>
<ul>
<li>TF binding to regulatory elements.</li>
<li>Chemical modification to genomic DNA.</li>
<li>Post-translational modifications of histones.</li>
<li>Chromatin accessibility.</li>
<li>3-D genome organization</li>
</ul>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-81c2c87772af4035.png" alt="epibox" loading="lazy"></figure>
<h2 id="一种组装大片段重复区域的方法">一种组装大片段重复区域的方法</h2>
<p><strong>文献题目</strong>：Long-read sequence and assembly of segmental duplications</p>
<p><strong>DOI(url):</strong> https://doi.org/10.1038/s41592-018-0236-3</p>
<p><strong>发表日期：</strong> 2019 Jan;16</p>
<h3 id="关键点-2">关键点</h3>
<p>作者开发了一种基于长序列 reads 的多倍体 <a href="https://isogg.org/wiki/Phasing">Pasing</a> 计算方法，用来解决基因组组装中大片段重复的折叠区域问题。</p>
<p><strong>方法流程图：</strong></p>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-745de7e5ba141c51.png" alt="Flowchart of the SDA method" loading="lazy"></figure>
<p>该方法利用旁系同源序列突变（PSV）和相关性聚类来精确组装先前在 long-read  人类基因组组装中塌陷的的 SD 的不同旁系同源大片段重复序列。</p>
<p>首先利用全基因组重测序数据根据深度的差异定义大片段的重复区域，其中大于 9kb 且覆盖度高于 3 个 sd 加上均值并且不全部是由 common repeat 组成的区域被认定为 SD ；接下来定义每个基因组塌陷区域的旁系同源序列突变 paralogous sequence variants (PSVs) 并对 PSV 进行聚类。随后根据 PSV 利用 WhatsHap 将 reads 进行分组然后分别对不同组的 reads 进行组装。</p>
<h3 id="参考意义-2">参考意义</h3>
<p>在基因组异染色质和基因富集区域存在很多大片段重复 segmental duplications (SDs)，长度大于 100kb 并且存在于多个位置。即便是基于 FALCON 的单分子实时（SMRT）测序的基因组组装在很多结果中也有一半类似的区域没有解决。</p>
<p>当前 ONT (https://nanoporetech.com/)  和 PacBio (https://www.pacb.com/) 的主要问题是测序的错误率会达到 10%-15%，主要针对 WGSA 的长序列组装方法是基于 reads 的矫正和矫正后的重叠来得到更长的 contig。但是这些高错误率对于区分旁系同源和等位基因序列特别成问题，因为很多序列重复序列高度相似（&gt; 95％）并且差异部分在测序的误差范围内。这篇发表在 nature methods 文章提供了一个不错的思路。</p>
<h3 id="相关内容-2">相关内容</h3>
<p>工具地址：Segmental Duplication Assembler (SDA; https://github.com/mvollger/SDA)</p>
<h2 id="rsem-转录本定量的提速升级版">RSEM 转录本定量的提速升级版</h2>
<p><strong>文献题目</strong>：A revisit of RSEM generative model and its EM algorithm for quantifying transcript abundances.</p>
<p><strong>DOI(url):</strong> https://doi.org/10.1101/503672</p>
<p><strong>发表日期：</strong> December 21, 2018 （预印本）</p>
<h3 id="关键点-3">关键点</h3>
<p>本文作者团队修改了 RSEM 的 EM 算法，而且将计算时间减少了大约 100 倍，而且准确性还略有提高。</p>
<p>以下是一些比对测试信息</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-e7de2f17d07cccc2.png" alt="time" loading="lazy"></figure>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/177622-a11ac69f5c07782d.png" alt="cor" loading="lazy"></figure>
<h3 id="参考意义-3">参考意义</h3>
<p>RSEM 定量相对其他软件是比较准确的但是速度比较感人，近几年 Salmon, Kallisto 这类软件因为采用了 pseudoalignment 极大的提高了运算速度也取得了很大的成功。但是，这类软件不能提供一些具体的比对信息，导致很多需要这些比对信息的软件无法进行。</p>
<p>现在，有团队出来管管这事儿了，不过实际的使用感受和测试还没有进行。另外，他们作为一个商业团队还有一个开源的 RNA-seq 分析，具体信息可以参考相关内容里的链接。</p>
<h3 id="相关内容-3">相关内容</h3>
<p>hear https://github.com/bioturing/hera</p>
<p>hear-EM https://github.com/bioturing/hera/tree/master/hera-EM</p>
<hr>
<figure data-type="image" tabindex="6"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt="" loading="lazy"></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20190628083754.png" alt="扫码即刻沟通" loading="lazy"></figure>

        </div>

        
          <a class="animated fadeInUp p-2 items-center text-sm text-gray-700 border hover:bg-gray-300 leading-none rounded-full flex lg:inline-flex m-4 " href="https://kaopubear.top/tag/EqrCpMmApu">
            <span class="flex-auto">文献</span>
          </a>
        


        <div class="flex justify-between py-8">
          
            <div class="prev-post">
              <a href="https://kaopubear.top/post/2018-12-29-my2018">
                <h3 class="post-title">
                  <i class="remixicon-arrow-left-line"></i>
                  2018，思考问题的熊
                </h3>
              </a>
            </div>
          

          
            <div class="next-post">
              <a href="https://kaopubear.top/post/2018-12-19-biolitmap">
                <h3 class="post-title">
                  生物信息发文章哪家强
                  <i class="remixicon-arrow-right-line"></i>
                </h3>
              </a>
            </div>
          
        </div>

        

      </div>
    </div>

    <script src="https://kaopubear.top//media/prism.js"></script>  
<script>

Prism.highlightAll()

let mainNavLinks = document.querySelectorAll(".markdownIt-TOC a");

// This should probably be throttled.
// Especially because it triggers during smooth scrolling.
// https://lodash.com/docs/4.17.10#throttle
// You could do like...
// window.addEventListener("scroll", () => {
//    _.throttle(doThatStuff, 100);
// });
// Only not doing it here to keep this Pen dependency-free.

window.addEventListener("scroll", event => {
  let fromTop = window.scrollY;

  mainNavLinks.forEach((link, index) => {
    let section = document.getElementById(decodeURI(link.hash).substring(1));
    let nextSection = null
    if (mainNavLinks[index + 1]) {
      nextSection = document.getElementById(decodeURI(mainNavLinks[index + 1].hash).substring(1));
    }
    if (section.offsetTop <= fromTop) {
      if (nextSection) {
        if (nextSection.offsetTop > fromTop) {
          link.classList.add("current");
        } else {
          link.classList.remove("current");    
        }
      } else {
        link.classList.add("current");
      }
    } else {
      link.classList.remove("current");
    }
  });
});


document.addEventListener("DOMContentLoaded", function() {
  var lazyImages = [].slice.call(document.querySelectorAll(".post-feature-image.lazy"));

  if ("IntersectionObserver" in window) {
    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
      entries.forEach(function(entry) {
        if (entry.isIntersecting) {
          let lazyImage = entry.target
          lazyImage.style.backgroundImage = `url(${lazyImage.dataset.bg})`
          lazyImage.classList.remove("lazy")
          lazyImageObserver.unobserve(lazyImage)
        }
      });
    });

    lazyImages.forEach(function(lazyImage) {
      lazyImageObserver.observe(lazyImage)
    })
  } else {
    // Possibly fall back to a more compatible method here
  }
});

const menuContainer = document.querySelector('.menu-container')
const menus = document.querySelectorAll('.menu-container .menu')
const mask = document.querySelector('.mask')
const contentWrapper = document.querySelector('.content-wrapper')
const latestArticle = document.querySelector('.latest-article')
const readMore = document.querySelector('.read-more')
const indexPage = document.querySelector('.index-page')

const isHome = location.pathname === '/'
if (latestArticle) {
  latestArticle.style.display = isHome ? 'block' : 'none'
  readMore.style.display = isHome ? 'block' : 'none'
  indexPage.style.display = isHome ? 'none' : 'block'
}

const openMenu = () => {
  menuContainer.classList.add('open')
  menus.forEach(menu => {
    menu.classList.add('animated', 'fadeInLeft')
  })
  mask.classList.add('open')
  contentWrapper.classList.add('is-second')
}

const closeMenu = () => {
  menuContainer.classList.remove('open')
  menus.forEach(menu => {
    menu.classList.remove('animated', 'fadeInLeft')
  })
  mask.classList.remove('open')
  contentWrapper.classList.remove('is-second')
}
</script>
  
  </body>
</html>
