<html>
  <head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>素材学习分享周刊 第1期 | 思考问题的熊</title>
<meta name="description" content="很高兴和你相遇<br>这里正在记录我的所思所学" />
<link rel="shortcut icon" href="https://kaopubear.top//favicon.ico?v=1582898991718">
<link href="https://cdn.remixicon.com/releases/v1.3.1/remixicon.css" rel="stylesheet">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">

<link rel="stylesheet" href="https://kaopubear.top//styles/main.css">
<link rel="alternate" type="application/atom+xml" title="素材学习分享周刊 第1期 | 思考问题的熊 - Atom Feed" href="https://kaopubear.top//atom.xml">

<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Droid+Serif:400,700">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  </head>
  <body>
    <div class="site-nav has-cover">
      <div class="head">
  <div class="inner">
    <a class="site-name" href="https://kaopubear.top/">
        <span>思考问题的熊</span>
    </a>
    <label class="burger" for="burger"></label>
    <input id="burger" type="checkbox">
    <button class="burger">
      <div>
        <span></span>
        <span></span>
      </div>
    </button>
    <nav class="info_nav">
      
        
          <a href="/">首页</a>
        
      
        
          <a href="/archives">归档</a>
        
      
        
          <a href="/tags">标签</a>
        
      
        
          <a href="/post/resource">资料</a>
        
      
        
          <a href="/post/project">项目</a>
        
      
        
          <a href="/post/about">关于</a>
        
      
    </nav>
  </div>
</div>
    </div>
    <article role="main" class="hentry has-cover">
      
        <div class="entry-cover js-cover" data-src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220902.png" data-width="900" data-height="423" style="background-image:url(https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220902.png); height: 80vw;">
          <h1 class="post-title">素材学习分享周刊 第1期</h1>
        </div>
      
      <h1 class="entry-title" itemprop="headline">素材学习分享周刊 第1期</h1>
      <div class="entry-meta">
        <time class="updated" datetime="2020-02-28 22:04:29">2020-02-28</time>
        <span class="author vcard">
          15 min read
        </span>
      </div>
      <div class="post-content yue">
        <h2 id="刊首语">刊首语</h2>
<p>在「素材学习分享小组」进行的第二个年头，为了让它能再向前走一小步，我们决定以电子杂志的形式发布「素材学习分享周刊」。</p>
<!-- more -->
<p>每期都会有一位小组成员作为轮值编辑为大家奉献一期内容，同时也作为小组成员的阶段性考核。**我们会提前一周提醒轮值编辑，如果断更则基本会取消下一阶段小组成员资格。**每位轮值编辑对自己当期的内容持有版权，可以决定除小组分享外是否公开分享，亦可发布在自己的博客或者公众号等个人平台。</p>
<p>当然，如果不是轮值编辑仍然可以记录自己的分享内容，只需要在题目最后增加你的个人 ID 即可。除了清退机制，我们也在思考如何引入奖励机制或成就值。分享的内容越多，在小组的成就值也就越高。</p>
<p>周刊的分享和发布形式都受到了阮一峰「<a href="https://www.yuque.com/ruanyf/weekly/">科技爱好者周刊</a>」的启发，在此表示感谢和敬意。</p>
<p>在以后的每期内容中，轮值编辑都会向大家推荐自己的心头好，其中包括但不限于和专业相关的科研文献、近期读到的书籍文章、值得学习的资料素材、自己喜欢的电影音乐和好用的应用工具等等。</p>
<p><strong>希望这些内容可以记录彼此的成长，留住易逝的时光。</strong></p>
<p>思考问题的熊<br>
2020年2月28日</p>
<h2 id="专业文献">专业文献</h2>
<h3 id="regulation-of-rice-tillering-by-rna-directed-dna-methylation-at-miniature-inverted-repeat-transposable-elements">Regulation of Rice Tillering by RNA-directed DNA Methylation at Miniature Inverted-repeat Transposable Elements</h3>
<p><em>Molecular Plant 2020 February 19</em></p>
<ul>
<li>PMID: 32087371</li>
<li>DOI: <a href="https://doi.org/10.1016/j.molp.2020.02.009">10.1016/j.molp.2020.02.009</a></li>
</ul>
<p>Tillering is a major determinant of rice plant architecture and grain yield. Here we report that depletion of rice OsNRPD1a and OsNRPD1b, two orthologues of the largest subunit of RNA polymerase IV, leads to a high-tillering phenotype, in addition to dwarfism and smaller panicles. OsNRPD1a and OsNRPD1b are required for the production of **24-nt siRNAs **that direct DNA methylation at transposable elements (TEs) including Miniature Inverted-repeat TEs (MITEs). Interestingly, many genes are regulated either positively or negatively by TE methylation. Among them, OsMIR156d and OsMIR156j, which promote rice tillering, are repressed by CHH methylation at two MITEs in the promoters. By contrast, D14, which suppresses rice tillering, is activated by CHH methylation at a MITE in its downstream. Our findings reveal control of rice tillering by RdDM at MITEs and provide potential targets for agronomic trait enhancement by epigenome editing.</p>
<blockquote>
<p>水稻分蘖是水稻结构和产量的主要决定因素。这篇文章揭示了RNA介导的DNA甲基化（RdDM）在MITE这个类型的转座子上可以分别正向和负向调节 D14 和 OsMIR156 家族成员，从而促进水稻分蘖的调节。</p>
</blockquote>
<h3 id="robustness-and-applicability-of-transcription-factor-and-pathway-analysis-tools-on-single-cell-rna-seq-data">Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data</h3>
<p><em>Genome Biology 2020 February 12, 21 (1): 36</em></p>
<ul>
<li>PMID: <strong>32051003</strong></li>
<li>DOI: <a href="https://doi.org/10.1186/s13059-020-1949-z">10.1186/s13059-020-1949-z</a></li>
</ul>
<p><strong>BACKGROUND:</strong> Many functional analysis tools have been developed to extract functional and mechanistic insight from bulk transcriptome data. With the advent of single-cell RNA sequencing (scRNA-seq), it is in principle possible to do such an analysis for single cells. However, scRNA-seq data has characteristics such as drop-out events and low library sizes. It is thus not clear if functional TF and pathway analysis tools established for bulk sequencing can be applied to scRNA-seq in a meaningful way.<br>
<strong>RESULTS:</strong> To address this question, we perform benchmark studies on simulated and real scRNA-seq data. We include the bulk-RNA tools PROGENy, GO enrichment, and DoRothEA that estimate pathway and transcription factor (TF) activities, respectively, and compare them against the tools SCENIC/AUCell and metaVIPER, designed for scRNA-seq. For the in silico study, we simulate single cells from TF/pathway perturbation bulk RNA-seq experiments. We complement the simulated data with real scRNA-seq data upon CRISPR-mediated knock-out. Our benchmarks on simulated and real data reveal comparable performance to the original bulk data. Additionally, we show that the TF and pathway activities preserve cell type-specific variability by analyzing a mixture sample sequenced with 13 scRNA-seq protocols. We also provide the benchmark data for further use by the community.**</p>
<blockquote>
<p>转录因子和通路分析工具对于单细胞数据是不是数据，以及有哪些影响，应该怎么用，这篇文章给出了一些介绍。这篇文章也提到了一些我们这方面可以借鉴的工具。</p>
</blockquote>
<h3 id="mumco-accurate-detection-of-all-sv-types-through-whole-genome-alignment">MUM&amp;Co: Accurate detection of all SV types through whole genome alignment</h3>
<p><em>Bioinformatics 2020 February 25</em></p>
<ul>
<li>PMID: <strong>32096823</strong></li>
<li>DOI: <a href="https://doi.org/10.1093/bioinformatics/btaa115">10.1093/bioinformatics/btaa115</a></li>
</ul>
<p><strong>SUMMARY:</strong> MUM&amp;Co is a single bash script to detect Structural Variations (SVs) utilizing Whole Genome Alignment (WGA). Using MUMmer's nucmer alignment, MUM&amp;Co can detect insertions, deletions, tandem duplications, inversions and translocations greater than 50bp. Its versatility depends upon the WGA and therefore benefits from contiguous de-novo assemblies generated by 3rd generation sequencing technologies. Benchmarked against 5 WGA SV-calling tools, MUM&amp;Co outperforms all tools on simulated SVs in yeast, plant and human genomes and performs similarly in two real human datasets. Additionally, MUM&amp;Co is particularly unique in its ability to find inversions in both simulated and real datasets. Lastly, MUM&amp;Co's primary output is an intuitive tabulated file containing a list of SVs with only necessary genomic details.<br>
<strong>AVAILABILITY:</strong> <a href="https://github.com/SAMtoBAM/MUMandCo.">https://github.com/SAMtoBAM/MUMandCo</a></p>
<blockquote>
<p>借助 MUMmer 写的一个一千多行的shell脚本，用来分析各种SV，测试了不同物种的分析效果，发到了bioinformatics。</p>
</blockquote>
<h3 id="small-rnas-in-the-transgenerational-inheritance-of-epigenetic-information">Small RNAs in the Transgenerational Inheritance of Epigenetic Information</h3>
<p>Published:January 14, 2020</p>
<ul>
<li>DOI:https://doi.org/10.1016/j.tig.2019.12.001</li>
</ul>
<p>In recent years it has become evident that RNA interference-related mechanisms can mediate the deposition and transgenerational inheritance of specific chromatin modifications in a truly epigenetic fashion. Rapid progress has been made in identifying the RNAi effector proteins and how they work together to confer long-lasting epigenetic responses, and initial studies hint at potential physiological relevance of such regulation. In this review, we highlight mechanistic studies in model organisms that advance our understanding of how small RNAs trigger long-lasting epigenetic changes in gene expression and we discuss observations that lend support for the idea that small RNAs might participate in mechanisms that trigger epigenetic gene expression changes in response to environmental cues and the effects these could have on population adaptation.</p>
<blockquote>
<p>这篇综述介绍了smRNA和表观的关系，推荐大家了解一下。</p>
</blockquote>
<h2 id="数据库">数据库</h2>
<p>_Genome Research _最近发表了一篇 Resource 文章，FANTOM 利用他们已有的数据对人的lncRNA数据进行了一次新的综合定量分析。<br>
长链非编码RNA (lncRNAs) 已经成为生物和细胞过程的关键调控子，在细胞和组织中表征 lncRNA 的表达是理解它们作用的关键。研究者们推出的 FC-R2 是一个转录组相关的综合表达图谱，包括了超过 109,000 个编码和非编码基因。<br>
该图谱极大地扩展了已有的基因注释。我们通过从已发表的大型研究中复制关键发现，并通过在正常和患病的人类样本中产生新的结果，证明了 FC-R2 图谱的实用性。<br>
该图谱有如下几个特点：</p>
<ul>
<li>确定不同类型编码和非编码基因的组织特异性转录谱</li>
<li>在 13 种癌症类型中进行差异表达分析, 识别潜在参与肿瘤发病和进展的新的非编码基因</li>
<li>确认癌症中几种增强子 lncRNAs 表达的预后价值</li>
</ul>
<p><strong>相关资料</strong></p>
<ul>
<li><a href="https://fantom.gsc.riken.jp/6/">FANTOM6</a> Functional annotation of long non-coding RNAs in the human genome</li>
<li><a href="https://jhubiostatistics.shinyapps.io/recount/">recount</a> A multi-experiment resource of analysis-ready RNA-seq gene and exon count datasets</li>
<li><a href="https://doi.org/10.1101/gr.254656.119">Recounting the FANTOM CAGEAssociated Transcriptome</a></li>
</ul>
<h2 id="好书好文">好书好文</h2>
<h3 id="如何学会所有编程语言">如何学会所有编程语言</h3>
<p>应该从哪些方面学习一门编程语言<br>
英文原文：<a href="https://hackernoon.com/how-to-learn-all-programming-languages-yes-all-c41j3t8l">This is How You Can Learn All Programming Languages, Yes - “all”</a><br>
中文译文：<a href="https://www.infoq.cn/article/8STbPDW9M9xVGVc2eVIP">如何学会所有的编程语言？</a></p>
<h3 id="一个普通985研究生的求职全纪录">一个普通985研究生的求职全纪录</h3>
<p>我也不是很差劲吧，985研究生，成绩前5%，发过论文，拿过国奖，怎么找工作就比不上别人？<br>
<a href="https://mp.weixin.qq.com/s/ITOV7cHB-KkEYceWYAw_RA">https://mp.weixin.qq.com/s/ITOV7cHB-KkEYceWYAw_RA</a></p>
<h3 id="梁文道-周轶君为什么假新闻总是跑得更快">梁文道、周轶君：为什么“假新闻”总是跑得更快？</h3>
<p>弹指之间就能获得大量讯息的今天，自然也伴随着漫天飞的截图、谣言、推测、和评论。这时，真相真的离我们更近了吗？我们该如何获得真相？新闻、媒体、平台与我们的关系又有了哪些改变？<br>
<a href="https://mp.weixin.qq.com/s/h-rkTyJ12Kn4olVYhegUSQ">https://mp.weixin.qq.com/s/h-rkTyJ12Kn4olVYhegUSQ</a></p>
<h2 id="学习素材">学习素材</h2>
<h3 id="如何使用r制作ppt">如何使用R制作PPT</h3>
<blockquote>
<p>前天收到了教育中心负责老师的微信，她说我之前作为助教带的一些学生这两年也陆续在当助教，这一届的助教有人推荐了我当时上第一节课的时候使用的PPT，问我能不能贡献出来给他们参考一下。这就想起了当时用R做PPT的经历。这几天又看到了一个不错的介绍</p>
</blockquote>
<p>地址 <a href="https://arm.rbind.io/slides/xaringan.html#17">https://arm.rbind.io/slides/xaringan.html</a></p>
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220029.png" alt="image.png" loading="lazy"></figure>
<h2 id="影音推荐">影音推荐</h2>
<p>北京时间2月25日凌晨，科比-布莱恩特追思会“生命的礼赞”在斯台普斯中心举行。科比遗孀瓦妮莎首次公开亮相，发表了感人至深的演讲，纪念科比以及两人的女儿吉安娜。迈克尔-乔丹全程流泪演讲，沙奎尔-奥尼尔等也都演讲致敬、怀念科比。<br>
这次推荐两个视频，一个是乔丹此次的演讲，一个是Jimmy Kimmel特别节目追忆科比，科比曾经15次录制过这个节目。</p>
<h3 id="乔丹回忆科比">乔丹回忆科比</h3>
<iframe src="https://player.bilibili.com/player.html?aid=91519948" frameborder="no" allowfullscreen="true"></iframe>
<h3 id="jimmy-kimmel特别节目追忆科比">Jimmy Kimmel特别节目追忆科比</h3>
<iframe src="https://player.bilibili.com/player.html?aid=85431520" frameborder="no" allowfullscreen="true"></iframe>
<h2 id="工具">工具</h2>
<h3 id="文章配图插画制作-itg-digital">文章配图插画制作 itg digital</h3>
<p>关于文章配图，推荐过几个比较不错的插画网站，这次推荐的工具支持对不同的元素进行自定义的组合。可以根据自己的需求更改元素、样式和颜色。jpg 格式内容的下载免费。</p>
<p>地址 <a href="https://app.itg.digital/">https://app.itg.digital/</a></p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220044.png" alt="image.png" loading="lazy"></figure>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220051.png" alt="image.png" loading="lazy"></figure>
<p>给我印象很深的是他的加载动画，是一只小狗向前走，然后越走自己的身体越长，如果你的网速不理想就可以看到一只加长小狗～</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/20200228220056.png" alt="image.png" loading="lazy"></figure>
<h2 id="讨论">讨论</h2>
<p>本周我遇到了两个比较有趣的问题，一并记录下来和大家分享。</p>
<h3 id="富集分析用哪些基因">富集分析用哪些基因</h3>
<p>第一个问题是</p>
<blockquote>
<p>转录组分析中，常常把上调表达基因和下调表达基因分别进行GO分析和Pathway富集分析，为什么不是放在一起进行GO和Pathway分析呢？上调下调的不同基因可能是同一种功能或者同一个通路的，整体分析不是能更全面的找出真正差异的功能或者通路了吗</p>
</blockquote>
<p>我的回答：</p>
<blockquote>
<p>这个问题很有意思，首先要不要把差异基因上调和下调分别进行GO和Pathway分析本身是没有定论的，上调下调不同基因当然可能是同一种功能或者同一个通路的。如果KEGG的一个通路里本身有抑制性的基因也有促进性的基因，那么使用所有表达差异基因去分析就是合理的。</p>
<p>但是2013年曾经有一篇文章专门研究了这个问题，他们使用了5种肿瘤的芯片和转录组数据集进行了分析，研究的结果是具有相关功能的基因在表达水平上往往是正相关的，这也就导致特定通路上某一类基因的比例是失衡的，不恰当的具体例子可以理解为一个通路上100个基因，可能有80个都是在一种处理中上调表达，而只有20个是下调表达的。</p>
<p>Hong G, Zhang W, Li H, Shen X, Guo Z. Separate enrichment analysis of pathways for up- and downregulated genes. J R Soc Interface. 2013;11(92):20130950. Published 2013 Dec 18. doi:10.1098/rsif.2013.0950</p>
<p>这种不平衡导致如果使用全部差异基因进行富集分析的统计学检验时会显著降低统计检验效力，在他们检测的5种肿瘤数据中，对上调和下调的基因进行单独分析可以确定与表型差异真正相关的更多通路。</p>
<p>这里就出现了一个富集分析的缺点，所谓上调基因和下调基因是不同处理下的相对表达量而言的，但是仅凭KEGG里的「富集」就把上调理解为对通路的激活，把下调理解为对通路的抑制是不合适的。</p>
<p>此外，富集分析通常用的都是Fisher精确检验或超几何检验来验证相关基因在背景中的比例是不是足够多，这里怎么定义上下调的倍数本身也存在很多争议，也就是为什么要引入某种程度上更有意义的GSEA分析。</p>
<p>当然，之前也有过一些其它的分析思路，但是都是发了文章没有人用。</p>
<p>Warden C D, Kanaya N, Chen S, et al. BD-Func: a streamlined algorithm for predicting activation and inhibition of pathways[J]. PeerJ, 2013, 1: e159.</p>
<p>Malathi S.I Dona, Luke A Prendergast, Suresh Mathivanan, Shivakumar Keerthikumar, Agus Salim, Powerful differential expression analysis incorporating network topology for next-generation sequencing data, Bioinformatics, Volume 33, Issue 10, 15 May 2017, Pages 1505–1513, <a href="https://doi.org/10.1093/bioinformatics/btw833">https://doi.org/10.1093/bioinformatics/btw833</a></p>
</blockquote>
<h3 id="rna和蛋白质的关系">RNA和蛋白质的关系</h3>
<blockquote>
<p>有没可能mrna没有高表达，但是蛋白高表达。即使发生了也没法解释吧？</p>
</blockquote>
<p>我的回答</p>
<blockquote>
<p>mRNA 和对应蛋白质的关系比较复杂，比如转录后修饰和miRNA起作用等等，这个推荐一篇研究二者关系的文献，16年发表在cell上</p>
<p>Liu Y, Beyer A, Aebersold R. On the Dependency of Cellular Protein Levels on mRNA Abundance. Cell. 2016;165(3):535–550. doi:10.1016/j.cell.2016.03.014</p>
</blockquote>
<h2 id="订阅">订阅</h2>
<p>该周刊每周六发布，暂时作为「素材分享学习小组」内部资料，我们会挑选部分内容进行公开分享。<br>
完成前期测试运行后将择期通过博客和公众号等形式同步更新，届时将发布具体的订阅形式。</p>
<h2 id="交流">交流</h2>
<p>如果你想要加入「素材分享学习小组」可以参考<a href="https://kaopubear.top/post/2020-01-19-refgroup2020/">申请说明，</a>如果想和我们交流，欢迎在评论区留言，我们下周见！</p>

      </div>
      <div class="entry-block">
        <div class="entry-tags">
          
            <a href="https://kaopubear.top/tag/tg-VHYKad/">
              工具
            </a>
          
            <a href="https://kaopubear.top/tag/EqrCpMmApu/">
              文献
            </a>
          
            <a href="https://kaopubear.top/tag/2IWlJgErkt/">
              日常
            </a>
          
        </div>
      </div>
    </article>
    
    
      <section class="post-section prev-post">
        <div class="inner">
          <h3>后一篇</h3>
          <a href="https://kaopubear.top/post/2020-01-06-7mooc/">
            <strong>停课不停学 推荐 7 门有趣实用的线上课程</strong>
          </a>
        </div>
      </section>
    

    

    <footer class="footer">
  <div class="main">
    <a href="https://kaopubear.top/">
      <img class="avatar" src="https://kaopubear.top//images/avatar.png?v=1582898991718" alt="思考问题的熊">
    </a>
    <div class="footer__social">
      
        
          <a href="https://github.com/fei0810" target="_blank"><i class="remixicon-github-line"></i></a>
        
      
        
      
        
      
        
          <a href="https://www.zhihu.com/people/ibear_share/posts" target="_blank"><i class="remixicon-zhihu-line"></i></a>
        
      
        
      
        
          <a href="https://t.me/kaopubear" target="_blank"><i class="remixicon-telegram-line"></i></a>
        
      
        
      
        
      
        
      
        
      
    </div>
    <p class="footer__sosumi">
      欢迎加入我的 <a href="https://t.zsxq.com/UjiEaYZ" target="_blank">知识星球「靠谱熊基地」</a>
    </p>
    <a class="footer-rss" href="https://kaopubear.top//atom.xml">RSS</a>
  </div>
</footer>

<script src="https://kaopubear.top//media/prism.js"></script>
<script>
  Prism.highlightAll()
</script>

  </body>
</html>
