<html lang="en">
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://fei0810.github.io/styles/main.css">
<script src="https://fei0810.github.io/media/script.js"></script>

    

<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title></title>
<link rel="stylesheet" href="https://fei0810.github.io/styles/main.css">
<script src="https://fei0810.github.io/media/script.js"></script>

    

  <script src="https://fei0810.github.io/media/prism.js"></script>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.0/katex.min.css">
</head>
<body>
  <div class="antialiased">
    <div class="max-w-5xl mx-auto px-6 md:px-8">
      <div class="flex justify-between items-center mt-10">
  <h1 class="text-lg sm:text-xl md:text-2xl lg:text-3xl text-gray-900 font-medium md:font-normal leading-none">
    <a href="https://fei0810.github.io">思考问题的熊</a>
  </h1>
  <div class="block lg:hidden" id="thumb-menu">
    <button id="thumb-open" type="button" class="block">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M2 6H22V7H2V6ZM2 12H22V13H2V12ZM22 18H2V19H22V18Z" class="fill-current text-gray-900"></path></svg>
    </button>
    <button id="thumb-close" type="button" class="hidden">
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M4 4L20 20M20 4L4 20" class="" stroke="#1A202C"></path></svg>
    </button>
  </div>
</div>
<div id="nav-list" class="lg:block hidden">
  <div class="w-full block lg:flex lg:justify-between lg:border-b lg:border-gray-400 mt-10 text-gray-700 font-medium lg:text-lg">
    <div class="lg:flex -mb-px">
      
        
          <a href="/" class="nav-link">首页</a>
        
      
        
          <a href="/archives" class="nav-link">归档</a>
        
      
        
          <a href="/tags" class="nav-link">标签</a>
        
      
        
          <a href="/post/about" class="nav-link">关于</a>
        
      
        
          <a href="/post/resource" class="nav-link">资料</a>
        
      
    </div>
    <div class="lg:flex">
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
      <a href="https://fei0810.github.io/atom.xml" class="block mt-6 lg:mt-0 lg:ml-8 flex align-center hover:text-gray-900" target="_blank">
        <div>RSS</div>
        <svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg" class="ml-1 -mt-px"><path d="M16.5 13V18.5H6.5V8.5H12" stroke="#4A5568"></path> <path d="M10 15L18.5 6.5" stroke="#4A5568"></path> <path d="M14 6.5H18.5V11" stroke="#4A5568"></path></svg>
      </a>
    </div>
  </div>
</div>
      <div class="mb-20 md:flex mt-10 lg:mt-6 md:border-t md:border-gray-400 md:pt-2 lg:border-t-0 lg:pt-0">
        <div class="w-full md:w-2/3 mt-6">
          <h1 class="text-4xl">每周文献-190411-lncRNA功能exRNA分析以及DNA甲基化</h1>
          
          <div class="mt-6 lg:mt-10 post-content">
            <h2 id="antisense-lncrna-transcription-mediates-dna-demethylation-to-drive-stochastic-protocadherin-α-promoter-choice">Antisense lncRNA Transcription Mediates DNA Demethylation to Drive Stochastic Protocadherin α Promoter Choice</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1016/j.cell.2019.03.008</p>
<p><strong>发表日期</strong>：4 April 2019</p>
<h3 id="关键点">关键点</h3>
<p>反义 lncRNA 转录可以影响 DNA 甲基化，进而改变染色体结构促进增强子与启动子结合，调控基因表达。</p>
<figure data-type="image" tabindex="1"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002036.png" alt=""></figure>
<!--more-->
<h3 id="参考意义">参考意义</h3>
<figure data-type="image" tabindex="2"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002114.png" alt=""></figure>
<p>Pcdhα基因有 13 个可以随机启动的可变外显子，每个启动子都可以由自身的启动子驱动，还有 3 个 c 型外显子以及稳定表达的编码 pcdh 结构域的外显子，同时，还有一个 enhancer 调控。</p>
<p>研究者发现，反义 lncRNA 的转录，会造成该位点 DNA 去甲基化的发生，从而使远端增强子靠近该外显子的启动子，促进它的表达。如图 2 所示，Pcdhα基因座位均带有抑制表达的 DNA 甲基化修饰，而当反义 lncRNA 表达时，该外显子附近的 DNA 被 DNA 去甲基化酶 TET3 识别，去除了甲基化修饰，cohesin 蛋白重塑了染色体的结构，HS5-1 增强子与该基因座位的启动子结合，启动相应 Pcdhα变体的表达。</p>
<h3 id="相关内容">相关内容</h3>
<p>关于反义 lncRNA 影响正义链基因表达的作用机制，主要有 3 类</p>
<ol>
<li>反义 lncRNA 的转录过程，抑制正义链基因的转录，该机制认为反义链转录事件本身，而不是反义 lncRNA，调控了基因的表达。</li>
<li>反义 lncRNA 结合 DNA 或组蛋白修饰酶，调控所在基因座位的表观遗传学，从而影响正义链基因的表达。</li>
<li>反义 lncRNA 与正义链 mRNA 通过碱基互补配对结合，影响 mRNA 的可变剪接等。</li>
</ol>
<p>CTCF:</p>
<blockquote>
<p>CTCF is an enhancer-blocking protein that inhibits the access of Igf2 to the enhancer elements located downstream from the H19 transcription start site.</p>
</blockquote>
<p>Cohesin</p>
<blockquote>
<p>Cohesin is a multiprotein complex that holds sister chromatids together from S phase until the start of mitosis, helping to ensure genomic integrity (Haarhuis, Elbatsh, &amp; Rowland, 2014).</p>
</blockquote>
<p>另一篇植物中相关 lncRNA 鉴定方法</p>
<blockquote>
<p>First, only transcripts with TAIR10 annotation [Cufflinks class codes ‘u’ (intergenic transcripts),’x’ (Exonic overlap with reference on the opposite strand),’i’ (transcripts entirely within intron) were retained. Second, transcripts of short length (length &lt;150 nt) or low abundance (FPKMmax &lt; 1, FPKMmax stands for the maximum expression level of a lncRNA from all samples) were removed. Third, transcripts with protein-coding potential were removed. Protein-coding potential was determined by using two programs: (1) transcripts were subjected to a BlastX search against all plant protein sequences in the Swiss-Prot database70 with a cutoff e-value &lt; 10-4 and the transcripts with strong hits (alignment length ≥40 aa, percent identity ≥35% and coverage of the alignment region in either query or subject sequence ≥35%) to known proteins were considered to have protein-coding potential; For antisense transcripts, open reading frames were checked. (2) the CPC (Coding Potential Calculator) score71, a value to assess protein-coding potential of a transcript based on six biologically meaningful sequence features, was calculated for each transcript. When the CPC score is positive, we considered the transcript to have protein-coding potential. Transcripts that passed the three filtering steps were annotated as lncRNAs.</p>
</blockquote>
<p><a href="https://www.nature.com/articles/s41467-018-07500-7">Global identification of Arabidopsis lncRNAs reveals the regulation of MAF4 by a natural antisense RNA</a></p>
<h2 id="excerpt-a-comprehensive-analytic-platform-for-extracellular-rna-profiling">exceRpt: A Comprehensive Analytic Platform for Extracellular RNA Profiling</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1016/j.cels.2019.03.004</p>
<p><strong>发表日期</strong>：April 4, 2019</p>
<h3 id="关键点-2">关键点</h3>
<p>一套分析 exRNA 的完整方案</p>
<h3 id="参考意义-2">参考意义</h3>
<p>exRNA 就是细胞外 RNA，也就是在细胞内转录但是在细胞外发挥功能？最近 cell 有一个专刊介绍了很多 exRNA 的文章，我扫了一眼，感觉主要内容是小 RNA 居多。所以这个分类和 lncRNA 类似，一个从长度一个从位置来进行区分。既然是这样，exRNA 的处理方法应该就和小 RNA 以及一般的 RNAseq 分析类似。看看这个流程里面有哪些不一样的地方。这里提供的分析方法从内容来看主要针对小 RNA，但是官方说也可以很方便的移植到其它 RNA，在 GitHub 的代码里也有针对 longRNA 的脚本。主要流程如下图，质控后会同时比对到多个数据库，具体内容可以参考 GitHub。</p>
<figure data-type="image" tabindex="3"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002139.png" alt=""></figure>
<h3 id="相关内容-2">相关内容</h3>
<p><a href="http://github.gersteinlab.org/exceRpt/">exceRpt</a></p>
<p>什么是 exRNA</p>
<blockquote>
<p>exRNA: Extracellular RNA (also known as exRNA or exosomal RNA) describes RNA species present outside of the cells from which they were transcribed. In Homo sapiens, exRNAs have been discovered in bodily fluids such as venous blood, saliva, breast milk, urine, semen, menstrual blood, and vaginal fluid. Although their biological function is not fully understood, exRNAs have been proposed to play a role in a variety of biological processes including syntrophy, intercellular communication, and cell regulation.</p>
</blockquote>
<p>exRNA 的种类</p>
<blockquote>
<p>Extracellular RNA should not be viewed as a category describing a set of RNAs with a specific biological function or belonging to a particular RNA family. Similar to the term &quot;non-coding RNA&quot;, &quot;extracellular RNA&quot; defines a group of several types of RNAs whose functions are diverse, yet they share a common attribute which, in the case of exRNAs, is existence in an extracellular environment. The following types of RNA have been found outside the cell:</p>
</blockquote>
<ul>
<li>Messenger RNA (mRNA)</li>
<li>Transfer RNA (tRNA)</li>
<li>MicroRNA (miRNA)</li>
<li>Small interfering RNA (siRNA)</li>
<li>Long non-coding RNA (lncRNA)</li>
</ul>
<p>研究 exRNA 的关键似乎应该是如何分离得到确实是细胞外的 RNA。</p>
<p>比如 Small RNA Sequencing across Diverse Biofluids Identifies Optimal Methods for exRNA Isolation. Cell. 2019 Apr 4;177(2):446-462.e16. doi: 10.1016/j.cell.2019.03.024. 这篇文章就比较了集中 exRNA Isolation Methods，通过对 5 种生物液体中 10 种 exRNA 分离方法的系统比较，发现所得到的小 RNA-seq 图谱的复杂性和重现性存在显著差异。每种方法对不同的 exRNA 载体亚类的相对效率是通过估计细胞外囊泡 (EV)-、核糖核蛋白(RNP)- 和高密度脂蛋白(HDL) 特异性 miRNA 在每个图谱中的比例来确定的。开发了一种基于 web 的交互式应用(miRDaR)，帮助研究人员为他们的研究选择最佳的 exRNA 分离方法。</p>
<p>另外，还有一篇文章：exRNA Atlas Analysis Reveals Distinct Extracellular RNACargo Types and Their Carriers Present across Human Biofluids. Cell. 2019 Apr 4;177(2):463-477.e15. doi: 10.1016/j.cell.2019.02.018.</p>
<p><a href="https://exrna-atlas.org">exRNA Atlas resource</a> 包含来自 19 项研究的 5309 个 exRNA-seq 和 exRNAqPCR 概要文件，以及一套分析和可视化工具。通过分析，该研究得到了一个包含六种 exRNA 类型 (CT1、CT2、CT3A、CT3B、CT3C、CT4) 的模型，每种 exRNA 类型都可以在多种生物体液 (血清、血浆、脑脊液、唾液、尿液) 中检测到。</p>
<h2 id="a-statistical-normalization-method-and-differential-expression-analysis-for-rna-seq-data-between-different-species">A statistical normalization method and differential expression analysis for RNA-seq data between different species</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1186/s12859-019-2745-1</p>
<p><strong>发表日期</strong>：29 March 2019</p>
<h3 id="关键点-3">关键点</h3>
<p>不同物种之间 RNA-seq 怎么分析确实是一个问题，那么 ChIP-seq 呢？</p>
<h3 id="参考意义-3">参考意义</h3>
<blockquote>
<p>propose a <strong>scale based normalization</strong> (SCBN) method by taking into account the available knowledge of conserved orthologous genes and by using the hypothesis testing framework.</p>
<p>Considering the different gene lengths and unmapped genes between different species, we formulate the problem from the perspective of hypothesis testing and search for the optimal scaling factor that minimizes the deviation between the empirical and nominal type I errors.</p>
</blockquote>
<p>用小鼠来研究人的疾病非常常见，在一些文章中也有人会用同源基因进行比较。对于不同物种的数据来说，除了基因数量基因长度的不同，也有测序深度的问题。让两个物种之间的数据可比，数据的标准化方法非常重要。在之前的一些研究中，有人使用 RPKM 值来进行比较找到一千个保守基因，然后评估每个基因在不同物种中的中位数水平，然后通过让中位值保持一致来得到一个校正因子（median method）。这篇文章作者利用直系同源基因通过对已有方法的改进来进行不同物种之间数据的矫正。</p>
<figure data-type="image" tabindex="4"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002150.png" alt=""></figure>
<h3 id="相关内容-3">相关内容</h3>
<p><a href="http://www.bioconductor.org/packages/devel/bioc/html/SCBN.html">SCBN</a></p>
<p>ChIP-seq 类的数据有哪些方法呢？</p>
<p><strong>雅卡尔指数</strong>（英语：Jaccard index），又称为并交比（Intersection over Union）、雅卡尔相似系数（Jaccard similarity coefficient），是用于比较样本集的相似性与多样性的统计量。雅卡尔系数能够量度有限样本集合的相似度，其定义为两个集合交集大小与并集大小之间的比例。在 bedtools 中有这个工具可以 <a href="https://bedtools.readthedocs.io/en/latest/content/tools/jaccard.html">对 jaccard 进行计算</a>。</p>
<p>还有 <strong>余弦相似度 cosine similarity score</strong> ，比如这篇文章 A Cosine Similarity-Based Method to Infer Variability of Chromatin Accessibility at the Single-Cell Level</p>
<p>另外，<a href="http://www.biostat.jhsph.edu/dpca/">dpca</a> 可以用于分析转录因子结合位点处和启动子的不同染色质模式，以及等位基因特异性蛋白 - DNA 之间的相互作用。</p>
<h2 id="dna-methylation-analysis-in-plants-review-of-computational-tools-and-future-perspectives">DNA methylation analysis in plants: review of computational tools and future perspectives</h2>
<p><strong>DOI(url)</strong>: https://doi.org/10.1093/bib/bbz039</p>
<p><strong>发表日期</strong>：09 April 2019</p>
<h3 id="关键点-4">关键点</h3>
<p>难得的植物 DNA 甲基化分析综述文章</p>
<h3 id="参考意义-4">参考意义</h3>
<p>在这篇综述中，作者概述了分析 DNA 甲基化数据（特别是亚硫酸氢盐测序数据）最常用的生物信息学工具，也分析了这些工具的性能并且比较了计算拟南芥以及小麦甲基数据的计算时间和一致性。同时举例说明了作物中 DNA 甲基化数据分析的应用。但从软件上看，BSMap 用是最短，尤其是当线程数上去之后，但是内存则是 Bismark 最省。</p>
<figure data-type="image" tabindex="5"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002159.png" alt=""></figure>
<p>关于内存的使用情况，不要被下图迷惑。小麦那里只是展示了处理 1 条染色体的需要的内存用量。要知道，小麦可是有 21 条染色体，16G 的基因组。</p>
<figure data-type="image" tabindex="6"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190606002207.png" alt=""></figure>
<h3 id="相关内容-4">相关内容</h3>
<p>另外一篇文章，<a href="https://doi.org/10.1016/j.jbiotec.2017.08.007">Strategies for analyzing bisulfite sequencing data</a> 。</p>
<hr>
<figure data-type="image" tabindex="7"><img src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" alt=""></figure>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名 - 非商业性使用 - 禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<figure data-type="image" tabindex="8"><img src="https://raw.githubusercontent.com/fei0810/image-host/master/img/20190628083754.png" alt="扫码即刻沟通"></figure>

          </div>
          
            
  
            
              <div class="mt-6">
                <link rel="stylesheet" href="https://unpkg.com/disqusjs@1.1/dist/disqusjs.css">
<script src="https://unpkg.com/disqusjs@1.1/dist/disqus.js"></script>

<div id="disqus_thread"></div>

<script>

var options = {
  shortname: 'kaopubear',
  apikey: 'S3aqfVwpSDyKQ2A8Zpg9bXlFmZue0eFVbb3JWkNZtefxqCH39Fd2cLkwP7X0MPDi',
}
if ('https://disqus.skk.moe/disqus/') {
  options.api = 'https://disqus.skk.moe/disqus/'
}
var dsqjs = new DisqusJS(options)

</script>

              </div>
            
          
        </div>
        
          <div class="w-full md:w-1/3">
            <div class="mt-6 md:pl-8">
              <h4 class="text-base font-thin text-gray-700 mb-2">发布时间</h4>
              <div class="text-lg mb-6">2019-04-11</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">标签</h4>
              <div class="mb-6">
                
                  <a href="https://fei0810.github.io/tag/NJiuPskcxN" class="py-1 px-4 border-gray-300 border text-sm inline-block text-gray-700 hover:text-gray-900 rounded mr-2 mb-2">文献</a>
                
              </div>
              <h4 class="text-base font-thin text-gray-700 mb-2">阅读时间</h4>
              <div class="text-lg mb-6">11 min read</div>
              <h4 class="text-base font-thin text-gray-700 mb-2">相关阅读</h4>
              <div class="border-b border-gray-400">
                

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-09-11-howtolearnzotero" class="text-lg hover:underline my-2 inline-block">Zotero入门学习路径</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-06-22-weeklypaper" class="text-lg hover:underline my-2 inline-block">每周文献-190622-ATAC-seq和转录本从头组装新方法</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
        
        <div class="border-t border-gray-400 py-6">
          <a href="https://fei0810.github.io/post/2019-06-06-weeklypaper" class="text-lg hover:underline my-2 inline-block">每周文献-190606-多篇结构变异和转录组分析方法文章</a>
        </div>
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  

  
    
      
    
  


              </div>
            </div>
          </div>
        
      </div>
      <div class="footer border-gray-400 border-t pt-6 mt-6 sm:mt-8 md:mt-10 mb-20 text-gray-700 text-sm sm:text-base lg:text-lg">
  Powered by <a href="https://github.com/getgridea/gridea" target="_blank">Gridea</a> 我的 <a href="https://t.zsxq.com/UjiEaYZ"target="_blank">知识星球</a>
</div>
    </div>
  </div>
  <script>
    Prism.highlightAll()
  </script>
</body>
</html>